<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Ben Umans" />

<meta name="date" content="2021-07-09" />

<title>combine_experiments</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Organoid_Stress_Pilot</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/bumans/Organoid_Stress_Pilot">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">combine_experiments</h1>
<h4 class="author">Ben Umans</h4>
<h4 class="date">2021-07-09</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-07-31
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Organoid_Stress_Pilot/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210705code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210705)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210705code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210705)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombumansOrganoidStressPilottree5d8afa5b0840cd65c8111901f4455b45b85abe82targetblank5d8afa5a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bumans/Organoid_Stress_Pilot/tree/5d8afa5b0840cd65c8111901f4455b45b85abe82" target="_blank">5d8afa5</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombumansOrganoidStressPilottree5d8afa5b0840cd65c8111901f4455b45b85abe82targetblank5d8afa5a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/bumans/Organoid_Stress_Pilot/tree/5d8afa5b0840cd65c8111901f4455b45b85abe82" target="_blank">5d8afa5</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    data/.DS_Store
    Ignored:    data/fastq/
    Ignored:    output/.DS_Store
    Ignored:    output/YG-BU-01_human/
    Ignored:    output/YG-BU-02_human/
    Ignored:    output/YG-BU-03_human/
    Ignored:    output/YG-BU-04_human/
    Ignored:    output/YG-BU-05_human/
    Ignored:    output/YG-BU-06_human/
    Ignored:    output/YG-BU-07_human/
    Ignored:    output/YG-BU-08_human/
    Ignored:    output/YG-BU-09_human/
    Ignored:    output/YG-BU-10_human/
    Ignored:    output/YG-BU-11_human/
    Ignored:    output/merged_full_dataset.RDS
    Ignored:    output/organoid.combined.full.sct.neuron.RDS
    Ignored:    output/organoid.combined.sct.neuron.markers.RDS
    Ignored:    output/organoid.combined.sct03.RDS
    Ignored:    output/organoid.combined.sct05.RDS
    Ignored:    output/organoid.combined.sct1.RDS

Untracked files:
    Untracked:  data/Snakefile_cellranger
    Untracked:  data/all.vcf.bed
    Untracked:  data/australia.rds
    Untracked:  data/cellranger.38.dict
    Untracked:  data/cluster.json
    Untracked:  data/config.yaml
    Untracked:  data/environment.yaml
    Untracked:  data/exclude_fulldata.rds
    Untracked:  data/human.unsorted.vcf
    Untracked:  data/human.vcf
    Untracked:  data/human.vcf.bed
    Untracked:  data/library9.rds
    Untracked:  data/log/
    Untracked:  data/mysterycell.rds
    Untracked:  data/organoid.combined.sct03.neuron.RDS
    Untracked:  data/organoid_combined_sct03.RDS
    Untracked:  data/reheaded.vcf
    Untracked:  data/snakelog.out
    Untracked:  data/snakemake_cellranger.batch
    Untracked:  data/sorted.vcf
    Untracked:  data/submit.sh
    Untracked:  data/unsorted.vcf
    Untracked:  data/unsorted_header
    Untracked:  data_S2/

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/index.Rmd
    Modified:   analysis/precheck.Rmd
    Modified:   analysis/remove-mesenchyme.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/combine_experiments.Rmd</code>) and HTML (<code>docs/combine_experiments.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bumans/Organoid_Stress_Pilot/blob/5d8afa5b0840cd65c8111901f4455b45b85abe82/analysis/combine_experiments.Rmd" target="_blank">5d8afa5</a>
</td>
<td>
Ben Umans
</td>
<td>
2021-07-30
</td>
<td>
Building all files for initial upload
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bumans/Organoid_Stress_Pilot/blob/38cb9d0d3b9e36dc18d5a14d0717df09a7cc31c6/analysis/combine_experiments.Rmd" target="_blank">38cb9d0</a>
</td>
<td>
Ben Umans
</td>
<td>
2021-07-28
</td>
<td>
Committing to move things online
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bumans/Organoid_Stress_Pilot/blob/d97e0a55536a926b6cadf2c343ee2cd67b68e736/analysis/combine_experiments.Rmd" target="_blank">d97e0a5</a>
</td>
<td>
Ben Umans
</td>
<td>
2021-07-14
</td>
<td>
wflow_publish(files = c(“analysis/index.Rmd”, “analysis/combine_experiments.Rmd”,
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="combining-demultiplexed-seurat-objects" class="section level2">
<h2>Combining demultiplexed Seurat objects</h2>
<p>The data here were collected on three separate days, from three separate replicates, each consisting of three individuals. Demuxlet has already resolved each library into its constituent individuals by genotype, and now I’d need to combine the various libraries in order to test for consistency.</p>
<pre class="r"><code>pacman::p_load(edgeR, variancePartition, BiocParallel, limma)
library(Seurat)
library(tidyverse)</code></pre>
<pre><code>── Attaching packages ────────────────────────────────── tidyverse 1.3.1 ──</code></pre>
<pre><code>✓ tibble  3.1.2     ✓ dplyr   1.0.7
✓ tidyr   1.1.3     ✓ stringr 1.4.0
✓ readr   1.4.0     ✓ forcats 0.5.1
✓ purrr   0.3.4     </code></pre>
<pre><code>── Conflicts ───────────────────────────────────── tidyverse_conflicts() ──
x purrr::accumulate()      masks foreach::accumulate()
x readr::col_factor()      masks scales::col_factor()
x dplyr::combine()         masks Biobase::combine(), BiocGenerics::combine()
x purrr::discard()         masks scales::discard()
x dplyr::filter()          masks stats::filter()
x dplyr::lag()             masks stats::lag()
x BiocGenerics::Position() masks ggplot2::Position(), base::Position()
x purrr::when()            masks foreach::when()</code></pre>
<pre class="r"><code>library(ggplot2)
library(RColorBrewer)
library(cowplot)</code></pre>
<pre><code>
********************************************************</code></pre>
<pre><code>Note: As of version 1.0.0, cowplot does not change the</code></pre>
<pre><code>  default ggplot2 theme anymore. To recover the previous</code></pre>
<pre><code>  behavior, execute:
  theme_set(theme_cowplot())</code></pre>
<pre><code>********************************************************</code></pre>
<pre class="r"><code>library(knitr)</code></pre>
<pre class="r"><code>control1 &lt;- readRDS(file = &quot;output/YG-BU-01_human/obj.rds&quot;)
#replace the names of all individuals with the actual line names
oldnames1 &lt;- control1$individual
newnames1 &lt;- str_replace(oldnames1, &quot;SCM-10&quot;, &quot;28126&quot;) %&gt;% str_replace(&quot;SCM-12&quot;, &quot;20157&quot;) %&gt;% str_replace(&quot;SCM-13&quot;, &quot;23555&quot;)
control1 &lt;- AddMetaData(control1, newnames1, col.name = &quot;cell.line&quot;)
control1$replicate &lt;- &quot;2&quot;
control1$stim &lt;- &quot;control&quot;


oxid1 &lt;- readRDS(file = &quot;output/YG-BU-02_human/obj.rds&quot;)
#replace the names of all individuals with the actual line names
oldnames2 &lt;- oxid1$individual
newnames2 &lt;- str_replace(oldnames2, &quot;SCM-10&quot;, &quot;28126&quot;) %&gt;% str_replace(&quot;SCM-12&quot;, &quot;20157&quot;) %&gt;% str_replace(&quot;SCM-13&quot;, &quot;23555&quot;)
oxid1 &lt;- AddMetaData(oxid1, newnames2, col.name = &quot;cell.line&quot;)
oxid1$replicate &lt;- &quot;2&quot;
oxid1$stim &lt;- &quot;oxidation&quot;

pff1 &lt;- readRDS(file = &quot;output/YG-BU-03_human/obj.rds&quot;)
#replace the names of all individuals with the actual line names
oldnames3 &lt;- pff1$individual
newnames3 &lt;- str_replace(oldnames3, &quot;SCM-10&quot;, &quot;28126&quot;) %&gt;% str_replace(&quot;SCM-12&quot;, &quot;20157&quot;) %&gt;% str_replace(&quot;SCM-13&quot;, &quot;23555&quot;)
pff1 &lt;- AddMetaData(pff1, newnames3, col.name = &quot;cell.line&quot;)
pff1$replicate &lt;- &quot;2&quot;
pff1$stim &lt;- &quot;pff&quot;

control2 &lt;- readRDS(file = &quot;output/YG-BU-06_human/obj.rds&quot;)
#replace the names of all individuals with the actual line names
oldnames6 &lt;- control2$individual
newnames6 &lt;- str_replace(oldnames6, &quot;SCM-10&quot;, &quot;28126&quot;) %&gt;% str_replace(&quot;SCM-12&quot;, &quot;20157&quot;) %&gt;% str_replace(&quot;SCM-13&quot;, &quot;23555&quot;)
control2 &lt;- AddMetaData(control2, newnames6, col.name = &quot;cell.line&quot;)
control2$replicate &lt;- &quot;3&quot;
control2$stim &lt;- &quot;control&quot;


oxid2 &lt;- readRDS(file = &quot;output/YG-BU-07_human/obj.rds&quot;)
#replace the names of all individuals with the actual line names
oldnames7 &lt;- oxid2$individual
newnames7 &lt;- str_replace(oldnames7, &quot;SCM-10&quot;, &quot;28126&quot;) %&gt;% str_replace(&quot;SCM-12&quot;, &quot;20157&quot;) %&gt;% str_replace(&quot;SCM-13&quot;, &quot;23555&quot;)
oxid2 &lt;- AddMetaData(oxid2, newnames7, col.name = &quot;cell.line&quot;)
oxid2$replicate &lt;- &quot;3&quot;
oxid2$stim &lt;- &quot;oxidation&quot;


control3 &lt;- readRDS(file = &quot;output/YG-BU-09_human/obj.rds&quot;)
#replace the names of all individuals with the actual line names
oldnames9 &lt;- control3$individual
newnames9 &lt;- str_replace(oldnames9, &quot;SCM-10&quot;, &quot;28126&quot;) %&gt;% str_replace(&quot;SCM-12&quot;, &quot;20157&quot;) %&gt;% str_replace(&quot;SCM-13&quot;, &quot;23555&quot;)
control3 &lt;- AddMetaData(control3, newnames9, col.name = &quot;cell.line&quot;)
control3$replicate &lt;- &quot;4&quot;
control3$stim &lt;- &quot;control&quot;

oxid3 &lt;- readRDS(file = &quot;output/YG-BU-10_human/obj.rds&quot;)
#replace the names of all individuals with the actual line names
oldnames10 &lt;- oxid3$individual
newnames10 &lt;- str_replace(oldnames10, &quot;SCM-10&quot;, &quot;28126&quot;) %&gt;% str_replace(&quot;SCM-12&quot;, &quot;20157&quot;) %&gt;% str_replace(&quot;SCM-13&quot;, &quot;23555&quot;)
oxid3 &lt;- AddMetaData(oxid3, newnames10, col.name = &quot;cell.line&quot;)
oxid3$replicate &lt;- &quot;4&quot;
oxid3$stim &lt;- &quot;oxidation&quot;

pff3 &lt;- readRDS(file = &quot;output/YG-BU-11_human/obj.rds&quot;)
#replace the names of all individuals with the actual line names
oldnames11 &lt;- pff3$individual
newnames11 &lt;- str_replace(oldnames11, &quot;SCM-10&quot;, &quot;28126&quot;) %&gt;% str_replace(&quot;SCM-12&quot;, &quot;20157&quot;) %&gt;% str_replace(&quot;SCM-13&quot;, &quot;23555&quot;)
pff3 &lt;- AddMetaData(pff3, newnames11, col.name = &quot;cell.line&quot;)
pff3$replicate &lt;- &quot;4&quot;
pff3$stim &lt;- &quot;oxidation&quot;


combined.data &lt;- list(control1, control2, control3, oxid1, oxid2, oxid3, pff1, pff3)

rm(control1, control2, control3, oxid1, oxid2, oxid3, pff1, pff3)</code></pre>
<p>Because each of these datasets was prepared using <code>SCTransform</code> it should be straightforward to now integrate them. Following the <a href="https://satijalab.org/seurat/articles/integration_introduction.html#performing-integration-on-datasets-normalized-with-sctransform-1">guide</a> from the Satija lab for integrating these datasets, I run the following steps:</p>
<pre class="r"><code>features &lt;- SelectIntegrationFeatures(object.list = combined.data, nfeatures = 3000, verbose = F)
combined.data &lt;- PrepSCTIntegration(object.list = combined.data, anchor.features = features)

organoid.anchors &lt;- FindIntegrationAnchors(object.list = combined.data, normalization.method = &quot;SCT&quot;,
    anchor.features = features)
organoid.combined.sct &lt;- IntegrateData(anchorset = organoid.anchors, normalization.method = &quot;SCT&quot;)

rm(combined.data)</code></pre>
<p>With the combined data I now look run through the normal workflow to find clusters.</p>
<pre class="r"><code>organoid.combined.sct &lt;- RunPCA(organoid.combined.sct, npcs = 60)
organoid.combined.sct &lt;- RunUMAP(organoid.combined.sct, reduction = &quot;pca&quot;, dims = 1:45)

DimPlot(organoid.combined.sct, reduction = &quot;umap&quot;, group.by = &quot;stim&quot;)
DimPlot(organoid.combined.sct, reduction = &quot;umap&quot;, group.by = &quot;replicate&quot;)</code></pre>
<p>And now we can look for clusters:</p>
<pre class="r"><code>#organoid.combined.sct &lt;- FindNeighbors(organoid.combined.sct, dims=1:65)
organoid.combined.sct03 &lt;- readRDS(&quot;output/organoid.combined.sct03.RDS&quot;)
organoid.combined.sct05 &lt;- readRDS(&quot;output/organoid.combined.sct05.RDS&quot;)
organoid.combined.sct1 &lt;- readRDS(&quot;output/organoid.combined.sct1.RDS&quot;)

#organoid.combined.sct03 &lt;- FindClusters(organoid.combined.sct, resolution=0.3)
DimPlot(organoid.combined.sct03, reduction = &quot;umap&quot;) + ggtitle(&quot;resolution=0.3&quot;)</code></pre>
<p><img src="figure/combine_experiments.Rmd/integrated-clustering-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#organoid.combined.sct05 &lt;- FindClusters(organoid.combined.sct, resolution=0.5)
DimPlot(organoid.combined.sct05, reduction = &quot;umap&quot;) + ggtitle(&quot;resolution=0.5&quot;)</code></pre>
<p><img src="figure/combine_experiments.Rmd/integrated-clustering-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#organoid.combined.sct1 &lt;- FindClusters(organoid.combined.sct, resolution=1)
DimPlot(organoid.combined.sct1, reduction = &quot;umap&quot;) + ggtitle(&quot;resolution=1&quot;)</code></pre>
<p><img src="figure/combine_experiments.Rmd/integrated-clustering-3.png" width="672" style="display: block; margin: auto;" /> A lot of these possibilities <em>seem</em> reasonable. Can cluster markers give us any idea of which of them make sense to pursue?</p>
<pre class="r"><code>organoid.combined.sct03.markers &lt;- FindAllMarkers(organoid.combined.sct03, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
organoid.combined.sct03.markers %&gt;% group_by(cluster) %&gt;% top_n(n = 5, wt = avg_logFC) %&gt;% kable()

organoid.combined.sct05.markers &lt;- FindAllMarkers(organoid.combined.sct05, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
organoid.combined.sct05.markers %&gt;% group_by(cluster) %&gt;% top_n(n = 5, wt = avg_logFC) %&gt;% kable()

organoid.combined.sct1.markers &lt;- FindAllMarkers(organoid.combined.sct1, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
organoid.combined.sct1.markers %&gt;% group_by(cluster) %&gt;% top_n(n = 5, wt = avg_logFC) %&gt;% kable()</code></pre>
<div id="cell-health-or-otherwise" class="section level3">
<h3>Cell health, or otherwise</h3>
<p>Are these cells hopelessly dead or otherwise inviable? Let’s look at some markers of cell death and distress. I looked to <a href="https://www.cell.com/cell-stem-cell/pdf/S1934-5909(21)00017-5.pdf">Machado et al.</a> and <a href="https://www.nature.com/articles/nmeth.4437">van den Brink et al.</a> for some marker genes</p>
<pre class="r"><code>FeaturePlot(organoid.combined.sct03, features = c(&quot;sct_DAP&quot;), max.cutoff = 5, split.by = &quot;stim&quot;)</code></pre>
<p><img src="figure/combine_experiments.Rmd/cell-health-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(organoid.combined.sct03, features = c(&quot;sct_CASP1&quot;), max.cutoff = 5, split.by = &quot;stim&quot;)</code></pre>
<p><img src="figure/combine_experiments.Rmd/cell-health-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(organoid.combined.sct03, features = c(&quot;sct_BCL10&quot;), max.cutoff = 5, split.by = &quot;stim&quot;)</code></pre>
<p><img src="figure/combine_experiments.Rmd/cell-health-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(organoid.combined.sct03, features = c(&quot;sct_CARD10&quot;), max.cutoff = 5, split.by = &quot;stim&quot;)</code></pre>
<p><img src="figure/combine_experiments.Rmd/cell-health-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(organoid.combined.sct03, features = c(&quot;sct_HIF1A&quot;), max.cutoff = 5, split.by = &quot;stim&quot;)</code></pre>
<p><img src="figure/combine_experiments.Rmd/cell-health-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(organoid.combined.sct03, features = c(&quot;sct_ATF3&quot;), max.cutoff = 5, split.by = &quot;stim&quot;)</code></pre>
<p><img src="figure/combine_experiments.Rmd/cell-health-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(organoid.combined.sct03, features = c(&quot;sct_SOCS3&quot;), max.cutoff = 5, split.by = &quot;stim&quot;)</code></pre>
<p><img src="figure/combine_experiments.Rmd/cell-health-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(organoid.combined.sct03, features = c(&quot;sct_ZFP36&quot;), max.cutoff = 5, split.by = &quot;stim&quot;)</code></pre>
<p><img src="figure/combine_experiments.Rmd/cell-health-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(organoid.combined.sct03, features = c(&quot;sct_EGR1&quot;), max.cutoff = 5, split.by = &quot;stim&quot;)</code></pre>
<p><img src="figure/combine_experiments.Rmd/cell-health-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(organoid.combined.sct03, features = c(&quot;sct_KLF4&quot;), max.cutoff = 5, split.by = &quot;stim&quot;)</code></pre>
<p><img src="figure/combine_experiments.Rmd/cell-health-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(organoid.combined.sct03, features = c(&quot;sct_UBC&quot;), split.by = &quot;stim&quot;)</code></pre>
<p><img src="figure/combine_experiments.Rmd/cell-health-11.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(organoid.combined.sct03, features = c(&quot;sct_HSP90AB1&quot;),  split.by = &quot;stim&quot;)</code></pre>
<p><img src="figure/combine_experiments.Rmd/cell-health-12.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(organoid.combined.sct03, features = c(&quot;sct_STAT3&quot;), max.cutoff = 5, split.by = &quot;stim&quot;)</code></pre>
<p><img src="figure/combine_experiments.Rmd/cell-health-13.png" width="672" style="display: block; margin: auto;" /> While there are some genes that are expressed fairly widely, actually a good number of these stress markers appear to be enriched in the stress treatment conditions, and particularly enriched in the weird mesenchymal cluster. This is not really convincing or direct evidence that my neurons are “healthy” <em>per se,</em> but it’s a little reassuring and might even indicate that the treatments were (at least at the margin) effective.</p>
</div>
</div>
<div id="obtain-pseudobulk-counts" class="section level1">
<h1>Obtain Pseudobulk counts</h1>
<p>Now that we’ve assigned clusters to each cell from each data collection, we want to go back to get the raw data for use in differential expression testing. Kenneth wrote a code snippet to do this. His code takes a Seurat object and a list of all the desired metadata columns that you want to subdivide, and it returns a matrix of pseudobulked counts (gene rows, columns as the cross product of all the metadata factors specified) and list of metadata.</p>
<pre class="r"><code>generate.pseudobulk &lt;- function(object, labels, assay=&quot;RNA&quot;,slot=&quot;counts&quot;) {
  factorlist &lt;- list()
  for (i in labels) factorlist[[i]] &lt;- unique(object@meta.data[,i])
  meta &lt;- expand.grid(factorlist, stringsAsFactors = FALSE)
  rownames(meta) &lt;- apply(meta, 1, function(x) paste0(x, collapse = &#39;.&#39;))
  # build the output matrix
  n &lt;- nrow(meta)
  out &lt;- matrix(nrow=dim(object[[assay]])[1], ncol=n, data=0)
  rownames(out) &lt;- rownames(object[[assay]])
  colnames(out) &lt;- rownames(meta)
  ncells &lt;- c()
  total.cells &lt;- dim(object[[assay]])[2]
  for (i in 1:n)
  {
    #prog(i,n)
    cells &lt;- 1:total.cells
    for (j in names(meta)) {
      keep  &lt;- which(object@meta.data[[j]] == meta[i,j])
      cells &lt;- cells[cells %in% keep]
    }
    ncells[i] &lt;- length(cells)
    #some other thing to measure
    if (length(cells)==1) {
      out[,i] &lt;- slot(object[[assay]], slot)[,cells]
    } else {
      out[,i] &lt;- Matrix::rowSums(slot(object[[assay]], slot)[,cells])
    }
  }
  meta$ncells &lt;- ncells
  #add that something else as metadata
  return(list(counts=out, meta=meta))
}

# keep only levels with more than &#39;threshold&#39; cells
filter.pseudobulk &lt;- function(pseudobulk, threshold = 0) {
  w &lt;- which(pseudobulk$meta$ncells &gt; threshold)
  pseudobulk$counts &lt;- pseudobulk$counts[,w]
  pseudobulk$meta   &lt;- pseudobulk$meta[w,]
  pseudobulk
}</code></pre>
<pre class="r"><code>pseudo3 &lt;- generate.pseudobulk(organoid.combined.sct03, labels = c(&quot;cell.line&quot;, &quot;stim&quot;, &quot;replicate&quot;, &quot;seurat_clusters&quot;))
pseudo3 &lt;- filter.pseudobulk(pseudo3)

d03 &lt;- DGEList(pseudo3$counts)
d03 &lt;- calcNormFactors(d03)

keep3 &lt;- filterByExpr(d03$counts)

d03 &lt;- d03[keep3,]

param = SnowParam(20, &quot;SOCK&quot;, progressbar=TRUE)
register(param)

fbase &lt;- ~  (1|cell.line) + (1|stim) + (1|seurat_clusters) + (1|replicate)

voomed3 &lt;- voomWithDreamWeights(d03, fbase, pseudo3$meta, plot=TRUE)</code></pre>
<pre><code>Memory usage to store result: &gt; 208 Mb 
Dividing work into 100 chunks...

Total: 111 s</code></pre>
<p><img src="figure/combine_experiments.Rmd/resolution03-varpart-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>varpart3 &lt;- fitExtractVarPartModel(voomed3, fbase, pseudo3$meta)</code></pre>
<pre><code>Dividing work into 100 chunks...

Total: 137 s</code></pre>
<pre class="r"><code>vp3 &lt;- sortCols(varpart3)
plotPercentBars(vp3[1:10,])</code></pre>
<p><img src="figure/combine_experiments.Rmd/resolution03-varpart-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotVarPart(vp3) + ggtitle(&quot;Variance partition clustered resolution=0.3&quot;)</code></pre>
<p><img src="figure/combine_experiments.Rmd/resolution03-varpart-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pseudo5 &lt;- generate.pseudobulk(organoid.combined.sct05, labels = c(&quot;cell.line&quot;, &quot;stim&quot;, &quot;replicate&quot;, &quot;seurat_clusters&quot;))
pseudo5 &lt;- filter.pseudobulk(pseudo5)

d05 &lt;- DGEList(pseudo5$counts)
d05 &lt;- calcNormFactors(d05)

keep5 &lt;- filterByExpr(d05$counts)

d05 &lt;- d05[keep5,]


fbase &lt;- ~ (1|cell.line) + (1|stim) + (1|seurat_clusters) + (1|replicate)

voomed5 &lt;- voomWithDreamWeights(d05, fbase, pseudo5$meta, plot=TRUE)</code></pre>
<pre><code>Memory usage to store result: &gt; 236.2 Mb 
Dividing work into 100 chunks...

Total: 119 s</code></pre>
<p><img src="figure/combine_experiments.Rmd/resolution05-varpart-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>varpart5 &lt;- fitExtractVarPartModel(voomed5, fbase, pseudo5$meta)</code></pre>
<pre><code>Dividing work into 100 chunks...

Total: 149 s</code></pre>
<pre class="r"><code>vp5 &lt;- sortCols(varpart5)
plotPercentBars(vp5[1:10,])</code></pre>
<p><img src="figure/combine_experiments.Rmd/resolution05-varpart-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotVarPart(vp5) + ggtitle(&quot;Variance partition clustered resolution=0.5&quot;)</code></pre>
<p><img src="figure/combine_experiments.Rmd/resolution05-varpart-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pseudo1 &lt;- generate.pseudobulk(organoid.combined.sct1, labels = c(&quot;cell.line&quot;, &quot;stim&quot;, &quot;replicate&quot;, &quot;seurat_clusters&quot;))
pseudo1 &lt;- filter.pseudobulk(pseudo1)

d1 &lt;- DGEList(pseudo1$counts)
d1 &lt;- calcNormFactors(d1)

keep1 &lt;- filterByExpr(d1$counts)

d1 &lt;- d1[keep1,]

fbase &lt;- ~ (1|cell.line) + (1|stim) + (1|seurat_clusters) + (1|replicate)

voomed1 &lt;- voomWithDreamWeights(d1, fbase, pseudo1$meta, plot=TRUE)</code></pre>
<pre><code>Memory usage to store result: &gt; 254.5 Mb 
Dividing work into 100 chunks...

Total: 121 s</code></pre>
<p><img src="figure/combine_experiments.Rmd/resolution1-varpart-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>varpart1 &lt;- fitExtractVarPartModel(cpm(d1$counts, log=TRUE), fbase, pseudo1$meta)</code></pre>
<pre><code>Dividing work into 100 chunks...

Total: 117 s</code></pre>
<pre class="r"><code>vp1 &lt;- sortCols(varpart1)
plotPercentBars(vp1[1:10,])</code></pre>
<p><img src="figure/combine_experiments.Rmd/resolution1-varpart-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotVarPart(vp1) + ggtitle(&quot;Variance partition clustered resolution=1&quot;)</code></pre>
<p><img src="figure/combine_experiments.Rmd/resolution1-varpart-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>saveRDS(organoid.combined.sct1, file = &quot;output/organoid.combined.sct1.RDS&quot;)
saveRDS(organoid.combined.sct03, file = &quot;output/organoid.combined.sct03.RDS&quot;)
saveRDS(organoid.combined.sct05, file = &quot;output/organoid.combined.sct05.RDS&quot;)</code></pre>
</div>
<div id="summary-discussion-with-yoav" class="section level1">
<h1>Summary, discussion with Yoav</h1>
<p>This is not what we expect, not at all. Discussed with Kenneth and Yoav. Yoav suggested this is not the right way to analyze the data. First of all, because cluster composition and membership are likely confounded with cell line and treatment, the analysis should be done on each cluster separately. Maybe the right way to start is to use large manually selected clusters.</p>
<p>Second, getting at a similar issue, Yoav suggests trying to ascertain whether the mean expression values in each cluster are stable with regards to cell number by jacknifing to generate the pseudobulk data.</p>
<p>Finally, Yoav suggests comparing these variancePartition results to the results obtained by PCA. If these are consistent, then PC2 should correspond to cluster and PC1 should correspond to…something else. Can compare to the experimental meta-data to try to figure out what that is. If PC2 is not Seurat cluster, then something doesn’t add up.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] knitr_1.23               cowplot_1.0.0           
 [3] RColorBrewer_1.1-2       forcats_0.5.1           
 [5] stringr_1.4.0            dplyr_1.0.7             
 [7] purrr_0.3.4              readr_1.4.0             
 [9] tidyr_1.1.3              tibble_3.1.2            
[11] tidyverse_1.3.1          Seurat_3.1.3            
[13] BiocParallel_1.18.0      variancePartition_1.14.1
[15] Biobase_2.44.0           BiocGenerics_0.30.0     
[17] scales_1.1.1             foreach_1.5.1           
[19] ggplot2_3.3.5            edgeR_3.26.5            
[21] limma_3.40.6             workflowr_1.6.2         

loaded via a namespace (and not attached):
  [1] snow_0.4-3          readxl_1.3.1        backports_1.2.1    
  [4] plyr_1.8.6          igraph_1.2.4.1      lazyeval_0.2.2     
  [7] splines_3.6.1       listenv_0.7.0       digest_0.6.27      
 [10] htmltools_0.5.1.1   gdata_2.18.0        fansi_0.5.0        
 [13] magrittr_2.0.1      cluster_2.1.0       doParallel_1.0.14  
 [16] ROCR_1.0-7          globals_0.12.4      modelr_0.1.8       
 [19] RcppParallel_5.1.4  prettyunits_1.1.1   colorspace_2.0-2   
 [22] rvest_1.0.0         rappdirs_0.3.1      ggrepel_0.8.1      
 [25] haven_2.3.1         xfun_0.24           crayon_1.4.1       
 [28] jsonlite_1.7.2      lme4_1.1-21         survival_3.2-11    
 [31] zoo_1.8-6           iterators_1.0.13    ape_5.4-1          
 [34] glue_1.4.2          gtable_0.3.0        leiden_0.3.1       
 [37] future.apply_1.3.0  DBI_1.1.0           bibtex_0.4.2       
 [40] Rcpp_1.0.6          metap_1.1           viridisLite_0.3.0  
 [43] progress_1.2.2      reticulate_1.16     rsvd_1.0.1         
 [46] tsne_0.1-3          htmlwidgets_1.5.3   httr_1.4.2         
 [49] gplots_3.0.1.1      ellipsis_0.3.2      ica_1.0-2          
 [52] farver_2.1.0        pkgconfig_2.0.3     uwot_0.1.5         
 [55] dbplyr_2.1.1        locfit_1.5-9.1      utf8_1.1.4         
 [58] labeling_0.4.2      tidyselect_1.1.0    rlang_0.4.11       
 [61] reshape2_1.4.4      later_1.2.0         cellranger_1.1.0   
 [64] munsell_0.5.0       tools_3.6.1         cli_2.5.0          
 [67] generics_0.1.0      pacman_0.5.1        broom_0.7.8        
 [70] ggridges_0.5.1      evaluate_0.14       yaml_2.2.1         
 [73] npsurv_0.4-0        fs_1.3.1            fitdistrplus_1.0-14
 [76] caTools_1.17.1.2    RANN_2.6.1          pbapply_1.4-0      
 [79] future_1.14.0       nlme_3.1-140        whisker_0.3-2      
 [82] xml2_1.3.2          rstudioapi_0.13     compiler_3.6.1     
 [85] pbkrtest_0.4-7      plotly_4.9.4.1      png_0.1-7          
 [88] lsei_1.2-0          reprex_2.0.0        stringi_1.6.2      
 [91] lattice_0.20-38     Matrix_1.2-18       nloptr_1.2.2.1     
 [94] vctrs_0.3.8         pillar_1.6.1        lifecycle_1.0.0    
 [97] Rdpack_0.11-0       lmtest_0.9-37       RcppAnnoy_0.0.12   
[100] data.table_1.14.0   bitops_1.0-6        irlba_2.3.3        
[103] gbRd_0.4-11         httpuv_1.6.1        colorRamps_2.3     
[106] R6_2.5.0            promises_1.2.0.1    KernSmooth_2.23-15 
[109] gridExtra_2.3       codetools_0.2-16    boot_1.3-23        
[112] MASS_7.3-51.4       gtools_3.8.1        assertthat_0.2.1   
[115] rprojroot_2.0.2     withr_2.4.2         sctransform_0.2.0  
[118] hms_1.1.0           grid_3.6.1          minqa_1.2.4        
[121] rmarkdown_1.13      Rtsne_0.15          git2r_0.26.1       
[124] lubridate_1.7.10   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
