<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Ben Umans" />

<meta name="date" content="2021-07-30" />

<title>de_full</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Organoid_Stress_Pilot</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/bumans/Organoid_Stress_Pilot">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">de_full</h1>
<h4 class="author">Ben Umans</h4>
<h4 class="date">2021-07-30</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-07-30
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Organoid_Stress_Pilot/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210705code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210705)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210705code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210705)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombumansOrganoidStressPilottree5d8afa5b0840cd65c8111901f4455b45b85abe82targetblank5d8afa5a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bumans/Organoid_Stress_Pilot/tree/5d8afa5b0840cd65c8111901f4455b45b85abe82" target="_blank">5d8afa5</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombumansOrganoidStressPilottree5d8afa5b0840cd65c8111901f4455b45b85abe82targetblank5d8afa5a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/bumans/Organoid_Stress_Pilot/tree/5d8afa5b0840cd65c8111901f4455b45b85abe82" target="_blank">5d8afa5</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    analysis/figure/
    Ignored:    data/.DS_Store
    Ignored:    data/fastq/
    Ignored:    output/.DS_Store
    Ignored:    output/YG-BU-01_human/
    Ignored:    output/YG-BU-02_human/
    Ignored:    output/YG-BU-03_human/
    Ignored:    output/YG-BU-04_human/
    Ignored:    output/YG-BU-05_human/
    Ignored:    output/YG-BU-06_human/
    Ignored:    output/YG-BU-07_human/
    Ignored:    output/YG-BU-08_human/
    Ignored:    output/YG-BU-09_human/
    Ignored:    output/YG-BU-10_human/
    Ignored:    output/YG-BU-11_human/
    Ignored:    output/merged_full_dataset.RDS
    Ignored:    output/organoid.combined.full.sct.neuron.RDS
    Ignored:    output/organoid.combined.sct.neuron.markers.RDS
    Ignored:    output/organoid.combined.sct03.RDS
    Ignored:    output/organoid.combined.sct05.RDS
    Ignored:    output/organoid.combined.sct1.RDS

Untracked files:
    Untracked:  data/Snakefile_cellranger
    Untracked:  data/all.vcf.bed
    Untracked:  data/australia.rds
    Untracked:  data/cellranger.38.dict
    Untracked:  data/cluster.json
    Untracked:  data/config.yaml
    Untracked:  data/environment.yaml
    Untracked:  data/exclude_fulldata.rds
    Untracked:  data/human.unsorted.vcf
    Untracked:  data/human.vcf
    Untracked:  data/human.vcf.bed
    Untracked:  data/library9.rds
    Untracked:  data/log/
    Untracked:  data/mysterycell.rds
    Untracked:  data/organoid.combined.sct03.neuron.RDS
    Untracked:  data/organoid_combined_sct03.RDS
    Untracked:  data/reheaded.vcf
    Untracked:  data/snakelog.out
    Untracked:  data/snakemake_cellranger.batch
    Untracked:  data/sorted.vcf
    Untracked:  data/submit.sh
    Untracked:  data/unsorted.vcf
    Untracked:  data/unsorted_header
    Untracked:  data_S2/

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/index.Rmd
    Modified:   analysis/precheck.Rmd
    Modified:   analysis/remove-mesenchyme.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/de_full.Rmd</code>) and HTML (<code>docs/de_full.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bumans/Organoid_Stress_Pilot/blob/5d8afa5b0840cd65c8111901f4455b45b85abe82/analysis/de_full.Rmd" target="_blank">5d8afa5</a>
</td>
<td>
Ben Umans
</td>
<td>
2021-07-30
</td>
<td>
Building all files for initial upload
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>I will now test for differential expression in each cluster independently, using <code>dream</code> on the combined sequencing data. I will do this on both the neuron-specific dataset and on the less filtered dataset, since I have the suspicion that some of the non-neuronal cells being removed might be interesting.</p>
<pre class="r"><code>pacman::p_load(edgeR, variancePartition, BiocParallel, limma)
library(Seurat)
library(tidyverse)</code></pre>
<pre><code>── Attaching packages ────────────────────────────────── tidyverse 1.3.1 ──</code></pre>
<pre><code>✓ tibble  3.1.2     ✓ dplyr   1.0.7
✓ tidyr   1.1.3     ✓ stringr 1.4.0
✓ readr   1.4.0     ✓ forcats 0.5.1
✓ purrr   0.3.4     </code></pre>
<pre><code>── Conflicts ───────────────────────────────────── tidyverse_conflicts() ──
x purrr::accumulate()      masks foreach::accumulate()
x readr::col_factor()      masks scales::col_factor()
x dplyr::combine()         masks Biobase::combine(), BiocGenerics::combine()
x purrr::discard()         masks scales::discard()
x dplyr::filter()          masks stats::filter()
x dplyr::lag()             masks stats::lag()
x BiocGenerics::Position() masks ggplot2::Position(), base::Position()
x purrr::when()            masks foreach::when()</code></pre>
<pre class="r"><code>library(ggplot2)
library(RColorBrewer)
library(cowplot)</code></pre>
<pre><code>
********************************************************</code></pre>
<pre><code>Note: As of version 1.0.0, cowplot does not change the</code></pre>
<pre><code>  default ggplot2 theme anymore. To recover the previous</code></pre>
<pre><code>  behavior, execute:
  theme_set(theme_cowplot())</code></pre>
<pre><code>********************************************************</code></pre>
<pre class="r"><code>library(knitr)
Sys.setenv(TZ=&#39;America/Chicago&#39;)

generate.pseudobulk &lt;- function(object, labels, assay=&quot;RNA&quot;,slot=&quot;counts&quot;) {
  factorlist &lt;- list()
  for (i in labels) factorlist[[i]] &lt;- unique(object@meta.data[,i])
  meta &lt;- expand.grid(factorlist, stringsAsFactors = FALSE)
  rownames(meta) &lt;- apply(meta, 1, function(x) paste0(x, collapse = &#39;.&#39;))
  # build the output matrix
  n &lt;- nrow(meta)
  out &lt;- matrix(nrow=dim(object[[assay]])[1], ncol=n, data=0)
  rownames(out) &lt;- rownames(object[[assay]])
  colnames(out) &lt;- rownames(meta)
  ncells &lt;- c()
  ncounts &lt;- c()
  total.cells &lt;- dim(object[[assay]])[2]
  for (i in 1:n)
  {
    #prog(i,n)
    cells &lt;- 1:total.cells
    for (j in names(meta)) {
      keep  &lt;- which(object@meta.data[[j]] == meta[i,j])
      cells &lt;- cells[cells %in% keep]
    }
    ncells[i] &lt;- length(cells)
    ncounts[i] &lt;- sum(slot(object[[assay]], slot)[,cells])
    #some other thing to measure
    if (length(cells)==1) {
      out[,i] &lt;- slot(object[[assay]], slot)[,cells]
    } else {
      out[,i] &lt;- Matrix::rowSums(slot(object[[assay]], slot)[,cells])
    }
  }
  meta$ncells &lt;- ncells
  meta$ncounts &lt;- ncounts/max(ncounts)
  #add that something else as metadata
  return(list(counts=out, meta=meta))
}

# keep only levels with more than &#39;threshold&#39; cells
filter.pseudobulk &lt;- function(pseudobulk, threshold = 0) {
  w &lt;- which(pseudobulk$meta$ncells &gt; threshold)
  pseudobulk$counts &lt;- pseudobulk$counts[,w]
  pseudobulk$meta &lt;- pseudobulk$meta[w,]
  pseudobulk
}

organoid.combined.sct &lt;- readRDS(file = &quot;output/merged_full_dataset.RDS&quot;)
organoid.combined.sct.neuron &lt;- readRDS(file = &quot;output/organoid.combined.full.sct.neuron.RDS&quot;)</code></pre>
<p>The neuronal dataset has 12 clusters and the full dataset has 17 clusters, as identified by very conservative (resolution 0.1) clustering in Seurat. I start by splitting each dataset into its clusters.</p>
<pre class="r"><code>cluster0.full &lt;- subset(organoid.combined.sct, subset = seurat_clusters == &#39;0&#39;)
cluster1.full &lt;- subset(organoid.combined.sct, subset = seurat_clusters == &#39;1&#39;)
cluster2.full &lt;- subset(organoid.combined.sct, subset = seurat_clusters == &#39;2&#39;)
cluster3.full &lt;- subset(organoid.combined.sct, subset = seurat_clusters == &#39;3&#39;)
cluster4.full &lt;- subset(organoid.combined.sct, subset = seurat_clusters == &#39;4&#39;)
cluster5.full &lt;- subset(organoid.combined.sct, subset = seurat_clusters == &#39;5&#39;)
cluster6.full &lt;- subset(organoid.combined.sct, subset = seurat_clusters == &#39;6&#39;)
cluster7.full &lt;- subset(organoid.combined.sct, subset = seurat_clusters == &#39;7&#39;)
cluster8.full &lt;- subset(organoid.combined.sct, subset = seurat_clusters == &#39;8&#39;)
cluster9.full &lt;- subset(organoid.combined.sct, subset = seurat_clusters == &#39;9&#39;)
cluster10.full &lt;- subset(organoid.combined.sct, subset = seurat_clusters == &#39;10&#39;)
cluster11.full &lt;- subset(organoid.combined.sct, subset = seurat_clusters == &#39;11&#39;)
cluster12.full &lt;- subset(organoid.combined.sct, subset = seurat_clusters == &#39;12&#39;)
cluster13.full &lt;- subset(organoid.combined.sct, subset = seurat_clusters == &#39;13&#39;)
cluster14.full &lt;- subset(organoid.combined.sct, subset = seurat_clusters == &#39;14&#39;)
cluster15.full &lt;- subset(organoid.combined.sct, subset = seurat_clusters == &#39;15&#39;)
cluster16.full &lt;- subset(organoid.combined.sct, subset = seurat_clusters == &#39;16&#39;)
rm(organoid.combined.sct)

cluster0.neuron &lt;- subset(organoid.combined.sct.neuron, subset = seurat_clusters == &#39;0&#39;)
cluster1.neuron &lt;- subset(organoid.combined.sct.neuron, subset = seurat_clusters == &#39;1&#39;)
cluster2.neuron &lt;- subset(organoid.combined.sct.neuron, subset = seurat_clusters == &#39;2&#39;)
cluster3.neuron &lt;- subset(organoid.combined.sct.neuron, subset = seurat_clusters == &#39;3&#39;)
cluster4.neuron &lt;- subset(organoid.combined.sct.neuron, subset = seurat_clusters == &#39;4&#39;)
cluster5.neuron &lt;- subset(organoid.combined.sct.neuron, subset = seurat_clusters == &#39;5&#39;)
cluster6.neuron &lt;- subset(organoid.combined.sct.neuron, subset = seurat_clusters == &#39;6&#39;)
cluster7.neuron &lt;- subset(organoid.combined.sct.neuron, subset = seurat_clusters == &#39;7&#39;)
cluster8.neuron &lt;- subset(organoid.combined.sct.neuron, subset = seurat_clusters == &#39;8&#39;)
cluster9.neuron &lt;- subset(organoid.combined.sct.neuron, subset = seurat_clusters == &#39;9&#39;)
cluster10.neuron &lt;- subset(organoid.combined.sct.neuron, subset = seurat_clusters == &#39;10&#39;)
cluster11.neuron &lt;- subset(organoid.combined.sct.neuron, subset = seurat_clusters == &#39;11&#39;)
rm(organoid.combined.sct.neuron)</code></pre>
<p>Set up a table to keep track of how many DE genes we find.</p>
<pre class="r"><code>de.summary.full &lt;- matrix(0, nrow = 17, ncol = 2)
colnames(de.summary.full) &lt;- c(&quot;oxidation&quot;, &quot;pff&quot;)
rownames(de.summary.full) &lt;- c(&quot;cluster0&quot;,&quot;cluster1&quot;,
                          &quot;cluster2&quot;,&quot;cluster3&quot;,
                          &quot;cluster4&quot;,&quot;cluster5&quot;,
                          &quot;cluster6&quot;,&quot;cluster7&quot;,
                          &quot;cluster8&quot;,&quot;cluster9&quot;,
                          &quot;cluster10&quot;,&quot;cluster11&quot;,
                          &quot;cluster12&quot;,&quot;cluster13&quot;,
                          &quot;cluster14&quot;,&quot;cluster15&quot;,
                          &quot;cluster16&quot;)

de.summary.neuron &lt;- matrix(0, nrow = 12, ncol = 2)
colnames(de.summary.neuron) &lt;- c(&quot;oxidation&quot;, &quot;pff&quot;)
rownames(de.summary.neuron) &lt;- c(&quot;cluster0&quot;,&quot;cluster1&quot;,
                          &quot;cluster2&quot;,&quot;cluster3&quot;,
                          &quot;cluster4&quot;,&quot;cluster5&quot;,
                          &quot;cluster6&quot;,&quot;cluster7&quot;,
                          &quot;cluster8&quot;,&quot;cluster9&quot;,
                          &quot;cluster10&quot;,&quot;cluster11&quot;)

param = SnowParam(20, &quot;SOCK&quot;, progressbar=TRUE)
register(param)
formde &lt;- ~  stim + (1|cell.line) + (1|replicate)</code></pre>
<p>Now we can use <code>dream</code> to look for differential expression using a repeated-measures model. For each cluster we can thank count the number of DE genes (FDR 0.05) for each control vs. stimulus comparison.</p>
<p>For cluster 0:</p>
<pre class="r"><code>pseudo0.full &lt;- generate.pseudobulk(cluster0.full, labels = c(&quot;stim&quot;, &quot;replicate&quot;, &quot;cell.line&quot;))
pseudo0.full &lt;- filter.pseudobulk(pseudo0.full)
d0.full &lt;- DGEList(pseudo0.full$counts)
d0.full &lt;- calcNormFactors(d0.full)
keep0.full &lt;- filterByExpr(d0.full$counts)
d0.full &lt;- d0.full[keep0.full,]

voomed0.full &lt;- voomWithDreamWeights(d0.full, formde, pseudo0.full$meta, plot=FALSE)</code></pre>
<pre><code>Memory usage to store result: &gt; 27.8 Mb 
Dividing work into 100 chunks...

Total: 83 s</code></pre>
<pre class="r"><code>fitmm0.full = dream(voomed0.full, formde, pseudo0.full$meta)</code></pre>
<pre><code>Dividing work into 100 chunks...

Total: 113 s</code></pre>
<pre class="r"><code>try(topTable(fitmm0.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.full[1,1] &lt;- try(topTable(fitmm0.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.full[1, 1] &lt;- try(topTable(fitmm0.full, coef = &quot;stimoxidation&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm0.full, coef=&#39;stimoxidation&#39;,  number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;Oxidation DE genes, cluster 0 of full dataset&quot;)</code></pre>
<table>
<caption>Oxidation DE genes, cluster 0 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>KDM3A</td>
<td align="right">-0.7673091</td>
<td align="right">5.602707</td>
<td align="right">-6.477892</td>
<td align="right">0.0000161</td>
<td align="right">0.1463350</td>
<td align="right">-4.312468</td>
</tr>
<tr class="even">
<td>MT-ND2</td>
<td align="right">0.5266483</td>
<td align="right">10.211493</td>
<td align="right">6.385952</td>
<td align="right">0.0000238</td>
<td align="right">0.1463350</td>
<td align="right">4.226235</td>
</tr>
<tr class="odd">
<td>CAV3</td>
<td align="right">-0.4996052</td>
<td align="right">3.592960</td>
<td align="right">-5.503474</td>
<td align="right">0.0000610</td>
<td align="right">0.2503909</td>
<td align="right">-4.008941</td>
</tr>
<tr class="even">
<td>MT-ND3</td>
<td align="right">0.4197293</td>
<td align="right">11.036739</td>
<td align="right">4.833743</td>
<td align="right">0.0002171</td>
<td align="right">0.4173653</td>
<td align="right">3.698226</td>
</tr>
<tr class="odd">
<td>C19orf44</td>
<td align="right">-0.4965437</td>
<td align="right">2.094527</td>
<td align="right">-4.786986</td>
<td align="right">0.0002370</td>
<td align="right">0.4173653</td>
<td align="right">-3.675943</td>
</tr>
<tr class="even">
<td>GYS1</td>
<td align="right">-0.3175370</td>
<td align="right">4.059502</td>
<td align="right">-4.731011</td>
<td align="right">0.0002424</td>
<td align="right">0.4173653</td>
<td align="right">-3.670204</td>
</tr>
<tr class="odd">
<td>LRP2BP</td>
<td align="right">-0.5509650</td>
<td align="right">3.827562</td>
<td align="right">-4.848434</td>
<td align="right">0.0003047</td>
<td align="right">0.4173653</td>
<td align="right">-3.611280</td>
</tr>
<tr class="even">
<td>DDX1</td>
<td align="right">0.1535710</td>
<td align="right">6.843366</td>
<td align="right">4.846704</td>
<td align="right">0.0003466</td>
<td align="right">0.4173653</td>
<td align="right">3.577692</td>
</tr>
<tr class="odd">
<td>VEGFA</td>
<td align="right">-1.0903778</td>
<td align="right">5.284983</td>
<td align="right">-4.437174</td>
<td align="right">0.0003612</td>
<td align="right">0.4173653</td>
<td align="right">-3.566915</td>
</tr>
<tr class="even">
<td>RLF</td>
<td align="right">-0.5341345</td>
<td align="right">6.745944</td>
<td align="right">-4.357912</td>
<td align="right">0.0004282</td>
<td align="right">0.4173653</td>
<td align="right">-3.522074</td>
</tr>
</tbody>
</table>
<pre class="r"><code>try(topTable(fitmm0.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.full[1,2] &lt;- try(topTable(fitmm0.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.full[1, 2] &lt;- try(topTable(fitmm0.full, coef = &quot;stimpff&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm0.full, coef=&#39;stimpff&#39;, number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;PFF DE genes, cluster 0 of full dataset&quot;)</code></pre>
<table>
<caption>PFF DE genes, cluster 0 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>AC068888.1</td>
<td align="right">0.5800141</td>
<td align="right">3.893120</td>
<td align="right">6.351180</td>
<td align="right">0.0000187</td>
<td align="right">0.2305727</td>
<td align="right">4.279618</td>
</tr>
<tr class="even">
<td>FZD6</td>
<td align="right">0.6922930</td>
<td align="right">3.253936</td>
<td align="right">4.464516</td>
<td align="right">0.0003982</td>
<td align="right">0.9862970</td>
<td align="right">3.541259</td>
</tr>
<tr class="odd">
<td>DNAJC9</td>
<td align="right">-0.4176203</td>
<td align="right">5.404620</td>
<td align="right">-4.379275</td>
<td align="right">0.0006632</td>
<td align="right">0.9862970</td>
<td align="right">-3.404341</td>
</tr>
<tr class="even">
<td>PWWP2A</td>
<td align="right">0.3127251</td>
<td align="right">5.775753</td>
<td align="right">4.124024</td>
<td align="right">0.0007089</td>
<td align="right">0.9862970</td>
<td align="right">3.386107</td>
</tr>
<tr class="odd">
<td>IFFO1</td>
<td align="right">-0.6038332</td>
<td align="right">2.773183</td>
<td align="right">-4.196973</td>
<td align="right">0.0007613</td>
<td align="right">0.9862970</td>
<td align="right">-3.366477</td>
</tr>
<tr class="even">
<td>RHNO1</td>
<td align="right">-0.4997454</td>
<td align="right">3.473608</td>
<td align="right">-3.993130</td>
<td align="right">0.0013614</td>
<td align="right">0.9862970</td>
<td align="right">-3.202702</td>
</tr>
<tr class="odd">
<td>TUBGCP3</td>
<td align="right">0.2272045</td>
<td align="right">5.291938</td>
<td align="right">3.928263</td>
<td align="right">0.0013787</td>
<td align="right">0.9862970</td>
<td align="right">3.199072</td>
</tr>
<tr class="even">
<td>PFKFB4</td>
<td align="right">0.5462308</td>
<td align="right">3.822067</td>
<td align="right">3.851347</td>
<td align="right">0.0015399</td>
<td align="right">0.9862970</td>
<td align="right">3.167055</td>
</tr>
<tr class="odd">
<td>PLCG1-AS1</td>
<td align="right">1.0906229</td>
<td align="right">1.871539</td>
<td align="right">3.802708</td>
<td align="right">0.0015741</td>
<td align="right">0.9862970</td>
<td align="right">3.160663</td>
</tr>
<tr class="even">
<td>UNG</td>
<td align="right">0.3492933</td>
<td align="right">3.524614</td>
<td align="right">3.868770</td>
<td align="right">0.0017298</td>
<td align="right">0.9862970</td>
<td align="right">3.133084</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#now repeat with the neuron dataset
pseudo0.neuron &lt;- generate.pseudobulk(cluster0.neuron, labels = c(&quot;stim&quot;, &quot;replicate&quot;, &quot;cell.line&quot;))
pseudo0.neuron &lt;- filter.pseudobulk(pseudo0.neuron)
d0.neuron &lt;- DGEList(pseudo0.neuron$counts)
d0.neuron &lt;- calcNormFactors(d0.neuron)
keep0.neuron &lt;- filterByExpr(d0.neuron$counts)
d0.neuron &lt;- d0.neuron[keep0.neuron,]

voomed0.neuron &lt;- voomWithDreamWeights(d0.neuron, formde, pseudo0.neuron$meta, plot=FALSE)</code></pre>
<pre><code>Memory usage to store result: &gt; 28 Mb 
Dividing work into 100 chunks...

Total: 83 s</code></pre>
<pre class="r"><code>fitmm0.neuron = dream(voomed0.neuron, formde, pseudo0.neuron$meta)</code></pre>
<pre><code>Dividing work into 100 chunks...

Total: 113 s</code></pre>
<pre class="r"><code>try(topTable(fitmm0.neuron, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.neuron[1,1] &lt;- try(topTable(fitmm0.neuron, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.neuron[1, 1] &lt;- try(topTable(fitmm0.neuron, coef = &quot;stimoxidation&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm0.neuron, coef=&#39;stimoxidation&#39;,  number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;Oxidation DE genes, cluster 0 of full dataset&quot;)</code></pre>
<table>
<caption>Oxidation DE genes, cluster 0 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>CAV3</td>
<td align="right">-0.5674932</td>
<td align="right">3.588982</td>
<td align="right">-6.889537</td>
<td align="right">0.0000051</td>
<td align="right">0.0635264</td>
<td align="right">-4.559700</td>
</tr>
<tr class="even">
<td>KDM3A</td>
<td align="right">-0.7817717</td>
<td align="right">5.609796</td>
<td align="right">-6.559811</td>
<td align="right">0.0000146</td>
<td align="right">0.0684807</td>
<td align="right">-4.335094</td>
</tr>
<tr class="odd">
<td>C19orf44</td>
<td align="right">-0.5315151</td>
<td align="right">2.080077</td>
<td align="right">-6.199607</td>
<td align="right">0.0000166</td>
<td align="right">0.0684807</td>
<td align="right">-4.306756</td>
</tr>
<tr class="even">
<td>MT-ND2</td>
<td align="right">0.5295625</td>
<td align="right">10.209263</td>
<td align="right">6.379629</td>
<td align="right">0.0000239</td>
<td align="right">0.0739557</td>
<td align="right">4.225371</td>
</tr>
<tr class="odd">
<td>MT-ND3</td>
<td align="right">0.4229258</td>
<td align="right">11.042302</td>
<td align="right">4.729425</td>
<td align="right">0.0002663</td>
<td align="right">0.3853219</td>
<td align="right">3.646034</td>
</tr>
<tr class="even">
<td>LRP2BP</td>
<td align="right">-0.5656928</td>
<td align="right">3.819085</td>
<td align="right">-4.866417</td>
<td align="right">0.0002953</td>
<td align="right">0.3853219</td>
<td align="right">-3.619364</td>
</tr>
<tr class="odd">
<td>PDK1</td>
<td align="right">-1.8095855</td>
<td align="right">4.874593</td>
<td align="right">-4.495991</td>
<td align="right">0.0003185</td>
<td align="right">0.3853219</td>
<td align="right">-3.599794</td>
</tr>
<tr class="even">
<td>VEGFA</td>
<td align="right">-1.0935612</td>
<td align="right">5.300377</td>
<td align="right">-4.485661</td>
<td align="right">0.0003256</td>
<td align="right">0.3853219</td>
<td align="right">-3.594043</td>
</tr>
<tr class="odd">
<td>LMLN</td>
<td align="right">-0.5664807</td>
<td align="right">2.828177</td>
<td align="right">-4.469714</td>
<td align="right">0.0003369</td>
<td align="right">0.3853219</td>
<td align="right">-3.585146</td>
</tr>
<tr class="even">
<td>RAPGEF4</td>
<td align="right">-1.3815273</td>
<td align="right">5.136007</td>
<td align="right">-4.465145</td>
<td align="right">0.0003402</td>
<td align="right">0.3853219</td>
<td align="right">-3.582592</td>
</tr>
</tbody>
</table>
<pre class="r"><code>try(topTable(fitmm0.neuron, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.neuron[1,2] &lt;- try(topTable(fitmm0.neuron, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.neuron[1, 2] &lt;- try(topTable(fitmm0.neuron, coef = &quot;stimpff&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm0.neuron, coef=&#39;stimpff&#39;, number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;PFF DE genes, cluster 0 of full dataset&quot;)</code></pre>
<table>
<caption>PFF DE genes, cluster 0 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>AC068888.1</td>
<td align="right">0.6010384</td>
<td align="right">3.912675</td>
<td align="right">5.789415</td>
<td align="right">0.0000376</td>
<td align="right">0.4666255</td>
<td align="right">4.121581</td>
</tr>
<tr class="even">
<td>UNG</td>
<td align="right">0.3640366</td>
<td align="right">3.507034</td>
<td align="right">4.828846</td>
<td align="right">0.0002848</td>
<td align="right">0.9812311</td>
<td align="right">3.628729</td>
</tr>
<tr class="odd">
<td>IFFO1</td>
<td align="right">-0.5739966</td>
<td align="right">2.754298</td>
<td align="right">-4.632184</td>
<td align="right">0.0003169</td>
<td align="right">0.9812311</td>
<td align="right">-3.601113</td>
</tr>
<tr class="even">
<td>FZD6</td>
<td align="right">0.6993105</td>
<td align="right">3.234707</td>
<td align="right">4.233247</td>
<td align="right">0.0005600</td>
<td align="right">0.9812311</td>
<td align="right">3.450290</td>
</tr>
<tr class="odd">
<td>RHNO1</td>
<td align="right">-0.4698866</td>
<td align="right">3.494592</td>
<td align="right">-4.423919</td>
<td align="right">0.0006093</td>
<td align="right">0.9812311</td>
<td align="right">-3.427435</td>
</tr>
<tr class="even">
<td>MAP3K15</td>
<td align="right">1.0685425</td>
<td align="right">2.721077</td>
<td align="right">4.092583</td>
<td align="right">0.0007588</td>
<td align="right">0.9812311</td>
<td align="right">3.367403</td>
</tr>
<tr class="odd">
<td>ERMARD</td>
<td align="right">-0.8705823</td>
<td align="right">3.704051</td>
<td align="right">-3.971397</td>
<td align="right">0.0009865</td>
<td align="right">0.9812311</td>
<td align="right">-3.294338</td>
</tr>
<tr class="even">
<td>PFKFB4</td>
<td align="right">0.5642126</td>
<td align="right">3.798203</td>
<td align="right">3.831037</td>
<td align="right">0.0015890</td>
<td align="right">0.9812311</td>
<td align="right">3.157923</td>
</tr>
<tr class="odd">
<td>SAMD1</td>
<td align="right">-0.3653188</td>
<td align="right">6.400913</td>
<td align="right">-3.832331</td>
<td align="right">0.0015896</td>
<td align="right">0.9812311</td>
<td align="right">-3.157807</td>
</tr>
<tr class="even">
<td>PWWP2A</td>
<td align="right">0.2845128</td>
<td align="right">5.785769</td>
<td align="right">3.741703</td>
<td align="right">0.0017923</td>
<td align="right">0.9812311</td>
<td align="right">3.122651</td>
</tr>
</tbody>
</table>
<p>For cluster 1:</p>
<pre class="r"><code>pseudo1.full &lt;- generate.pseudobulk(cluster1.full, labels = c(&quot;stim&quot;, &quot;replicate&quot;, &quot;cell.line&quot;))
pseudo1.full &lt;- filter.pseudobulk(pseudo1.full)
d1.full &lt;- DGEList(pseudo1.full$counts)
d1.full &lt;- calcNormFactors(d1.full)
keep1.full &lt;- filterByExpr(d1.full$counts)
d1.full &lt;- d1.full[keep1.full,]

voomed1.full &lt;- voomWithDreamWeights(d1.full, formde, pseudo1.full$meta, plot=FALSE)</code></pre>
<pre><code>Memory usage to store result: &gt; 28.8 Mb 
Dividing work into 100 chunks...

Total: 83 s</code></pre>
<pre class="r"><code>fitmm1.full = dream(voomed1.full, formde, pseudo1.full$meta)</code></pre>
<pre><code>Dividing work into 100 chunks...

Total: 113 s</code></pre>
<pre class="r"><code>try(topTable(fitmm1.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>[1] 1360</code></pre>
<pre class="r"><code>de.summary.full[2,1] &lt;- try(topTable(fitmm1.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()
topTable(fitmm1.full, coef=&#39;stimoxidation&#39;,  number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;Oxidation DE genes, cluster 1 of full dataset&quot;)</code></pre>
<table>
<caption>Oxidation DE genes, cluster 1 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>MIR210HG</td>
<td align="right">-1.4861649</td>
<td align="right">4.545416</td>
<td align="right">-10.120440</td>
<td align="right">0e+00</td>
<td align="right">0.0002810</td>
<td align="right">-5.589244</td>
</tr>
<tr class="even">
<td>ANKRD12</td>
<td align="right">-0.7149233</td>
<td align="right">7.352097</td>
<td align="right">-8.996813</td>
<td align="right">1e-07</td>
<td align="right">0.0004988</td>
<td align="right">-5.299412</td>
</tr>
<tr class="odd">
<td>SLC25A36</td>
<td align="right">-0.7282901</td>
<td align="right">6.333232</td>
<td align="right">-9.625732</td>
<td align="right">1e-07</td>
<td align="right">0.0004988</td>
<td align="right">-5.256748</td>
</tr>
<tr class="even">
<td>FAM210A</td>
<td align="right">-1.0850808</td>
<td align="right">5.431128</td>
<td align="right">-8.456170</td>
<td align="right">3e-07</td>
<td align="right">0.0004988</td>
<td align="right">-5.147805</td>
</tr>
<tr class="odd">
<td>MXI1</td>
<td align="right">-0.9297223</td>
<td align="right">6.186212</td>
<td align="right">-9.053788</td>
<td align="right">3e-07</td>
<td align="right">0.0004988</td>
<td align="right">-5.131769</td>
</tr>
<tr class="even">
<td>POLR1D</td>
<td align="right">-0.4162784</td>
<td align="right">7.377297</td>
<td align="right">-7.890080</td>
<td align="right">3e-07</td>
<td align="right">0.0004988</td>
<td align="right">-5.124532</td>
</tr>
<tr class="odd">
<td>NFKBIL1</td>
<td align="right">-0.7921862</td>
<td align="right">5.174177</td>
<td align="right">-8.855684</td>
<td align="right">3e-07</td>
<td align="right">0.0004988</td>
<td align="right">-5.110982</td>
</tr>
<tr class="even">
<td>CCDC9</td>
<td align="right">-0.4534393</td>
<td align="right">3.812727</td>
<td align="right">-8.287774</td>
<td align="right">3e-07</td>
<td align="right">0.0004988</td>
<td align="right">-5.094867</td>
</tr>
<tr class="odd">
<td>ZC3H6</td>
<td align="right">-0.9555621</td>
<td align="right">4.794496</td>
<td align="right">-8.636903</td>
<td align="right">4e-07</td>
<td align="right">0.0004988</td>
<td align="right">-5.086733</td>
</tr>
<tr class="even">
<td>SNAPC3</td>
<td align="right">-0.5812198</td>
<td align="right">5.068041</td>
<td align="right">-8.760951</td>
<td align="right">4e-07</td>
<td align="right">0.0005230</td>
<td align="right">-5.057672</td>
</tr>
</tbody>
</table>
<pre class="r"><code>try(topTable(fitmm1.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code>de.summary.full[2,2] &lt;- try(topTable(fitmm1.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()
topTable(fitmm1.full, coef=&#39;stimpff&#39;, number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;PFF DE genes, cluster 1 of full dataset&quot;)</code></pre>
<table>
<caption>PFF DE genes, cluster 1 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>DEPDC5</td>
<td align="right">0.5113727</td>
<td align="right">3.615869</td>
<td align="right">7.828587</td>
<td align="right">0.0000009</td>
<td align="right">0.0104764</td>
<td align="right">4.923489</td>
</tr>
<tr class="even">
<td>ZNF529</td>
<td align="right">0.4995452</td>
<td align="right">4.573766</td>
<td align="right">5.700794</td>
<td align="right">0.0000209</td>
<td align="right">0.1288884</td>
<td align="right">4.254856</td>
</tr>
<tr class="odd">
<td>AC012447.1</td>
<td align="right">1.7563688</td>
<td align="right">2.462530</td>
<td align="right">5.496095</td>
<td align="right">0.0000531</td>
<td align="right">0.1646350</td>
<td align="right">4.041675</td>
</tr>
<tr class="even">
<td>LZTR1</td>
<td align="right">0.5497957</td>
<td align="right">2.972287</td>
<td align="right">5.507253</td>
<td align="right">0.0000534</td>
<td align="right">0.1646350</td>
<td align="right">4.040050</td>
</tr>
<tr class="odd">
<td>ZNF783</td>
<td align="right">0.4224450</td>
<td align="right">3.442421</td>
<td align="right">4.889662</td>
<td align="right">0.0001180</td>
<td align="right">0.2907034</td>
<td align="right">3.850345</td>
</tr>
<tr class="even">
<td>ALKBH1</td>
<td align="right">0.6435701</td>
<td align="right">3.576970</td>
<td align="right">5.760180</td>
<td align="right">0.0001864</td>
<td align="right">0.2920017</td>
<td align="right">3.736826</td>
</tr>
<tr class="odd">
<td>CROCC</td>
<td align="right">0.7524263</td>
<td align="right">4.808688</td>
<td align="right">4.971323</td>
<td align="right">0.0001868</td>
<td align="right">0.2920017</td>
<td align="right">3.736222</td>
</tr>
<tr class="even">
<td>ZNF672</td>
<td align="right">-0.8595912</td>
<td align="right">2.397761</td>
<td align="right">-4.952524</td>
<td align="right">0.0001896</td>
<td align="right">0.2920017</td>
<td align="right">-3.732531</td>
</tr>
<tr class="odd">
<td>TRMT13</td>
<td align="right">0.4932977</td>
<td align="right">4.181665</td>
<td align="right">4.767658</td>
<td align="right">0.0002149</td>
<td align="right">0.2942165</td>
<td align="right">3.700850</td>
</tr>
<tr class="even">
<td>TMEM177</td>
<td align="right">0.5539319</td>
<td align="right">2.358721</td>
<td align="right">4.638138</td>
<td align="right">0.0002758</td>
<td align="right">0.3160229</td>
<td align="right">3.637057</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#now repeat with the neuron dataset
pseudo1.neuron &lt;- generate.pseudobulk(cluster1.neuron, labels = c(&quot;stim&quot;, &quot;replicate&quot;, &quot;cell.line&quot;))
pseudo1.neuron &lt;- filter.pseudobulk(pseudo1.neuron)
d1.neuron &lt;- DGEList(pseudo1.neuron$counts)
d1.neuron &lt;- calcNormFactors(d1.neuron)
keep1.neuron &lt;- filterByExpr(d1.neuron$counts)
d1.neuron &lt;- d1.neuron[keep1.neuron,]

voomed1.neuron &lt;- voomWithDreamWeights(d1.neuron, formde, pseudo1.neuron$meta, plot=FALSE)</code></pre>
<pre><code>Memory usage to store result: &gt; 29.4 Mb 
Dividing work into 100 chunks...

Total: 84 s</code></pre>
<pre class="r"><code>fitmm1.neuron = dream(voomed1.neuron, formde, pseudo1.neuron$meta)</code></pre>
<pre><code>Dividing work into 100 chunks...

Total: 115 s</code></pre>
<pre class="r"><code>try(topTable(fitmm1.neuron, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.neuron[2,1] &lt;- try(topTable(fitmm1.neuron, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.neuron[2, 1] &lt;- try(topTable(fitmm1.neuron, coef = &quot;stimoxidation&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm1.neuron, coef=&#39;stimoxidation&#39;,  number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;Oxidation DE genes, cluster 1 of full dataset&quot;)</code></pre>
<table>
<caption>Oxidation DE genes, cluster 1 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ARL2BP</td>
<td align="right">0.1693057</td>
<td align="right">5.809936</td>
<td align="right">6.570752</td>
<td align="right">0.0000217</td>
<td align="right">0.1607343</td>
<td align="right">4.246699</td>
</tr>
<tr class="even">
<td>CFAP97</td>
<td align="right">-0.4051432</td>
<td align="right">6.207754</td>
<td align="right">-6.531684</td>
<td align="right">0.0000329</td>
<td align="right">0.1607343</td>
<td align="right">-4.152412</td>
</tr>
<tr class="odd">
<td>P4HA1</td>
<td align="right">-0.9316080</td>
<td align="right">5.895728</td>
<td align="right">-5.747116</td>
<td align="right">0.0000654</td>
<td align="right">0.1607343</td>
<td align="right">-3.992572</td>
</tr>
<tr class="even">
<td>MRPL17</td>
<td align="right">-0.3879544</td>
<td align="right">4.376946</td>
<td align="right">-5.802094</td>
<td align="right">0.0000697</td>
<td align="right">0.1607343</td>
<td align="right">-3.977448</td>
</tr>
<tr class="odd">
<td>IFNAR1</td>
<td align="right">-0.2695187</td>
<td align="right">5.574632</td>
<td align="right">-6.076647</td>
<td align="right">0.0000796</td>
<td align="right">0.1607343</td>
<td align="right">-3.945555</td>
</tr>
<tr class="even">
<td>LINC01618</td>
<td align="right">0.9421520</td>
<td align="right">3.530375</td>
<td align="right">5.402144</td>
<td align="right">0.0000898</td>
<td align="right">0.1607343</td>
<td align="right">3.916551</td>
</tr>
<tr class="odd">
<td>GALNT17</td>
<td align="right">-0.8086929</td>
<td align="right">5.481707</td>
<td align="right">-5.737438</td>
<td align="right">0.0000906</td>
<td align="right">0.1607343</td>
<td align="right">-3.914454</td>
</tr>
<tr class="even">
<td>AGGF1</td>
<td align="right">-0.1678874</td>
<td align="right">5.376466</td>
<td align="right">-5.509918</td>
<td align="right">0.0001134</td>
<td align="right">0.1607343</td>
<td align="right">-3.859949</td>
</tr>
<tr class="odd">
<td>SF3B1</td>
<td align="right">-0.2162695</td>
<td align="right">7.203681</td>
<td align="right">-5.173583</td>
<td align="right">0.0001136</td>
<td align="right">0.1607343</td>
<td align="right">-3.859542</td>
</tr>
<tr class="even">
<td>UBAC2</td>
<td align="right">0.2009153</td>
<td align="right">6.189419</td>
<td align="right">5.485343</td>
<td align="right">0.0001193</td>
<td align="right">0.1607343</td>
<td align="right">3.847612</td>
</tr>
</tbody>
</table>
<pre class="r"><code>try(topTable(fitmm1.neuron, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.neuron[2,2] &lt;- try(topTable(fitmm1.neuron, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.neuron[2, 2] &lt;- try(topTable(fitmm1.neuron, coef = &quot;stimpff&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm1.neuron, coef=&#39;stimpff&#39;, number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;PFF DE genes, cluster 1 of full dataset&quot;)</code></pre>
<table>
<caption>PFF DE genes, cluster 1 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>AL121900.1</td>
<td align="right">1.0555931</td>
<td align="right">1.448619</td>
<td align="right">6.457476</td>
<td align="right">0.0000180</td>
<td align="right">0.2422689</td>
<td align="right">4.288631</td>
</tr>
<tr class="even">
<td>DLGAP1-AS5</td>
<td align="right">0.4924793</td>
<td align="right">5.782916</td>
<td align="right">5.114082</td>
<td align="right">0.0001743</td>
<td align="right">0.9975140</td>
<td align="right">3.753658</td>
</tr>
<tr class="odd">
<td>PIGN</td>
<td align="right">-0.5734598</td>
<td align="right">4.725704</td>
<td align="right">-5.220067</td>
<td align="right">0.0002258</td>
<td align="right">0.9975140</td>
<td align="right">-3.688257</td>
</tr>
<tr class="even">
<td>SGTA</td>
<td align="right">-0.3380298</td>
<td align="right">4.565339</td>
<td align="right">-4.365982</td>
<td align="right">0.0006676</td>
<td align="right">0.9975140</td>
<td align="right">-3.402544</td>
</tr>
<tr class="odd">
<td>PCNA</td>
<td align="right">-0.5634151</td>
<td align="right">3.125940</td>
<td align="right">-4.182058</td>
<td align="right">0.0009035</td>
<td align="right">0.9975140</td>
<td align="right">-3.318980</td>
</tr>
<tr class="even">
<td>RASGRP3</td>
<td align="right">0.7281125</td>
<td align="right">1.852595</td>
<td align="right">3.995450</td>
<td align="right">0.0013706</td>
<td align="right">0.9975140</td>
<td align="right">3.200769</td>
</tr>
<tr class="odd">
<td>VARS</td>
<td align="right">-0.2761577</td>
<td align="right">4.860495</td>
<td align="right">-4.011752</td>
<td align="right">0.0015680</td>
<td align="right">0.9975140</td>
<td align="right">-3.161802</td>
</tr>
<tr class="even">
<td>CYSTM1</td>
<td align="right">0.2783571</td>
<td align="right">5.637518</td>
<td align="right">3.875293</td>
<td align="right">0.0020615</td>
<td align="right">0.9975140</td>
<td align="right">3.081233</td>
</tr>
<tr class="odd">
<td>ZBTB49</td>
<td align="right">0.3307848</td>
<td align="right">3.525877</td>
<td align="right">3.791935</td>
<td align="right">0.0022624</td>
<td align="right">0.9975140</td>
<td align="right">3.053432</td>
</tr>
<tr class="even">
<td>AC099778.1</td>
<td align="right">1.1481636</td>
<td align="right">1.784776</td>
<td align="right">3.607745</td>
<td align="right">0.0023603</td>
<td align="right">0.9975140</td>
<td align="right">3.040702</td>
</tr>
</tbody>
</table>
<p>For cluster 2:</p>
<pre class="r"><code>pseudo2.full &lt;- generate.pseudobulk(cluster2.full, labels = c(&quot;stim&quot;, &quot;replicate&quot;, &quot;cell.line&quot;))
pseudo2.full &lt;- filter.pseudobulk(pseudo2.full)
d2.full &lt;- DGEList(pseudo2.full$counts)
d2.full &lt;- calcNormFactors(d2.full)
keep2.full &lt;- filterByExpr(d2.full$counts)
d2.full &lt;- d2.full[keep2.full,]

voomed2.full &lt;- voomWithDreamWeights(d2.full, formde, pseudo2.full$meta, plot=FALSE)</code></pre>
<pre><code>Memory usage to store result: &gt; 29.6 Mb 
Dividing work into 100 chunks...

Total: 85 s</code></pre>
<pre class="r"><code>fitmm2.full = dream(voomed2.full, formde, pseudo2.full$meta)</code></pre>
<pre><code>Dividing work into 100 chunks...

Total: 117 s</code></pre>
<pre class="r"><code>try(topTable(fitmm2.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.full[3,1] &lt;- try(topTable(fitmm2.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.full[3, 1] &lt;- try(topTable(fitmm2.full, coef = &quot;stimoxidation&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm2.full, coef=&#39;stimoxidation&#39;,  number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;Oxidation DE genes, cluster 2 of full dataset&quot;)</code></pre>
<table>
<caption>Oxidation DE genes, cluster 2 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>XIAP</td>
<td align="right">-0.1864341</td>
<td align="right">5.958063</td>
<td align="right">-7.570081</td>
<td align="right">6.00e-06</td>
<td align="right">0.0697605</td>
<td align="right">-4.527042</td>
</tr>
<tr class="even">
<td>ARL2BP</td>
<td align="right">0.1673053</td>
<td align="right">5.824260</td>
<td align="right">6.717033</td>
<td align="right">1.86e-05</td>
<td align="right">0.0697605</td>
<td align="right">4.280700</td>
</tr>
<tr class="odd">
<td>MTFMT</td>
<td align="right">0.2916516</td>
<td align="right">5.107276</td>
<td align="right">6.756897</td>
<td align="right">1.96e-05</td>
<td align="right">0.0697605</td>
<td align="right">4.268915</td>
</tr>
<tr class="even">
<td>ADNP2</td>
<td align="right">0.4657531</td>
<td align="right">4.900196</td>
<td align="right">6.671036</td>
<td align="right">2.38e-05</td>
<td align="right">0.0697605</td>
<td align="right">4.225886</td>
</tr>
<tr class="odd">
<td>PIK3C3</td>
<td align="right">-0.2298161</td>
<td align="right">5.907815</td>
<td align="right">-6.259271</td>
<td align="right">3.21e-05</td>
<td align="right">0.0697605</td>
<td align="right">-4.157816</td>
</tr>
<tr class="even">
<td>IFITM3</td>
<td align="right">-2.3090427</td>
<td align="right">2.440484</td>
<td align="right">-6.357182</td>
<td align="right">3.28e-05</td>
<td align="right">0.0697605</td>
<td align="right">-4.153270</td>
</tr>
<tr class="odd">
<td>CFAP97</td>
<td align="right">-0.4138790</td>
<td align="right">6.223015</td>
<td align="right">-6.535221</td>
<td align="right">3.60e-05</td>
<td align="right">0.0697605</td>
<td align="right">-4.131833</td>
</tr>
<tr class="even">
<td>SF3B1</td>
<td align="right">-0.2164410</td>
<td align="right">7.168483</td>
<td align="right">-5.609188</td>
<td align="right">5.30e-05</td>
<td align="right">0.0898259</td>
<td align="right">-4.042167</td>
</tr>
<tr class="odd">
<td>P4HA1</td>
<td align="right">-1.0380718</td>
<td align="right">6.005931</td>
<td align="right">-5.753167</td>
<td align="right">6.49e-05</td>
<td align="right">0.0979144</td>
<td align="right">-3.994086</td>
</tr>
<tr class="even">
<td>RFC1</td>
<td align="right">-0.2500093</td>
<td align="right">5.714798</td>
<td align="right">-5.423565</td>
<td align="right">8.41e-05</td>
<td align="right">0.1120628</td>
<td align="right">-3.932281</td>
</tr>
</tbody>
</table>
<pre class="r"><code>try(topTable(fitmm2.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.full[3,2] &lt;- try(topTable(fitmm2.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.full[3, 2] &lt;- try(topTable(fitmm2.full, coef = &quot;stimpff&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm2.full, coef=&#39;stimpff&#39;, number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;PFF DE genes, cluster 2 of full dataset&quot;)</code></pre>
<table>
<caption>PFF DE genes, cluster 2 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PLA2G4B</td>
<td align="right">1.1913565</td>
<td align="right">1.220316</td>
<td align="right">4.395943</td>
<td align="right">0.0007213</td>
<td align="right">0.9998068</td>
<td align="right">3.381341</td>
</tr>
<tr class="even">
<td>BOLA2B</td>
<td align="right">0.6244385</td>
<td align="right">2.574218</td>
<td align="right">4.414108</td>
<td align="right">0.0007662</td>
<td align="right">0.9998068</td>
<td align="right">3.364730</td>
</tr>
<tr class="odd">
<td>IAH1</td>
<td align="right">0.6582530</td>
<td align="right">2.862523</td>
<td align="right">4.299066</td>
<td align="right">0.0008016</td>
<td align="right">0.9998068</td>
<td align="right">3.352241</td>
</tr>
<tr class="even">
<td>SUPT6H</td>
<td align="right">0.2341494</td>
<td align="right">5.699308</td>
<td align="right">4.184495</td>
<td align="right">0.0008294</td>
<td align="right">0.9998068</td>
<td align="right">3.342789</td>
</tr>
<tr class="odd">
<td>MYO1B</td>
<td align="right">-0.7802405</td>
<td align="right">3.271614</td>
<td align="right">-4.252760</td>
<td align="right">0.0008804</td>
<td align="right">0.9998068</td>
<td align="right">-3.326183</td>
</tr>
<tr class="even">
<td>ZNF655</td>
<td align="right">-0.3628586</td>
<td align="right">5.145655</td>
<td align="right">-4.080912</td>
<td align="right">0.0009228</td>
<td align="right">0.9998068</td>
<td align="right">-3.313064</td>
</tr>
<tr class="odd">
<td>CYSTM1</td>
<td align="right">0.2870296</td>
<td align="right">5.655375</td>
<td align="right">4.282610</td>
<td align="right">0.0009755</td>
<td align="right">0.9998068</td>
<td align="right">3.297486</td>
</tr>
<tr class="even">
<td>INPP5B</td>
<td align="right">0.4214799</td>
<td align="right">4.413828</td>
<td align="right">4.005708</td>
<td align="right">0.0010202</td>
<td align="right">0.9998068</td>
<td align="right">3.284895</td>
</tr>
<tr class="odd">
<td>GALNT1</td>
<td align="right">-0.1362841</td>
<td align="right">6.930216</td>
<td align="right">-4.156226</td>
<td align="right">0.0012361</td>
<td align="right">0.9998068</td>
<td align="right">-3.230412</td>
</tr>
<tr class="even">
<td>TMEM106C</td>
<td align="right">-0.3135089</td>
<td align="right">4.337968</td>
<td align="right">-4.209453</td>
<td align="right">0.0012599</td>
<td align="right">0.9998068</td>
<td align="right">-3.224959</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#now repeat with the neuron dataset
pseudo2.neuron &lt;- generate.pseudobulk(cluster2.neuron, labels = c(&quot;stim&quot;, &quot;replicate&quot;, &quot;cell.line&quot;))
pseudo2.neuron &lt;- filter.pseudobulk(pseudo2.neuron)
d2.neuron &lt;- DGEList(pseudo2.neuron$counts)
d2.neuron &lt;- calcNormFactors(d2.neuron)
keep2.neuron &lt;- filterByExpr(d2.neuron$counts)
d2.neuron &lt;- d2.neuron[keep2.neuron,]

voomed2.neuron &lt;- voomWithDreamWeights(d2.neuron, formde, pseudo2.neuron$meta, plot=FALSE)</code></pre>
<pre><code>Memory usage to store result: &gt; 27.7 Mb 
Dividing work into 100 chunks...

Total: 83 s</code></pre>
<pre class="r"><code>fitmm2.neuron = dream(voomed2.neuron, formde, pseudo2.neuron$meta)</code></pre>
<pre><code>Dividing work into 100 chunks...

Total: 113 s</code></pre>
<pre class="r"><code>try(topTable(fitmm2.neuron, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>[1] 92</code></pre>
<pre class="r"><code>de.summary.neuron[3,1] &lt;- try(topTable(fitmm2.neuron, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()
topTable(fitmm2.neuron, coef=&#39;stimoxidation&#39;,  number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;Oxidation DE genes, cluster 2 of full dataset&quot;)</code></pre>
<table>
<caption>Oxidation DE genes, cluster 2 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>KCNMB2-AS1</td>
<td align="right">1.7348209</td>
<td align="right">6.834342</td>
<td align="right">7.285786</td>
<td align="right">2.70e-06</td>
<td align="right">0.0145610</td>
<td align="right">4.691129</td>
</tr>
<tr class="even">
<td>FUT9</td>
<td align="right">0.2994957</td>
<td align="right">8.563892</td>
<td align="right">7.709804</td>
<td align="right">3.30e-06</td>
<td align="right">0.0145610</td>
<td align="right">4.651283</td>
</tr>
<tr class="odd">
<td>NUP210L</td>
<td align="right">1.4434435</td>
<td align="right">3.025174</td>
<td align="right">7.666420</td>
<td align="right">3.60e-06</td>
<td align="right">0.0145610</td>
<td align="right">4.635452</td>
</tr>
<tr class="even">
<td>SOX21-AS1</td>
<td align="right">0.6314250</td>
<td align="right">4.567713</td>
<td align="right">6.981071</td>
<td align="right">9.20e-06</td>
<td align="right">0.0237983</td>
<td align="right">4.434281</td>
</tr>
<tr class="odd">
<td>HK2</td>
<td align="right">-1.8673542</td>
<td align="right">4.481175</td>
<td align="right">-6.515259</td>
<td align="right">1.06e-05</td>
<td align="right">0.0237983</td>
<td align="right">-4.404732</td>
</tr>
<tr class="even">
<td>TECPR2</td>
<td align="right">0.4244876</td>
<td align="right">5.444734</td>
<td align="right">6.733368</td>
<td align="right">1.28e-05</td>
<td align="right">0.0237983</td>
<td align="right">4.363811</td>
</tr>
<tr class="odd">
<td>HMGXB4</td>
<td align="right">-0.4005906</td>
<td align="right">6.095883</td>
<td align="right">-6.271780</td>
<td align="right">1.36e-05</td>
<td align="right">0.0237983</td>
<td align="right">-4.350506</td>
</tr>
<tr class="even">
<td>KRT10</td>
<td align="right">0.2434293</td>
<td align="right">7.154008</td>
<td align="right">6.531612</td>
<td align="right">1.88e-05</td>
<td align="right">0.0287555</td>
<td align="right">4.279206</td>
</tr>
<tr class="odd">
<td>TST</td>
<td align="right">0.8605936</td>
<td align="right">4.005944</td>
<td align="right">5.936928</td>
<td align="right">2.64e-05</td>
<td align="right">0.0332056</td>
<td align="right">4.202254</td>
</tr>
<tr class="even">
<td>PDK1</td>
<td align="right">-1.6125347</td>
<td align="right">5.627979</td>
<td align="right">-5.884568</td>
<td align="right">2.71e-05</td>
<td align="right">0.0332056</td>
<td align="right">-4.196782</td>
</tr>
</tbody>
</table>
<pre class="r"><code>try(topTable(fitmm2.neuron, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.neuron[3,2] &lt;- try(topTable(fitmm2.neuron, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.neuron[3, 2] &lt;- try(topTable(fitmm2.neuron, coef = &quot;stimpff&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm2.neuron, coef=&#39;stimpff&#39;, number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;PFF DE genes, cluster 2 of full dataset&quot;)</code></pre>
<table>
<caption>PFF DE genes, cluster 2 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>AC017101.1</td>
<td align="right">0.9519897</td>
<td align="right">3.175794</td>
<td align="right">4.938418</td>
<td align="right">0.0001247</td>
<td align="right">0.9252425</td>
<td align="right">3.836752</td>
</tr>
<tr class="even">
<td>CAV3</td>
<td align="right">1.2230091</td>
<td align="right">2.282294</td>
<td align="right">4.630913</td>
<td align="right">0.0002475</td>
<td align="right">0.9252425</td>
<td align="right">3.664839</td>
</tr>
<tr class="odd">
<td>KRT10</td>
<td align="right">0.2956825</td>
<td align="right">7.154008</td>
<td align="right">4.949392</td>
<td align="right">0.0002609</td>
<td align="right">0.9252425</td>
<td align="right">3.651268</td>
</tr>
<tr class="even">
<td>PACRGL</td>
<td align="right">-0.3141855</td>
<td align="right">5.445977</td>
<td align="right">-4.786914</td>
<td align="right">0.0003314</td>
<td align="right">0.9252425</td>
<td align="right">-3.589430</td>
</tr>
<tr class="odd">
<td>SAXO2</td>
<td align="right">-1.3861226</td>
<td align="right">3.419174</td>
<td align="right">-4.317978</td>
<td align="right">0.0007637</td>
<td align="right">0.9252425</td>
<td align="right">-3.365626</td>
</tr>
<tr class="even">
<td>LMBRD2</td>
<td align="right">-0.8482119</td>
<td align="right">4.123852</td>
<td align="right">-4.096463</td>
<td align="right">0.0007814</td>
<td align="right">0.9252425</td>
<td align="right">-3.359294</td>
</tr>
<tr class="odd">
<td>ARGLU1</td>
<td align="right">-0.3709277</td>
<td align="right">7.775373</td>
<td align="right">-4.244502</td>
<td align="right">0.0008085</td>
<td align="right">0.9252425</td>
<td align="right">-3.349863</td>
</tr>
<tr class="even">
<td>ZNF518A</td>
<td align="right">-0.3287731</td>
<td align="right">6.312342</td>
<td align="right">-3.945344</td>
<td align="right">0.0012845</td>
<td align="right">0.9252425</td>
<td align="right">-3.219424</td>
</tr>
<tr class="odd">
<td>DCAF11</td>
<td align="right">-0.7466575</td>
<td align="right">3.933147</td>
<td align="right">-3.713700</td>
<td align="right">0.0017257</td>
<td align="right">0.9252425</td>
<td align="right">-3.133776</td>
</tr>
<tr class="even">
<td>DHRS4</td>
<td align="right">-0.8131216</td>
<td align="right">3.111429</td>
<td align="right">-3.706478</td>
<td align="right">0.0020872</td>
<td align="right">0.9252425</td>
<td align="right">-3.077539</td>
</tr>
</tbody>
</table>
<p>For cluster 3:</p>
<pre class="r"><code>pseudo3.full &lt;- generate.pseudobulk(cluster3.full, labels = c(&quot;stim&quot;, &quot;replicate&quot;, &quot;cell.line&quot;))
pseudo3.full &lt;- filter.pseudobulk(pseudo3.full)
d3.full &lt;- DGEList(pseudo3.full$counts)
d3.full &lt;- calcNormFactors(d3.full)
keep3.full &lt;- filterByExpr(d3.full$counts)
d3.full &lt;- d3.full[keep3.full,]

voomed3.full &lt;- voomWithDreamWeights(d3.full, formde, pseudo3.full$meta, plot=FALSE)</code></pre>
<pre><code>Memory usage to store result: &gt; 25 Mb 
Dividing work into 100 chunks...

Total: 81 s</code></pre>
<pre class="r"><code>fitmm3.full = dream(voomed3.full, formde, pseudo3.full$meta)</code></pre>
<pre><code>Dividing work into 100 chunks...

Total: 107 s</code></pre>
<pre class="r"><code>try(topTable(fitmm3.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>[1] 30</code></pre>
<pre class="r"><code>de.summary.full[4,1] &lt;- try(topTable(fitmm3.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()
topTable(fitmm3.full, coef=&#39;stimoxidation&#39;,  number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;Oxidation DE genes, cluster 3 of full dataset&quot;)</code></pre>
<table>
<caption>Oxidation DE genes, cluster 3 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>MIR210HG</td>
<td align="right">-1.9785129</td>
<td align="right">4.629151</td>
<td align="right">-8.955030</td>
<td align="right">6.00e-07</td>
<td align="right">0.0066982</td>
<td align="right">-4.977548</td>
</tr>
<tr class="even">
<td>EGLN1</td>
<td align="right">-1.1781445</td>
<td align="right">6.045484</td>
<td align="right">-7.464960</td>
<td align="right">1.30e-06</td>
<td align="right">0.0066982</td>
<td align="right">-4.847357</td>
</tr>
<tr class="odd">
<td>LNPK</td>
<td align="right">-0.5581982</td>
<td align="right">5.938798</td>
<td align="right">-6.501533</td>
<td align="right">6.80e-06</td>
<td align="right">0.0189237</td>
<td align="right">-4.499248</td>
</tr>
<tr class="even">
<td>KDM4B</td>
<td align="right">-0.4113887</td>
<td align="right">5.851409</td>
<td align="right">-6.148095</td>
<td align="right">8.30e-06</td>
<td align="right">0.0189237</td>
<td align="right">-4.456436</td>
</tr>
<tr class="odd">
<td>ZNF160</td>
<td align="right">-0.5986243</td>
<td align="right">4.669685</td>
<td align="right">-6.739653</td>
<td align="right">8.80e-06</td>
<td align="right">0.0189237</td>
<td align="right">-4.443832</td>
</tr>
<tr class="even">
<td>PDK1</td>
<td align="right">-1.4533313</td>
<td align="right">4.728879</td>
<td align="right">-5.783801</td>
<td align="right">1.76e-05</td>
<td align="right">0.0314077</td>
<td align="right">-4.293343</td>
</tr>
<tr class="odd">
<td>DARS</td>
<td align="right">-0.5559017</td>
<td align="right">6.673600</td>
<td align="right">-5.708037</td>
<td align="right">2.06e-05</td>
<td align="right">0.0315165</td>
<td align="right">-4.258235</td>
</tr>
<tr class="even">
<td>NAA15</td>
<td align="right">0.3841580</td>
<td align="right">5.903543</td>
<td align="right">5.922204</td>
<td align="right">2.93e-05</td>
<td align="right">0.0370849</td>
<td align="right">4.178895</td>
</tr>
<tr class="odd">
<td>CEP250</td>
<td align="right">-0.5496013</td>
<td align="right">4.448796</td>
<td align="right">-6.063847</td>
<td align="right">3.12e-05</td>
<td align="right">0.0370849</td>
<td align="right">-4.164727</td>
</tr>
<tr class="even">
<td>CDKN1B</td>
<td align="right">-0.3917505</td>
<td align="right">5.695571</td>
<td align="right">-5.783999</td>
<td align="right">3.64e-05</td>
<td align="right">0.0389842</td>
<td align="right">-4.129154</td>
</tr>
</tbody>
</table>
<pre class="r"><code>try(topTable(fitmm3.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.full[4,2] &lt;- try(topTable(fitmm3.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.full[4, 2] &lt;- try(topTable(fitmm3.full, coef = &quot;stimpff&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm3.full, coef=&#39;stimpff&#39;, number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;PFF DE genes, cluster 3 of full dataset&quot;)</code></pre>
<table>
<caption>PFF DE genes, cluster 3 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>AC020916.1</td>
<td align="right">1.1429718</td>
<td align="right">3.683339</td>
<td align="right">5.613559</td>
<td align="right">0.0000598</td>
<td align="right">0.6398057</td>
<td align="right">4.013773</td>
</tr>
<tr class="even">
<td>ADAMTSL4-AS1</td>
<td align="right">0.9094911</td>
<td align="right">6.551624</td>
<td align="right">4.802219</td>
<td align="right">0.0002741</td>
<td align="right">0.9998251</td>
<td align="right">3.638632</td>
</tr>
<tr class="odd">
<td>PMPCB</td>
<td align="right">-0.3396362</td>
<td align="right">5.820159</td>
<td align="right">-4.002783</td>
<td align="right">0.0009937</td>
<td align="right">0.9998251</td>
<td align="right">-3.292303</td>
</tr>
<tr class="even">
<td>SIK3</td>
<td align="right">0.4247392</td>
<td align="right">6.542107</td>
<td align="right">3.862890</td>
<td align="right">0.0014697</td>
<td align="right">0.9998251</td>
<td align="right">3.180598</td>
</tr>
<tr class="odd">
<td>GRID1</td>
<td align="right">1.0941608</td>
<td align="right">4.606572</td>
<td align="right">3.575888</td>
<td align="right">0.0023933</td>
<td align="right">0.9998251</td>
<td align="right">3.036521</td>
</tr>
<tr class="even">
<td>TUBGCP2</td>
<td align="right">-0.1858209</td>
<td align="right">5.809549</td>
<td align="right">-3.500010</td>
<td align="right">0.0029593</td>
<td align="right">0.9998251</td>
<td align="right">-2.971935</td>
</tr>
<tr class="odd">
<td>ANP32B</td>
<td align="right">-0.2559796</td>
<td align="right">8.078451</td>
<td align="right">-3.431157</td>
<td align="right">0.0035162</td>
<td align="right">0.9998251</td>
<td align="right">-2.918589</td>
</tr>
<tr class="even">
<td>NUP37</td>
<td align="right">-0.4266955</td>
<td align="right">4.414016</td>
<td align="right">-3.415672</td>
<td align="right">0.0037520</td>
<td align="right">0.9998251</td>
<td align="right">-2.898295</td>
</tr>
<tr class="odd">
<td>RYR2</td>
<td align="right">1.7971076</td>
<td align="right">3.766548</td>
<td align="right">3.327662</td>
<td align="right">0.0037528</td>
<td align="right">0.9998251</td>
<td align="right">2.898223</td>
</tr>
<tr class="even">
<td>KDELR1</td>
<td align="right">-0.2160168</td>
<td align="right">8.200013</td>
<td align="right">-3.271174</td>
<td align="right">0.0047919</td>
<td align="right">0.9998251</td>
<td align="right">-2.820702</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#now repeat with the neuron dataset
pseudo3.neuron &lt;- generate.pseudobulk(cluster3.neuron, labels = c(&quot;stim&quot;, &quot;replicate&quot;, &quot;cell.line&quot;))
pseudo3.neuron &lt;- filter.pseudobulk(pseudo3.neuron)
d3.neuron &lt;- DGEList(pseudo3.neuron$counts)
d3.neuron &lt;- calcNormFactors(d3.neuron)
keep3.neuron &lt;- filterByExpr(d3.neuron$counts)
d3.neuron &lt;- d3.neuron[keep3.neuron,]

voomed3.neuron &lt;- voomWithDreamWeights(d3.neuron, formde, pseudo3.neuron$meta, plot=FALSE)</code></pre>
<pre><code>Memory usage to store result: &gt; 23.3 Mb 
Dividing work into 100 chunks...

Total: 81 s</code></pre>
<pre class="r"><code>fitmm3.neuron = dream(voomed3.neuron, formde, pseudo3.neuron$meta)</code></pre>
<pre><code>Dividing work into 100 chunks...

Total: 105 s</code></pre>
<pre class="r"><code>try(topTable(fitmm3.neuron, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>[1] 152</code></pre>
<pre class="r"><code>de.summary.neuron[4,1] &lt;- try(topTable(fitmm3.neuron, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()
topTable(fitmm3.neuron, coef=&#39;stimoxidation&#39;,  number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;Oxidation DE genes, cluster 3 of full dataset&quot;)</code></pre>
<table>
<caption>Oxidation DE genes, cluster 3 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>BNIP3</td>
<td align="right">-1.4539980</td>
<td align="right">8.730915</td>
<td align="right">-7.332267</td>
<td align="right">8.00e-07</td>
<td align="right">0.0082781</td>
<td align="right">-4.928500</td>
</tr>
<tr class="even">
<td>SPG7</td>
<td align="right">0.3759640</td>
<td align="right">5.862912</td>
<td align="right">7.091072</td>
<td align="right">2.40e-06</td>
<td align="right">0.0119616</td>
<td align="right">4.716895</td>
</tr>
<tr class="odd">
<td>DNAAF2</td>
<td align="right">0.7591070</td>
<td align="right">4.788517</td>
<td align="right">6.903826</td>
<td align="right">4.00e-06</td>
<td align="right">0.0132999</td>
<td align="right">4.611697</td>
</tr>
<tr class="even">
<td>MAP3K20</td>
<td align="right">-1.1974567</td>
<td align="right">4.974694</td>
<td align="right">-6.093347</td>
<td align="right">9.30e-06</td>
<td align="right">0.0223511</td>
<td align="right">-4.432513</td>
</tr>
<tr class="odd">
<td>PDK3</td>
<td align="right">-1.0317518</td>
<td align="right">5.382387</td>
<td align="right">-5.787479</td>
<td align="right">1.75e-05</td>
<td align="right">0.0223511</td>
<td align="right">-4.295037</td>
</tr>
<tr class="even">
<td>MTFP1</td>
<td align="right">-1.3830615</td>
<td align="right">4.236329</td>
<td align="right">-5.890055</td>
<td align="right">2.19e-05</td>
<td align="right">0.0223511</td>
<td align="right">-4.244634</td>
</tr>
<tr class="odd">
<td>AP002812.2</td>
<td align="right">1.5899153</td>
<td align="right">4.587028</td>
<td align="right">5.887744</td>
<td align="right">2.26e-05</td>
<td align="right">0.0223511</td>
<td align="right">4.237534</td>
</tr>
<tr class="even">
<td>PSMG4</td>
<td align="right">1.0983645</td>
<td align="right">5.720094</td>
<td align="right">5.879150</td>
<td align="right">2.33e-05</td>
<td align="right">0.0223511</td>
<td align="right">4.231033</td>
</tr>
<tr class="odd">
<td>ZBED5</td>
<td align="right">0.6601952</td>
<td align="right">6.227438</td>
<td align="right">5.844512</td>
<td align="right">2.43e-05</td>
<td align="right">0.0223511</td>
<td align="right">4.221257</td>
</tr>
<tr class="even">
<td>ADAMTS7</td>
<td align="right">1.0729560</td>
<td align="right">3.181875</td>
<td align="right">5.568918</td>
<td align="right">2.76e-05</td>
<td align="right">0.0223511</td>
<td align="right">4.192655</td>
</tr>
</tbody>
</table>
<pre class="r"><code>try(topTable(fitmm3.neuron, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.neuron[4,2] &lt;- try(topTable(fitmm3.neuron, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.neuron[4, 2] &lt;- try(topTable(fitmm3.neuron, coef = &quot;stimpff&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm3.neuron, coef=&#39;stimpff&#39;, number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;PFF DE genes, cluster 3 of full dataset&quot;)</code></pre>
<table>
<caption>PFF DE genes, cluster 3 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>MT-CO2</td>
<td align="right">0.6314077</td>
<td align="right">11.016584</td>
<td align="right">4.864757</td>
<td align="right">0.0001721</td>
<td align="right">0.9999021</td>
<td align="right">3.756797</td>
</tr>
<tr class="even">
<td>MT-ND6</td>
<td align="right">0.6385164</td>
<td align="right">6.730837</td>
<td align="right">4.414291</td>
<td align="right">0.0004554</td>
<td align="right">0.9999021</td>
<td align="right">3.505688</td>
</tr>
<tr class="odd">
<td>MED17</td>
<td align="right">-0.5022371</td>
<td align="right">6.263339</td>
<td align="right">-3.954988</td>
<td align="right">0.0011286</td>
<td align="right">0.9999021</td>
<td align="right">-3.256335</td>
</tr>
<tr class="even">
<td>MT-ND3</td>
<td align="right">0.7823187</td>
<td align="right">10.190854</td>
<td align="right">3.770109</td>
<td align="right">0.0018238</td>
<td align="right">0.9999021</td>
<td align="right">3.117512</td>
</tr>
<tr class="odd">
<td>NAA15</td>
<td align="right">-0.3619248</td>
<td align="right">7.443102</td>
<td align="right">-3.597385</td>
<td align="right">0.0024194</td>
<td align="right">0.9999021</td>
<td align="right">-3.033249</td>
</tr>
<tr class="even">
<td>WDPCP</td>
<td align="right">-0.4924760</td>
<td align="right">7.242165</td>
<td align="right">-3.603724</td>
<td align="right">0.0026906</td>
<td align="right">0.9999021</td>
<td align="right">-3.001044</td>
</tr>
<tr class="odd">
<td>C4orf47</td>
<td align="right">0.7250651</td>
<td align="right">4.332384</td>
<td align="right">3.521895</td>
<td align="right">0.0028611</td>
<td align="right">0.9999021</td>
<td align="right">2.982283</td>
</tr>
<tr class="even">
<td>MIR762HG</td>
<td align="right">0.5507415</td>
<td align="right">4.432379</td>
<td align="right">3.376487</td>
<td align="right">0.0034605</td>
<td align="right">0.9999021</td>
<td align="right">2.923560</td>
</tr>
<tr class="odd">
<td>MACROD1</td>
<td align="right">0.5571052</td>
<td align="right">4.034856</td>
<td align="right">3.424726</td>
<td align="right">0.0035445</td>
<td align="right">0.9999021</td>
<td align="right">2.916091</td>
</tr>
<tr class="even">
<td>DHFR2</td>
<td align="right">0.8018443</td>
<td align="right">3.742089</td>
<td align="right">3.396533</td>
<td align="right">0.0036267</td>
<td align="right">0.9999021</td>
<td align="right">2.908929</td>
</tr>
</tbody>
</table>
<p>For cluster 4:</p>
<pre class="r"><code>pseudo4.full &lt;- generate.pseudobulk(cluster4.full, labels = c(&quot;stim&quot;, &quot;replicate&quot;, &quot;cell.line&quot;))
pseudo4.full &lt;- filter.pseudobulk(pseudo4.full)
d4.full &lt;- DGEList(pseudo4.full$counts)
d4.full &lt;- calcNormFactors(d4.full)
keep4.full &lt;- filterByExpr(d4.full$counts)
d4.full &lt;- d4.full[keep4.full,]

voomed4.full &lt;- voomWithDreamWeights(d4.full, formde, pseudo4.full$meta, plot=FALSE)</code></pre>
<pre><code>Memory usage to store result: &gt; 27.8 Mb 
Dividing work into 100 chunks...

Total: 83 s</code></pre>
<pre class="r"><code>fitmm4.full = dream(voomed4.full, formde, pseudo4.full$meta)</code></pre>
<pre><code>Dividing work into 100 chunks...

Total: 112 s</code></pre>
<pre class="r"><code>try(topTable(fitmm4.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>[1] 93</code></pre>
<pre class="r"><code>de.summary.full[5,1] &lt;- try(topTable(fitmm4.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()
topTable(fitmm4.full, coef=&#39;stimoxidation&#39;,  number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;Oxidation DE genes, cluster 4 of full dataset&quot;)</code></pre>
<table>
<caption>Oxidation DE genes, cluster 4 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>KCNMB2-AS1</td>
<td align="right">1.7506890</td>
<td align="right">6.833487</td>
<td align="right">7.491134</td>
<td align="right">2.00e-06</td>
<td align="right">0.0134276</td>
<td align="right">4.758234</td>
</tr>
<tr class="even">
<td>NUP210L</td>
<td align="right">1.4235512</td>
<td align="right">3.048672</td>
<td align="right">7.766317</td>
<td align="right">3.10e-06</td>
<td align="right">0.0134276</td>
<td align="right">4.666538</td>
</tr>
<tr class="odd">
<td>FUT9</td>
<td align="right">0.2903107</td>
<td align="right">8.563347</td>
<td align="right">7.714747</td>
<td align="right">3.30e-06</td>
<td align="right">0.0134276</td>
<td align="right">4.652706</td>
</tr>
<tr class="even">
<td>HMGXB4</td>
<td align="right">-0.4126549</td>
<td align="right">6.097346</td>
<td align="right">-6.586916</td>
<td align="right">7.90e-06</td>
<td align="right">0.0237601</td>
<td align="right">-4.467614</td>
</tr>
<tr class="odd">
<td>HK2</td>
<td align="right">-1.8690628</td>
<td align="right">4.472322</td>
<td align="right">-6.576990</td>
<td align="right">9.70e-06</td>
<td align="right">0.0237601</td>
<td align="right">-4.424609</td>
</tr>
<tr class="even">
<td>KRT10</td>
<td align="right">0.2260484</td>
<td align="right">7.158633</td>
<td align="right">6.596478</td>
<td align="right">1.69e-05</td>
<td align="right">0.0299964</td>
<td align="right">4.301729</td>
</tr>
<tr class="odd">
<td>TECPR2</td>
<td align="right">0.4275456</td>
<td align="right">5.436455</td>
<td align="right">6.547176</td>
<td align="right">1.71e-05</td>
<td align="right">0.0299964</td>
<td align="right">4.300041</td>
</tr>
<tr class="even">
<td>SOX21-AS1</td>
<td align="right">0.6219661</td>
<td align="right">4.562402</td>
<td align="right">6.437972</td>
<td align="right">2.13e-05</td>
<td align="right">0.0317830</td>
<td align="right">4.251297</td>
</tr>
<tr class="odd">
<td>PDK1</td>
<td align="right">-1.6145790</td>
<td align="right">5.639091</td>
<td align="right">-5.924557</td>
<td align="right">2.51e-05</td>
<td align="right">0.0317830</td>
<td align="right">-4.213836</td>
</tr>
<tr class="even">
<td>ANGPTL4</td>
<td align="right">-1.2893588</td>
<td align="right">3.380501</td>
<td align="right">-6.234102</td>
<td align="right">2.87e-05</td>
<td align="right">0.0317830</td>
<td align="right">-4.183551</td>
</tr>
</tbody>
</table>
<pre class="r"><code>try(topTable(fitmm4.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.full[5,2] &lt;- try(topTable(fitmm4.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.full[5, 2] &lt;- try(topTable(fitmm4.full, coef = &quot;stimpff&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm4.full, coef=&#39;stimpff&#39;, number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;PFF DE genes, cluster 4 of full dataset&quot;)</code></pre>
<table>
<caption>PFF DE genes, cluster 4 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>AC017101.1</td>
<td align="right">0.9405469</td>
<td align="right">3.168264</td>
<td align="right">4.839276</td>
<td align="right">0.0002345</td>
<td align="right">0.9271403</td>
<td align="right">3.678568</td>
</tr>
<tr class="even">
<td>CAV3</td>
<td align="right">1.2173238</td>
<td align="right">2.303246</td>
<td align="right">4.677827</td>
<td align="right">0.0002916</td>
<td align="right">0.9271403</td>
<td align="right">3.622677</td>
</tr>
<tr class="odd">
<td>KRT10</td>
<td align="right">0.2634934</td>
<td align="right">7.158633</td>
<td align="right">4.785050</td>
<td align="right">0.0003505</td>
<td align="right">0.9271403</td>
<td align="right">3.574832</td>
</tr>
<tr class="even">
<td>PACRGL</td>
<td align="right">-0.3068847</td>
<td align="right">5.451265</td>
<td align="right">-4.616364</td>
<td align="right">0.0004545</td>
<td align="right">0.9271403</td>
<td align="right">-3.506206</td>
</tr>
<tr class="odd">
<td>LMBRD2</td>
<td align="right">-0.8389002</td>
<td align="right">4.124352</td>
<td align="right">-4.172038</td>
<td align="right">0.0006391</td>
<td align="right">0.9271403</td>
<td align="right">-3.414471</td>
</tr>
<tr class="even">
<td>ARGLU1</td>
<td align="right">-0.3791068</td>
<td align="right">7.773896</td>
<td align="right">-4.314113</td>
<td align="right">0.0007155</td>
<td align="right">0.9271403</td>
<td align="right">-3.383580</td>
</tr>
<tr class="odd">
<td>SAXO2</td>
<td align="right">-1.3623949</td>
<td align="right">3.437534</td>
<td align="right">-4.278611</td>
<td align="right">0.0008179</td>
<td align="right">0.9271403</td>
<td align="right">-3.346677</td>
</tr>
<tr class="even">
<td>ZNF518A</td>
<td align="right">-0.3203980</td>
<td align="right">6.307487</td>
<td align="right">-3.673040</td>
<td align="right">0.0022402</td>
<td align="right">0.9271403</td>
<td align="right">-3.056389</td>
</tr>
<tr class="odd">
<td>CD302</td>
<td align="right">0.7113723</td>
<td align="right">3.385529</td>
<td align="right">3.631201</td>
<td align="right">0.0022529</td>
<td align="right">0.9271403</td>
<td align="right">3.054690</td>
</tr>
<tr class="even">
<td>DHRS4</td>
<td align="right">-0.8045192</td>
<td align="right">3.115295</td>
<td align="right">-3.639206</td>
<td align="right">0.0023956</td>
<td align="right">0.9271403</td>
<td align="right">-3.036225</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#now repeat with the neuron dataset
pseudo4.neuron &lt;- generate.pseudobulk(cluster4.neuron, labels = c(&quot;stim&quot;, &quot;replicate&quot;, &quot;cell.line&quot;))
pseudo4.neuron &lt;- filter.pseudobulk(pseudo4.neuron)
d4.neuron &lt;- DGEList(pseudo4.neuron$counts)
d4.neuron &lt;- calcNormFactors(d4.neuron)
keep4.neuron &lt;- filterByExpr(d4.neuron$counts)
d4.neuron &lt;- d4.neuron[keep4.neuron,]

voomed4.neuron &lt;- voomWithDreamWeights(d4.neuron, formde, pseudo4.neuron$meta, plot=FALSE)</code></pre>
<pre><code>Memory usage to store result: &gt; 26.6 Mb 
Dividing work into 100 chunks...

Total: 83 s</code></pre>
<pre class="r"><code>fitmm4.neuron = dream(voomed4.neuron, formde, pseudo4.neuron$meta)</code></pre>
<pre><code>Dividing work into 100 chunks...

Total: 110 s</code></pre>
<pre class="r"><code>try(topTable(fitmm4.neuron, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>[1] 9</code></pre>
<pre class="r"><code>de.summary.neuron[5,1] &lt;- try(topTable(fitmm4.neuron, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()
topTable(fitmm4.neuron, coef=&#39;stimoxidation&#39;,  number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;Oxidation DE genes, cluster 4 of full dataset&quot;)</code></pre>
<table>
<caption>Oxidation DE genes, cluster 4 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>NXN</td>
<td align="right">0.3603022</td>
<td align="right">6.631067</td>
<td align="right">7.332037</td>
<td align="right">1.70e-06</td>
<td align="right">0.0191960</td>
<td align="right">4.787937</td>
</tr>
<tr class="even">
<td>TNIP1</td>
<td align="right">-0.8050977</td>
<td align="right">4.661778</td>
<td align="right">-6.017840</td>
<td align="right">1.09e-05</td>
<td align="right">0.0431595</td>
<td align="right">-4.399184</td>
</tr>
<tr class="odd">
<td>EGLN1</td>
<td align="right">-0.6822963</td>
<td align="right">6.166482</td>
<td align="right">-6.256884</td>
<td align="right">1.14e-05</td>
<td align="right">0.0431595</td>
<td align="right">-4.389412</td>
</tr>
<tr class="even">
<td>PDK1</td>
<td align="right">-1.4330293</td>
<td align="right">5.134943</td>
<td align="right">-5.668574</td>
<td align="right">2.24e-05</td>
<td align="right">0.0492382</td>
<td align="right">-4.239780</td>
</tr>
<tr class="odd">
<td>P4HA1</td>
<td align="right">-0.9977953</td>
<td align="right">6.636905</td>
<td align="right">-5.901018</td>
<td align="right">2.27e-05</td>
<td align="right">0.0492382</td>
<td align="right">-4.236963</td>
</tr>
<tr class="even">
<td>INSIG2</td>
<td align="right">-0.8130412</td>
<td align="right">5.152190</td>
<td align="right">-5.533468</td>
<td align="right">2.97e-05</td>
<td align="right">0.0492382</td>
<td align="right">-4.175709</td>
</tr>
<tr class="odd">
<td>KDM3A</td>
<td align="right">-0.6704342</td>
<td align="right">5.869513</td>
<td align="right">-5.887675</td>
<td align="right">3.78e-05</td>
<td align="right">0.0492382</td>
<td align="right">-4.120662</td>
</tr>
<tr class="even">
<td>MT-ND1</td>
<td align="right">0.3820881</td>
<td align="right">10.215100</td>
<td align="right">5.623431</td>
<td align="right">3.81e-05</td>
<td align="right">0.0492382</td>
<td align="right">4.118545</td>
</tr>
<tr class="odd">
<td>PRPF38B</td>
<td align="right">-0.2231305</td>
<td align="right">6.215668</td>
<td align="right">-5.873444</td>
<td align="right">3.89e-05</td>
<td align="right">0.0492382</td>
<td align="right">-4.113920</td>
</tr>
<tr class="even">
<td>PLOD2</td>
<td align="right">-0.8758898</td>
<td align="right">6.001853</td>
<td align="right">-5.509124</td>
<td align="right">4.81e-05</td>
<td align="right">0.0507314</td>
<td align="right">-4.064523</td>
</tr>
</tbody>
</table>
<pre class="r"><code>try(topTable(fitmm4.neuron, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.neuron[5,2] &lt;- try(topTable(fitmm4.neuron, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.neuron[5, 2] &lt;- try(topTable(fitmm4.neuron, coef = &quot;stimpff&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm4.neuron, coef=&#39;stimpff&#39;, number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;PFF DE genes, cluster 4 of full dataset&quot;)</code></pre>
<table>
<caption>PFF DE genes, cluster 4 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>KPNA6</td>
<td align="right">0.3496555</td>
<td align="right">5.795902</td>
<td align="right">4.412612</td>
<td align="right">0.0004327</td>
<td align="right">0.9999474</td>
<td align="right">3.519283</td>
</tr>
<tr class="even">
<td>KCNAB2</td>
<td align="right">-1.1770133</td>
<td align="right">2.574797</td>
<td align="right">-4.320307</td>
<td align="right">0.0005773</td>
<td align="right">0.9999474</td>
<td align="right">-3.442061</td>
</tr>
<tr class="odd">
<td>AC002558.3</td>
<td align="right">0.7514077</td>
<td align="right">2.799136</td>
<td align="right">3.939928</td>
<td align="right">0.0009601</td>
<td align="right">0.9999474</td>
<td align="right">3.301970</td>
</tr>
<tr class="even">
<td>PYM1</td>
<td align="right">0.3587808</td>
<td align="right">4.275162</td>
<td align="right">3.968848</td>
<td align="right">0.0010817</td>
<td align="right">0.9999474</td>
<td align="right">3.268362</td>
</tr>
<tr class="odd">
<td>DHODH</td>
<td align="right">-0.7350070</td>
<td align="right">3.011927</td>
<td align="right">-3.974690</td>
<td align="right">0.0010892</td>
<td align="right">0.9999474</td>
<td align="right">-3.266423</td>
</tr>
<tr class="even">
<td>ERCC4</td>
<td align="right">-0.5713403</td>
<td align="right">4.832603</td>
<td align="right">-3.731553</td>
<td align="right">0.0015274</td>
<td align="right">0.9999474</td>
<td align="right">-3.169418</td>
</tr>
<tr class="odd">
<td>THUMPD1</td>
<td align="right">0.2343289</td>
<td align="right">5.696851</td>
<td align="right">3.719175</td>
<td align="right">0.0018679</td>
<td align="right">0.9999474</td>
<td align="right">3.110473</td>
</tr>
<tr class="even">
<td>LCA5</td>
<td align="right">0.5874278</td>
<td align="right">4.188372</td>
<td align="right">3.683834</td>
<td align="right">0.0020621</td>
<td align="right">0.9999474</td>
<td align="right">3.081133</td>
</tr>
<tr class="odd">
<td>GTF2F1</td>
<td align="right">-0.3333143</td>
<td align="right">5.747489</td>
<td align="right">-3.661073</td>
<td align="right">0.0020810</td>
<td align="right">0.9999474</td>
<td align="right">-3.078416</td>
</tr>
<tr class="even">
<td>KIAA0100</td>
<td align="right">-0.4770489</td>
<td align="right">5.411110</td>
<td align="right">-3.574942</td>
<td align="right">0.0025672</td>
<td align="right">0.9999474</td>
<td align="right">-3.015306</td>
</tr>
</tbody>
</table>
<p>For cluster 5:</p>
<pre class="r"><code>pseudo5.full &lt;- generate.pseudobulk(cluster5.full, labels = c(&quot;stim&quot;, &quot;replicate&quot;, &quot;cell.line&quot;))
pseudo5.full &lt;- filter.pseudobulk(pseudo5.full)
d5.full &lt;- DGEList(pseudo5.full$counts)
d5.full &lt;- calcNormFactors(d5.full)
keep5.full &lt;- filterByExpr(d5.full$counts)
d5.full &lt;- d5.full[keep5.full,]

voomed5.full &lt;- voomWithDreamWeights(d5.full, formde, pseudo5.full$meta, plot=FALSE)</code></pre>
<pre><code>Memory usage to store result: &gt; 29.4 Mb 
Dividing work into 100 chunks...

Total: 83 s</code></pre>
<pre class="r"><code>fitmm5.full = dream(voomed5.full, formde, pseudo5.full$meta)</code></pre>
<pre><code>Dividing work into 100 chunks...

Total: 113 s</code></pre>
<pre class="r"><code>try(topTable(fitmm5.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>[1] 63</code></pre>
<pre class="r"><code>de.summary.full[6,1] &lt;- try(topTable(fitmm5.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()
topTable(fitmm5.full, coef=&#39;stimoxidation&#39;,  number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;Oxidation DE genes, cluster 5 of full dataset&quot;)</code></pre>
<table>
<caption>Oxidation DE genes, cluster 5 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>CABIN1</td>
<td align="right">-0.3226990</td>
<td align="right">5.712278</td>
<td align="right">-7.693902</td>
<td align="right">9.00e-07</td>
<td align="right">0.0114363</td>
<td align="right">-4.910284</td>
</tr>
<tr class="even">
<td>LINC00641</td>
<td align="right">0.5824095</td>
<td align="right">2.820588</td>
<td align="right">6.922621</td>
<td align="right">3.40e-06</td>
<td align="right">0.0155825</td>
<td align="right">4.646743</td>
</tr>
<tr class="odd">
<td>ATP13A2</td>
<td align="right">0.2652062</td>
<td align="right">4.052815</td>
<td align="right">6.814627</td>
<td align="right">4.10e-06</td>
<td align="right">0.0155825</td>
<td align="right">4.604713</td>
</tr>
<tr class="even">
<td>SNHG30</td>
<td align="right">0.2797867</td>
<td align="right">5.026505</td>
<td align="right">6.217065</td>
<td align="right">7.20e-06</td>
<td align="right">0.0155825</td>
<td align="right">4.486288</td>
</tr>
<tr class="odd">
<td>ZNF160</td>
<td align="right">-0.4825544</td>
<td align="right">5.044213</td>
<td align="right">-6.887812</td>
<td align="right">7.40e-06</td>
<td align="right">0.0155825</td>
<td align="right">-4.482428</td>
</tr>
<tr class="even">
<td>KCNMB2-AS1</td>
<td align="right">1.4149666</td>
<td align="right">5.101941</td>
<td align="right">6.792413</td>
<td align="right">7.40e-06</td>
<td align="right">0.0155825</td>
<td align="right">4.480840</td>
</tr>
<tr class="odd">
<td>TET1</td>
<td align="right">-0.4458017</td>
<td align="right">7.402212</td>
<td align="right">-6.667060</td>
<td align="right">1.01e-05</td>
<td align="right">0.0181621</td>
<td align="right">-4.414805</td>
</tr>
<tr class="even">
<td>ESRRB</td>
<td align="right">1.0663174</td>
<td align="right">2.020671</td>
<td align="right">6.004214</td>
<td align="right">1.71e-05</td>
<td align="right">0.0193160</td>
<td align="right">4.299490</td>
</tr>
<tr class="odd">
<td>LINC01159</td>
<td align="right">0.4438393</td>
<td align="right">4.039626</td>
<td align="right">5.777702</td>
<td align="right">1.78e-05</td>
<td align="right">0.0193160</td>
<td align="right">4.290532</td>
</tr>
<tr class="even">
<td>MAT2A</td>
<td align="right">0.3251612</td>
<td align="right">6.844898</td>
<td align="right">6.315252</td>
<td align="right">1.81e-05</td>
<td align="right">0.0193160</td>
<td align="right">4.286744</td>
</tr>
</tbody>
</table>
<pre class="r"><code>try(topTable(fitmm5.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.full[6,2] &lt;- try(topTable(fitmm5.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.full[6, 2] &lt;- try(topTable(fitmm5.full, coef = &quot;stimpff&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm5.full, coef=&#39;stimpff&#39;, number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;PFF DE genes, cluster 5 of full dataset&quot;)</code></pre>
<table>
<caption>PFF DE genes, cluster 5 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>DUS1L</td>
<td align="right">-0.4208671</td>
<td align="right">3.805922</td>
<td align="right">-5.527615</td>
<td align="right">0.0000301</td>
<td align="right">0.3781863</td>
<td align="right">-4.172902</td>
</tr>
<tr class="even">
<td>DUSP6</td>
<td align="right">1.2298776</td>
<td align="right">2.980812</td>
<td align="right">4.598290</td>
<td align="right">0.0002373</td>
<td align="right">0.9717995</td>
<td align="right">3.675631</td>
</tr>
<tr class="odd">
<td>FAXDC2</td>
<td align="right">0.6521356</td>
<td align="right">3.937118</td>
<td align="right">4.549936</td>
<td align="right">0.0002749</td>
<td align="right">0.9717995</td>
<td align="right">3.637837</td>
</tr>
<tr class="even">
<td>ERCC4</td>
<td align="right">-0.4475125</td>
<td align="right">4.832427</td>
<td align="right">-4.450290</td>
<td align="right">0.0003091</td>
<td align="right">0.9717995</td>
<td align="right">-3.607534</td>
</tr>
<tr class="odd">
<td>NCAPG</td>
<td align="right">-0.3714164</td>
<td align="right">7.473330</td>
<td align="right">-4.399441</td>
<td align="right">0.0005133</td>
<td align="right">0.9998241</td>
<td align="right">-3.473739</td>
</tr>
<tr class="even">
<td>PRIMPOL</td>
<td align="right">-0.3381303</td>
<td align="right">5.115861</td>
<td align="right">-4.270537</td>
<td align="right">0.0005829</td>
<td align="right">0.9998241</td>
<td align="right">-3.439432</td>
</tr>
<tr class="odd">
<td>OSGEP</td>
<td align="right">0.2379493</td>
<td align="right">5.142415</td>
<td align="right">4.224235</td>
<td align="right">0.0008121</td>
<td align="right">0.9998241</td>
<td align="right">3.348642</td>
</tr>
<tr class="even">
<td>MBD4</td>
<td align="right">-0.3957156</td>
<td align="right">5.154262</td>
<td align="right">-4.022324</td>
<td align="right">0.0008259</td>
<td align="right">0.9998241</td>
<td align="right">-3.343982</td>
</tr>
<tr class="odd">
<td>ZBTB1</td>
<td align="right">-0.4628341</td>
<td align="right">5.648933</td>
<td align="right">-4.051802</td>
<td align="right">0.0009247</td>
<td align="right">0.9998241</td>
<td align="right">-3.312482</td>
</tr>
<tr class="even">
<td>ZNF184</td>
<td align="right">-0.4232080</td>
<td align="right">3.971922</td>
<td align="right">-4.118462</td>
<td align="right">0.0009481</td>
<td align="right">0.9998241</td>
<td align="right">-3.305504</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#now repeat with the neuron dataset
pseudo5.neuron &lt;- generate.pseudobulk(cluster5.neuron, labels = c(&quot;stim&quot;, &quot;replicate&quot;, &quot;cell.line&quot;))
pseudo5.neuron &lt;- filter.pseudobulk(pseudo5.neuron)
d5.neuron &lt;- DGEList(pseudo5.neuron$counts)
d5.neuron &lt;- calcNormFactors(d5.neuron)
keep5.neuron &lt;- filterByExpr(d5.neuron$counts)
d5.neuron &lt;- d5.neuron[keep5.neuron,]

voomed5.neuron &lt;- voomWithDreamWeights(d5.neuron, formde, pseudo5.neuron$meta, plot=FALSE)</code></pre>
<pre><code>Memory usage to store result: &gt; 29.6 Mb 
Dividing work into 100 chunks...

Total: 84 s</code></pre>
<pre class="r"><code>fitmm5.neuron = dream(voomed5.neuron, formde, pseudo5.neuron$meta)</code></pre>
<pre><code>Dividing work into 100 chunks...

Total: 113 s</code></pre>
<pre class="r"><code>try(topTable(fitmm5.neuron, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>[1] 628</code></pre>
<pre class="r"><code>de.summary.neuron[6,1] &lt;- try(topTable(fitmm5.neuron, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()
topTable(fitmm5.neuron, coef=&#39;stimoxidation&#39;,  number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;Oxidation DE genes, cluster 5 of full dataset&quot;)</code></pre>
<table>
<caption>Oxidation DE genes, cluster 5 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>OXLD1</td>
<td align="right">0.6506033</td>
<td align="right">5.032072</td>
<td align="right">9.638792</td>
<td align="right">0.0e+00</td>
<td align="right">0.0005479</td>
<td align="right">5.477367</td>
</tr>
<tr class="even">
<td>BRWD1</td>
<td align="right">-0.3023209</td>
<td align="right">6.565273</td>
<td align="right">-8.018519</td>
<td align="right">2.0e-07</td>
<td align="right">0.0010952</td>
<td align="right">-5.167592</td>
</tr>
<tr class="odd">
<td>LINC01937</td>
<td align="right">1.9765433</td>
<td align="right">3.579522</td>
<td align="right">9.139882</td>
<td align="right">3.0e-07</td>
<td align="right">0.0010952</td>
<td align="right">5.151160</td>
</tr>
<tr class="even">
<td>KAT6B</td>
<td align="right">-0.5639522</td>
<td align="right">6.844290</td>
<td align="right">-7.925762</td>
<td align="right">6.0e-07</td>
<td align="right">0.0018977</td>
<td align="right">-4.987275</td>
</tr>
<tr class="odd">
<td>RLF</td>
<td align="right">-0.8086849</td>
<td align="right">6.255824</td>
<td align="right">-7.329228</td>
<td align="right">8.0e-07</td>
<td align="right">0.0018977</td>
<td align="right">-4.927390</td>
</tr>
<tr class="even">
<td>AC010210.1</td>
<td align="right">1.4062683</td>
<td align="right">3.299308</td>
<td align="right">8.285449</td>
<td align="right">9.0e-07</td>
<td align="right">0.0018977</td>
<td align="right">4.912955</td>
</tr>
<tr class="odd">
<td>GLG1</td>
<td align="right">-0.2618926</td>
<td align="right">7.013333</td>
<td align="right">-7.904429</td>
<td align="right">1.6e-06</td>
<td align="right">0.0023382</td>
<td align="right">-4.803689</td>
</tr>
<tr class="even">
<td>XPO1</td>
<td align="right">-0.4312338</td>
<td align="right">6.715333</td>
<td align="right">-8.082489</td>
<td align="right">1.6e-06</td>
<td align="right">0.0023382</td>
<td align="right">-4.803363</td>
</tr>
<tr class="odd">
<td>PIAS2</td>
<td align="right">-0.5179374</td>
<td align="right">6.325787</td>
<td align="right">-7.809022</td>
<td align="right">1.7e-06</td>
<td align="right">0.0023382</td>
<td align="right">-4.791179</td>
</tr>
<tr class="even">
<td>SPICE1</td>
<td align="right">-0.6261213</td>
<td align="right">4.913316</td>
<td align="right">-7.604679</td>
<td align="right">2.2e-06</td>
<td align="right">0.0026318</td>
<td align="right">-4.733843</td>
</tr>
</tbody>
</table>
<pre class="r"><code>try(topTable(fitmm5.neuron, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.neuron[6,2] &lt;- try(topTable(fitmm5.neuron, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.neuron[6, 2] &lt;- try(topTable(fitmm5.neuron, coef = &quot;stimpff&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm5.neuron, coef=&#39;stimpff&#39;, number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;PFF DE genes, cluster 5 of full dataset&quot;)</code></pre>
<table>
<caption>PFF DE genes, cluster 5 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>BMPR1A</td>
<td align="right">0.3863877</td>
<td align="right">6.507488</td>
<td align="right">5.482367</td>
<td align="right">0.0000710</td>
<td align="right">0.9007691</td>
<td align="right">3.973006</td>
</tr>
<tr class="even">
<td>PRR5</td>
<td align="right">0.9878129</td>
<td align="right">4.219951</td>
<td align="right">4.605775</td>
<td align="right">0.0002992</td>
<td align="right">0.9255204</td>
<td align="right">3.616025</td>
</tr>
<tr class="odd">
<td>BICRAL</td>
<td align="right">0.4529146</td>
<td align="right">5.248111</td>
<td align="right">4.576396</td>
<td align="right">0.0003043</td>
<td align="right">0.9255204</td>
<td align="right">3.611624</td>
</tr>
<tr class="even">
<td>HAUS2</td>
<td align="right">-0.3301033</td>
<td align="right">4.719675</td>
<td align="right">-4.609727</td>
<td align="right">0.0003449</td>
<td align="right">0.9255204</td>
<td align="right">-3.579037</td>
</tr>
<tr class="odd">
<td>MCM4</td>
<td align="right">-0.5992250</td>
<td align="right">4.494232</td>
<td align="right">-4.374155</td>
<td align="right">0.0006202</td>
<td align="right">0.9255204</td>
<td align="right">-3.422633</td>
</tr>
<tr class="even">
<td>SLC41A3</td>
<td align="right">0.3827104</td>
<td align="right">4.574850</td>
<td align="right">4.226655</td>
<td align="right">0.0006434</td>
<td align="right">0.9255204</td>
<td align="right">3.412646</td>
</tr>
<tr class="odd">
<td>CXorf40B</td>
<td align="right">-0.8306051</td>
<td align="right">2.633517</td>
<td align="right">-4.177124</td>
<td align="right">0.0006759</td>
<td align="right">0.9255204</td>
<td align="right">-3.399180</td>
</tr>
<tr class="even">
<td>CP</td>
<td align="right">-1.5400332</td>
<td align="right">5.573696</td>
<td align="right">-4.152740</td>
<td align="right">0.0007520</td>
<td align="right">0.9255204</td>
<td align="right">-3.369870</td>
</tr>
<tr class="odd">
<td>CNNM2</td>
<td align="right">0.4792987</td>
<td align="right">4.898028</td>
<td align="right">4.122968</td>
<td align="right">0.0008106</td>
<td align="right">0.9255204</td>
<td align="right">3.349144</td>
</tr>
<tr class="even">
<td>CLCN5</td>
<td align="right">0.5734715</td>
<td align="right">4.251115</td>
<td align="right">4.152036</td>
<td align="right">0.0008667</td>
<td align="right">0.9255204</td>
<td align="right">3.330570</td>
</tr>
</tbody>
</table>
<p>For cluster 6:</p>
<pre class="r"><code>pseudo6.full &lt;- generate.pseudobulk(cluster6.full, labels = c(&quot;stim&quot;, &quot;replicate&quot;, &quot;cell.line&quot;))
pseudo6.full &lt;- filter.pseudobulk(pseudo6.full)
d6.full &lt;- DGEList(pseudo6.full$counts)
d6.full &lt;- calcNormFactors(d6.full)
keep6.full &lt;- filterByExpr(d6.full$counts)
d6.full &lt;- d6.full[keep6.full,]

voomed6.full &lt;- voomWithDreamWeights(d6.full, formde, pseudo6.full$meta, plot=FALSE)</code></pre>
<pre><code>Memory usage to store result: &gt; 23.2 Mb 
Dividing work into 100 chunks...

Total: 81 s</code></pre>
<pre class="r"><code>fitmm6.full = dream(voomed6.full, formde, pseudo6.full$meta)</code></pre>
<pre><code>Dividing work into 100 chunks...

Total: 105 s</code></pre>
<pre class="r"><code>try(topTable(fitmm6.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>[1] 108</code></pre>
<pre class="r"><code>de.summary.full[7,1] &lt;- try(topTable(fitmm6.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()
topTable(fitmm6.full, coef=&#39;stimoxidation&#39;,  number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;Oxidation DE genes, cluster 6 of full dataset&quot;)</code></pre>
<table>
<caption>Oxidation DE genes, cluster 6 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>BNIP3</td>
<td align="right">-1.4412810</td>
<td align="right">8.761414</td>
<td align="right">-6.919662</td>
<td align="right">1.80e-06</td>
<td align="right">0.0161107</td>
<td align="right">-4.773271</td>
</tr>
<tr class="even">
<td>DNAAF2</td>
<td align="right">0.7666519</td>
<td align="right">4.800939</td>
<td align="right">7.014520</td>
<td align="right">3.20e-06</td>
<td align="right">0.0161107</td>
<td align="right">4.654804</td>
</tr>
<tr class="odd">
<td>MRGBP</td>
<td align="right">-0.4446116</td>
<td align="right">4.614929</td>
<td align="right">-6.153146</td>
<td align="right">8.20e-06</td>
<td align="right">0.0169048</td>
<td align="right">-4.458633</td>
</tr>
<tr class="even">
<td>MTFP1</td>
<td align="right">-1.4431802</td>
<td align="right">4.249224</td>
<td align="right">-6.313521</td>
<td align="right">1.00e-05</td>
<td align="right">0.0169048</td>
<td align="right">-4.417629</td>
</tr>
<tr class="odd">
<td>PSMG4</td>
<td align="right">1.1586464</td>
<td align="right">5.707574</td>
<td align="right">6.317818</td>
<td align="right">1.02e-05</td>
<td align="right">0.0169048</td>
<td align="right">4.412327</td>
</tr>
<tr class="even">
<td>MAP3K20</td>
<td align="right">-1.2011682</td>
<td align="right">5.008467</td>
<td align="right">-5.987766</td>
<td align="right">1.16e-05</td>
<td align="right">0.0169048</td>
<td align="right">-4.385800</td>
</tr>
<tr class="odd">
<td>CEBPG</td>
<td align="right">-0.6599012</td>
<td align="right">6.742575</td>
<td align="right">-6.582657</td>
<td align="right">1.20e-05</td>
<td align="right">0.0169048</td>
<td align="right">-4.377118</td>
</tr>
<tr class="even">
<td>STAG3</td>
<td align="right">0.9059157</td>
<td align="right">5.362265</td>
<td align="right">6.445060</td>
<td align="right">1.50e-05</td>
<td align="right">0.0169048</td>
<td align="right">4.329217</td>
</tr>
<tr class="odd">
<td>NIFK-AS1</td>
<td align="right">0.8079359</td>
<td align="right">5.574152</td>
<td align="right">6.055403</td>
<td align="right">1.64e-05</td>
<td align="right">0.0169048</td>
<td align="right">4.308549</td>
</tr>
<tr class="even">
<td>PDK3</td>
<td align="right">-1.0586455</td>
<td align="right">5.372225</td>
<td align="right">-5.773256</td>
<td align="right">1.80e-05</td>
<td align="right">0.0169048</td>
<td align="right">-4.288482</td>
</tr>
</tbody>
</table>
<pre class="r"><code>try(topTable(fitmm6.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.full[7,2] &lt;- try(topTable(fitmm6.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.full[7, 2] &lt;- try(topTable(fitmm6.full, coef = &quot;stimpff&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm6.full, coef=&#39;stimpff&#39;, number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;PFF DE genes, cluster 6 of full dataset&quot;)</code></pre>
<table>
<caption>PFF DE genes, cluster 6 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>MT-CO2</td>
<td align="right">0.6761947</td>
<td align="right">11.016669</td>
<td align="right">5.900091</td>
<td align="right">0.0000224</td>
<td align="right">0.2224028</td>
<td align="right">4.239683</td>
</tr>
<tr class="even">
<td>MT-ND3</td>
<td align="right">0.7983265</td>
<td align="right">10.178307</td>
<td align="right">4.262758</td>
<td align="right">0.0005965</td>
<td align="right">0.9983607</td>
<td align="right">3.433190</td>
</tr>
<tr class="odd">
<td>FAM104B</td>
<td align="right">-0.5342687</td>
<td align="right">4.275711</td>
<td align="right">-4.038762</td>
<td align="right">0.0011256</td>
<td align="right">0.9983607</td>
<td align="right">-3.257098</td>
</tr>
<tr class="even">
<td>MT-ND6</td>
<td align="right">0.6030821</td>
<td align="right">6.691457</td>
<td align="right">3.952450</td>
<td align="right">0.0012637</td>
<td align="right">0.9983607</td>
<td align="right">3.224088</td>
</tr>
<tr class="odd">
<td>MED17</td>
<td align="right">-0.4854982</td>
<td align="right">6.271810</td>
<td align="right">-3.737776</td>
<td align="right">0.0018018</td>
<td align="right">0.9983607</td>
<td align="right">-3.121097</td>
</tr>
<tr class="even">
<td>MT-ATP6</td>
<td align="right">0.8109730</td>
<td align="right">10.717826</td>
<td align="right">3.635792</td>
<td align="right">0.0022263</td>
<td align="right">0.9983607</td>
<td align="right">3.058256</td>
</tr>
<tr class="odd">
<td>TMEM177</td>
<td align="right">0.7745938</td>
<td align="right">3.289449</td>
<td align="right">3.560103</td>
<td align="right">0.0022371</td>
<td align="right">0.9983607</td>
<td align="right">3.056800</td>
</tr>
<tr class="even">
<td>DHFR2</td>
<td align="right">0.8239828</td>
<td align="right">3.718825</td>
<td align="right">3.608116</td>
<td align="right">0.0023191</td>
<td align="right">0.9983607</td>
<td align="right">3.046002</td>
</tr>
<tr class="odd">
<td>PDCD5</td>
<td align="right">0.4425578</td>
<td align="right">6.320520</td>
<td align="right">3.495936</td>
<td align="right">0.0031004</td>
<td align="right">0.9983607</td>
<td align="right">2.957600</td>
</tr>
<tr class="even">
<td>LIMD1</td>
<td align="right">1.0802217</td>
<td align="right">3.316949</td>
<td align="right">3.448920</td>
<td align="right">0.0034068</td>
<td align="right">0.9983607</td>
<td align="right">2.928426</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#now repeat with the neuron dataset
pseudo6.neuron &lt;- generate.pseudobulk(cluster6.neuron, labels = c(&quot;stim&quot;, &quot;replicate&quot;, &quot;cell.line&quot;))
pseudo6.neuron &lt;- filter.pseudobulk(pseudo6.neuron)
d6.neuron &lt;- DGEList(pseudo6.neuron$counts)
d6.neuron &lt;- calcNormFactors(d6.neuron)
keep6.neuron &lt;- filterByExpr(d6.neuron$counts)
d6.neuron &lt;- d6.neuron[keep6.neuron,]

voomed6.neuron &lt;- voomWithDreamWeights(d6.neuron, formde, pseudo6.neuron$meta, plot=FALSE)</code></pre>
<pre><code>Memory usage to store result: &gt; 25.9 Mb 
Dividing work into 100 chunks...

Total: 82 s</code></pre>
<pre class="r"><code>fitmm6.neuron = dream(voomed6.neuron, formde, pseudo6.neuron$meta)</code></pre>
<pre><code>Dividing work into 100 chunks...

Total: 110 s</code></pre>
<pre class="r"><code>try(topTable(fitmm6.neuron, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>[1] 91</code></pre>
<pre class="r"><code>de.summary.neuron[7,1] &lt;- try(topTable(fitmm6.neuron, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()
topTable(fitmm6.neuron, coef=&#39;stimoxidation&#39;,  number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;Oxidation DE genes, cluster 6 of full dataset&quot;)</code></pre>
<table>
<caption>Oxidation DE genes, cluster 6 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>OGG1</td>
<td align="right">0.4061381</td>
<td align="right">5.580582</td>
<td align="right">9.567692</td>
<td align="right">2.0e-07</td>
<td align="right">0.0019583</td>
<td align="right">5.228800</td>
</tr>
<tr class="even">
<td>SLC2A1</td>
<td align="right">-1.1719850</td>
<td align="right">3.249230</td>
<td align="right">-7.977235</td>
<td align="right">1.7e-06</td>
<td align="right">0.0085267</td>
<td align="right">-4.784307</td>
</tr>
<tr class="odd">
<td>AP3B2</td>
<td align="right">0.3723062</td>
<td align="right">6.005660</td>
<td align="right">7.321230</td>
<td align="right">2.2e-06</td>
<td align="right">0.0085267</td>
<td align="right">4.731504</td>
</tr>
<tr class="even">
<td>RLF</td>
<td align="right">-0.8721759</td>
<td align="right">6.913154</td>
<td align="right">-6.511614</td>
<td align="right">5.3e-06</td>
<td align="right">0.0114563</td>
<td align="right">-4.551549</td>
</tr>
<tr class="odd">
<td>KDM3A</td>
<td align="right">-1.6378514</td>
<td align="right">5.543659</td>
<td align="right">-6.704668</td>
<td align="right">6.4e-06</td>
<td align="right">0.0114563</td>
<td align="right">-4.512568</td>
</tr>
<tr class="even">
<td>WNK3</td>
<td align="right">-0.3425302</td>
<td align="right">6.623782</td>
<td align="right">-6.736769</td>
<td align="right">6.6e-06</td>
<td align="right">0.0114563</td>
<td align="right">-4.506318</td>
</tr>
<tr class="odd">
<td>AC100810.1</td>
<td align="right">0.6169508</td>
<td align="right">3.010015</td>
<td align="right">7.157406</td>
<td align="right">7.0e-06</td>
<td align="right">0.0114563</td>
<td align="right">4.494091</td>
</tr>
<tr class="even">
<td>STIM1</td>
<td align="right">0.7892676</td>
<td align="right">4.414586</td>
<td align="right">6.918648</td>
<td align="right">9.3e-06</td>
<td align="right">0.0115153</td>
<td align="right">4.433909</td>
</tr>
<tr class="odd">
<td>MIR210HG</td>
<td align="right">-1.6526982</td>
<td align="right">3.479361</td>
<td align="right">-6.524671</td>
<td align="right">9.8e-06</td>
<td align="right">0.0115153</td>
<td align="right">-4.421007</td>
</tr>
<tr class="even">
<td>TFRC</td>
<td align="right">-1.0936335</td>
<td align="right">5.058094</td>
<td align="right">-6.342139</td>
<td align="right">1.2e-05</td>
<td align="right">0.0115153</td>
<td align="right">-4.378312</td>
</tr>
</tbody>
</table>
<pre class="r"><code>try(topTable(fitmm6.neuron, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.neuron[7,2] &lt;- try(topTable(fitmm6.neuron, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.neuron[7, 2] &lt;- try(topTable(fitmm6.neuron, coef = &quot;stimpff&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm6.neuron, coef=&#39;stimpff&#39;, number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;PFF DE genes, cluster 6 of full dataset&quot;)</code></pre>
<table>
<caption>PFF DE genes, cluster 6 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>AP3B2</td>
<td align="right">0.3493287</td>
<td align="right">6.005660</td>
<td align="right">4.633762</td>
<td align="right">0.0002409</td>
<td align="right">0.9994322</td>
<td align="right">3.671705</td>
</tr>
<tr class="even">
<td>STIM1</td>
<td align="right">0.6525948</td>
<td align="right">4.414586</td>
<td align="right">4.116702</td>
<td align="right">0.0011738</td>
<td align="right">0.9994322</td>
<td align="right">3.245178</td>
</tr>
<tr class="odd">
<td>AL049840.2</td>
<td align="right">0.7727177</td>
<td align="right">2.921171</td>
<td align="right">3.841311</td>
<td align="right">0.0013081</td>
<td align="right">0.9994322</td>
<td align="right">3.214193</td>
</tr>
<tr class="even">
<td>LDOC1</td>
<td align="right">-0.4738864</td>
<td align="right">5.040902</td>
<td align="right">-3.891656</td>
<td align="right">0.0017351</td>
<td align="right">0.9994322</td>
<td align="right">-3.132185</td>
</tr>
<tr class="odd">
<td>EMC1</td>
<td align="right">-0.3227851</td>
<td align="right">4.310517</td>
<td align="right">-3.699554</td>
<td align="right">0.0021756</td>
<td align="right">0.9994322</td>
<td align="right">-3.065145</td>
</tr>
<tr class="even">
<td>LINC02789</td>
<td align="right">1.0064553</td>
<td align="right">3.427892</td>
<td align="right">3.612559</td>
<td align="right">0.0021767</td>
<td align="right">0.9994322</td>
<td align="right">3.064999</td>
</tr>
<tr class="odd">
<td>GCA</td>
<td align="right">-0.7520607</td>
<td align="right">3.449475</td>
<td align="right">-3.620794</td>
<td align="right">0.0024958</td>
<td align="right">0.9994322</td>
<td align="right">-3.023853</td>
</tr>
<tr class="even">
<td>SPSB3</td>
<td align="right">-0.4324635</td>
<td align="right">5.080607</td>
<td align="right">-3.558854</td>
<td align="right">0.0025639</td>
<td align="right">0.9994322</td>
<td align="right">-3.015691</td>
</tr>
<tr class="odd">
<td>CHKB-DT</td>
<td align="right">0.6338893</td>
<td align="right">4.241227</td>
<td align="right">3.571986</td>
<td align="right">0.0030046</td>
<td align="right">0.9994322</td>
<td align="right">2.967268</td>
</tr>
<tr class="even">
<td>GABPB1-IT1</td>
<td align="right">-0.3061776</td>
<td align="right">5.116526</td>
<td align="right">-3.565932</td>
<td align="right">0.0035049</td>
<td align="right">0.9994322</td>
<td align="right">-2.919594</td>
</tr>
</tbody>
</table>
<p>For cluster 7:</p>
<pre class="r"><code>pseudo7.full &lt;- generate.pseudobulk(cluster7.full, labels = c(&quot;stim&quot;, &quot;replicate&quot;, &quot;cell.line&quot;))
pseudo7.full &lt;- filter.pseudobulk(pseudo7.full)
d7.full &lt;- DGEList(pseudo7.full$counts)
d7.full &lt;- calcNormFactors(d7.full)
keep7.full &lt;- filterByExpr(d7.full$counts)
d7.full &lt;- d7.full[keep7.full,]

voomed7.full &lt;- voomWithDreamWeights(d7.full, formde, pseudo7.full$meta, plot=FALSE)</code></pre>
<pre><code>Memory usage to store result: &gt; 22 Mb 
Dividing work into 100 chunks...

Total: 80 s</code></pre>
<pre class="r"><code>fitmm7.full = dream(voomed7.full, formde, pseudo7.full$meta)</code></pre>
<pre><code>Dividing work into 100 chunks...

Total: 104 s</code></pre>
<pre class="r"><code>try(topTable(fitmm7.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.full[8,1] &lt;- try(topTable(fitmm7.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.full[8, 1] &lt;- try(topTable(fitmm7.full, coef = &quot;stimoxidation&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm7.full, coef=&#39;stimoxidation&#39;,  number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;Oxidation DE genes, cluster 7 of full dataset&quot;)</code></pre>
<table>
<caption>Oxidation DE genes, cluster 7 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>SEMA4C</td>
<td align="right">-0.2762940</td>
<td align="right">3.380888</td>
<td align="right">-7.069519</td>
<td align="right">0.0000080</td>
<td align="right">0.0776038</td>
<td align="right">-4.466010</td>
</tr>
<tr class="even">
<td>KDM4C</td>
<td align="right">-0.4852838</td>
<td align="right">6.047073</td>
<td align="right">-6.403564</td>
<td align="right">0.0000212</td>
<td align="right">0.1031466</td>
<td align="right">-4.252028</td>
</tr>
<tr class="odd">
<td>BCHE</td>
<td align="right">-1.3649188</td>
<td align="right">4.397220</td>
<td align="right">-5.330040</td>
<td align="right">0.0000827</td>
<td align="right">0.2149479</td>
<td align="right">-3.936459</td>
</tr>
<tr class="even">
<td>SPINDOC</td>
<td align="right">0.6174755</td>
<td align="right">3.517549</td>
<td align="right">5.113726</td>
<td align="right">0.0001209</td>
<td align="right">0.2149479</td>
<td align="right">3.844209</td>
</tr>
<tr class="odd">
<td>P4HA1</td>
<td align="right">-1.1097698</td>
<td align="right">6.110686</td>
<td align="right">-5.350676</td>
<td align="right">0.0001327</td>
<td align="right">0.2149479</td>
<td align="right">-3.821482</td>
</tr>
<tr class="even">
<td>GPN2</td>
<td align="right">0.5735638</td>
<td align="right">3.808511</td>
<td align="right">5.066217</td>
<td align="right">0.0001374</td>
<td align="right">0.2149479</td>
<td align="right">3.812850</td>
</tr>
<tr class="odd">
<td>BROX</td>
<td align="right">-0.3978903</td>
<td align="right">5.021557</td>
<td align="right">-5.192062</td>
<td align="right">0.0001720</td>
<td align="right">0.2149479</td>
<td align="right">-3.756943</td>
</tr>
<tr class="even">
<td>UBE2E3</td>
<td align="right">-0.3358330</td>
<td align="right">8.029897</td>
<td align="right">-5.193155</td>
<td align="right">0.0001766</td>
<td align="right">0.2149479</td>
<td align="right">-3.750354</td>
</tr>
<tr class="odd">
<td>SETD1B</td>
<td align="right">0.5352956</td>
<td align="right">3.916539</td>
<td align="right">4.695771</td>
<td align="right">0.0002081</td>
<td align="right">0.2251263</td>
<td align="right">3.709018</td>
</tr>
<tr class="even">
<td>CHD4</td>
<td align="right">0.2336914</td>
<td align="right">7.010355</td>
<td align="right">5.016731</td>
<td align="right">0.0002384</td>
<td align="right">0.2321063</td>
<td align="right">3.674462</td>
</tr>
</tbody>
</table>
<pre class="r"><code>try(topTable(fitmm7.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.full[8,2] &lt;- try(topTable(fitmm7.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.full[8, 2] &lt;- try(topTable(fitmm7.full, coef = &quot;stimpff&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm7.full, coef=&#39;stimpff&#39;, number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;PFF DE genes, cluster 7 of full dataset&quot;)</code></pre>
<table>
<caption>PFF DE genes, cluster 7 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PTK7</td>
<td align="right">-0.7857255</td>
<td align="right">4.346043</td>
<td align="right">-5.244802</td>
<td align="right">0.0001362</td>
<td align="right">0.7983159</td>
<td align="right">-3.814931</td>
</tr>
<tr class="even">
<td>LSM12</td>
<td align="right">-0.5332427</td>
<td align="right">5.728801</td>
<td align="right">-4.812242</td>
<td align="right">0.0001640</td>
<td align="right">0.7983159</td>
<td align="right">-3.768915</td>
</tr>
<tr class="odd">
<td>HIKESHI</td>
<td align="right">-0.4080186</td>
<td align="right">5.961141</td>
<td align="right">-4.420679</td>
<td align="right">0.0003742</td>
<td align="right">0.8580650</td>
<td align="right">-3.557634</td>
</tr>
<tr class="even">
<td>YLPM1</td>
<td align="right">-0.5194360</td>
<td align="right">6.900117</td>
<td align="right">-4.247523</td>
<td align="right">0.0005430</td>
<td align="right">0.8580650</td>
<td align="right">-3.458589</td>
</tr>
<tr class="odd">
<td>ST7L</td>
<td align="right">-0.4918986</td>
<td align="right">5.512936</td>
<td align="right">-4.273772</td>
<td align="right">0.0006082</td>
<td align="right">0.8580650</td>
<td align="right">-3.427929</td>
</tr>
<tr class="even">
<td>ZDHHC5</td>
<td align="right">-0.6608337</td>
<td align="right">4.422092</td>
<td align="right">-4.094331</td>
<td align="right">0.0007572</td>
<td align="right">0.8580650</td>
<td align="right">-3.367978</td>
</tr>
<tr class="odd">
<td>MORC2-AS1</td>
<td align="right">0.5133991</td>
<td align="right">4.215137</td>
<td align="right">4.171615</td>
<td align="right">0.0007606</td>
<td align="right">0.8580650</td>
<td align="right">3.366740</td>
</tr>
<tr class="even">
<td>DNAJC18</td>
<td align="right">-0.4017006</td>
<td align="right">5.717145</td>
<td align="right">-4.207709</td>
<td align="right">0.0008354</td>
<td align="right">0.8580650</td>
<td align="right">-3.340807</td>
</tr>
<tr class="odd">
<td>SHARPIN</td>
<td align="right">-0.3787221</td>
<td align="right">4.885176</td>
<td align="right">-4.313177</td>
<td align="right">0.0008715</td>
<td align="right">0.8580650</td>
<td align="right">-3.329019</td>
</tr>
<tr class="even">
<td>MTRNR2L12</td>
<td align="right">-0.6231421</td>
<td align="right">7.546277</td>
<td align="right">-4.105749</td>
<td align="right">0.0009211</td>
<td align="right">0.8580650</td>
<td align="right">-3.313581</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#now repeat with the neuron dataset
pseudo7.neuron &lt;- generate.pseudobulk(cluster7.neuron, labels = c(&quot;stim&quot;, &quot;replicate&quot;, &quot;cell.line&quot;))
pseudo7.neuron &lt;- filter.pseudobulk(pseudo7.neuron)
d7.neuron &lt;- DGEList(pseudo7.neuron$counts)
d7.neuron &lt;- calcNormFactors(d7.neuron)
keep7.neuron &lt;- filterByExpr(d7.neuron$counts)
d7.neuron &lt;- d7.neuron[keep7.neuron,]

voomed7.neuron &lt;- voomWithDreamWeights(d7.neuron, formde, pseudo7.neuron$meta, plot=FALSE)</code></pre>
<pre><code>Memory usage to store result: &gt; 20.1 Mb 
Dividing work into 100 chunks...

Total: 66 s</code></pre>
<pre class="r"><code>fitmm7.neuron = dream(voomed7.neuron, formde, pseudo7.neuron$meta)</code></pre>
<pre><code>Dividing work into 100 chunks...

Total: 100 s</code></pre>
<pre class="r"><code>try(topTable(fitmm7.neuron, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.neuron[8,1] &lt;- try(topTable(fitmm7.neuron, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.neuron[8, 1] &lt;- try(topTable(fitmm7.neuron, coef = &quot;stimoxidation&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm7.neuron, coef=&#39;stimoxidation&#39;,  number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;Oxidation DE genes, cluster 7 of full dataset&quot;)</code></pre>
<table>
<caption>Oxidation DE genes, cluster 7 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>CASTOR3</td>
<td align="right">0.3569498</td>
<td align="right">6.968379</td>
<td align="right">6.309544</td>
<td align="right">0.0000060</td>
<td align="right">0.0517629</td>
<td align="right">4.525821</td>
</tr>
<tr class="even">
<td>MRPL17</td>
<td align="right">-0.7846283</td>
<td align="right">4.835736</td>
<td align="right">-5.673256</td>
<td align="right">0.0000333</td>
<td align="right">0.0899409</td>
<td align="right">-4.149964</td>
</tr>
<tr class="odd">
<td>MT-ND2</td>
<td align="right">0.4983064</td>
<td align="right">10.235438</td>
<td align="right">5.994458</td>
<td align="right">0.0000356</td>
<td align="right">0.0899409</td>
<td align="right">4.134211</td>
</tr>
<tr class="even">
<td>STAU2</td>
<td align="right">0.3654272</td>
<td align="right">7.627762</td>
<td align="right">5.833084</td>
<td align="right">0.0000418</td>
<td align="right">0.0899409</td>
<td align="right">4.097223</td>
</tr>
<tr class="odd">
<td>MT-ND1</td>
<td align="right">0.5441213</td>
<td align="right">10.219673</td>
<td align="right">5.481575</td>
<td align="right">0.0000800</td>
<td align="right">0.1204966</td>
<td align="right">3.944524</td>
</tr>
<tr class="even">
<td>MT-ND3</td>
<td align="right">0.5545342</td>
<td align="right">10.973308</td>
<td align="right">5.454620</td>
<td align="right">0.0000840</td>
<td align="right">0.1204966</td>
<td align="right">3.932611</td>
</tr>
<tr class="odd">
<td>BNIP3</td>
<td align="right">-1.6817032</td>
<td align="right">8.337418</td>
<td align="right">-4.913700</td>
<td align="right">0.0001119</td>
<td align="right">0.1337440</td>
<td align="right">-3.863122</td>
</tr>
<tr class="even">
<td>IDI1</td>
<td align="right">0.5967459</td>
<td align="right">6.318019</td>
<td align="right">5.122484</td>
<td align="right">0.0001399</td>
<td align="right">0.1337440</td>
<td align="right">3.808366</td>
</tr>
<tr class="odd">
<td>TNIP1</td>
<td align="right">-1.0036789</td>
<td align="right">4.948658</td>
<td align="right">-4.962139</td>
<td align="right">0.0001399</td>
<td align="right">0.1337440</td>
<td align="right">-3.808346</td>
</tr>
<tr class="even">
<td>ZCCHC17</td>
<td align="right">0.2787882</td>
<td align="right">6.357254</td>
<td align="right">4.637591</td>
<td align="right">0.0002046</td>
<td align="right">0.1714341</td>
<td align="right">3.713232</td>
</tr>
</tbody>
</table>
<pre class="r"><code>try(topTable(fitmm7.neuron, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.neuron[8,2] &lt;- try(topTable(fitmm7.neuron, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.neuron[8, 2] &lt;- try(topTable(fitmm7.neuron, coef = &quot;stimpff&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm7.neuron, coef=&#39;stimpff&#39;, number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;PFF DE genes, cluster 7 of full dataset&quot;)</code></pre>
<table>
<caption>PFF DE genes, cluster 7 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PAPSS1</td>
<td align="right">-0.4319394</td>
<td align="right">6.318047</td>
<td align="right">-5.104977</td>
<td align="right">0.0001059</td>
<td align="right">0.9110047</td>
<td align="right">-3.876703</td>
</tr>
<tr class="even">
<td>CLTA</td>
<td align="right">-0.3203198</td>
<td align="right">7.946916</td>
<td align="right">-4.543230</td>
<td align="right">0.0002867</td>
<td align="right">0.9530321</td>
<td align="right">-3.627013</td>
</tr>
<tr class="odd">
<td>PRPF19</td>
<td align="right">-0.6794699</td>
<td align="right">4.469610</td>
<td align="right">-4.425788</td>
<td align="right">0.0004315</td>
<td align="right">0.9530321</td>
<td align="right">-3.520048</td>
</tr>
<tr class="even">
<td>DPH3</td>
<td align="right">-0.7999538</td>
<td align="right">4.709590</td>
<td align="right">-4.098552</td>
<td align="right">0.0006744</td>
<td align="right">0.9530321</td>
<td align="right">-3.399784</td>
</tr>
<tr class="odd">
<td>SF3B4</td>
<td align="right">-0.9352325</td>
<td align="right">4.929091</td>
<td align="right">-4.197428</td>
<td align="right">0.0006757</td>
<td align="right">0.9530321</td>
<td align="right">-3.399247</td>
</tr>
<tr class="even">
<td>CNOT3</td>
<td align="right">-0.6178248</td>
<td align="right">4.424536</td>
<td align="right">-4.096391</td>
<td align="right">0.0006974</td>
<td align="right">0.9530321</td>
<td align="right">-3.390601</td>
</tr>
<tr class="odd">
<td>METTL23</td>
<td align="right">-0.7600086</td>
<td align="right">4.422313</td>
<td align="right">-3.942273</td>
<td align="right">0.0009551</td>
<td align="right">0.9530321</td>
<td align="right">-3.303436</td>
</tr>
<tr class="even">
<td>WBP1L</td>
<td align="right">-0.7905323</td>
<td align="right">4.645184</td>
<td align="right">-4.027870</td>
<td align="right">0.0010310</td>
<td align="right">0.9530321</td>
<td align="right">-3.281923</td>
</tr>
<tr class="odd">
<td>MBD4</td>
<td align="right">-0.8338809</td>
<td align="right">4.671269</td>
<td align="right">-3.660893</td>
<td align="right">0.0020469</td>
<td align="right">0.9530321</td>
<td align="right">-3.083338</td>
</tr>
<tr class="even">
<td>ABHD5</td>
<td align="right">-1.0134761</td>
<td align="right">4.512543</td>
<td align="right">-3.566664</td>
<td align="right">0.0022086</td>
<td align="right">0.9530321</td>
<td align="right">-3.060650</td>
</tr>
</tbody>
</table>
<p>For cluster 8:</p>
<pre class="r"><code>pseudo8.full &lt;- generate.pseudobulk(cluster8.full, labels = c(&quot;stim&quot;, &quot;replicate&quot;, &quot;cell.line&quot;))
pseudo8.full &lt;- filter.pseudobulk(pseudo8.full)
d8.full &lt;- DGEList(pseudo8.full$counts)
d8.full &lt;- calcNormFactors(d8.full)
keep8.full &lt;- filterByExpr(d8.full$counts)
d8.full &lt;- d8.full[keep8.full,]

voomed8.full &lt;- voomWithDreamWeights(d8.full, formde, pseudo8.full$meta, plot=FALSE)</code></pre>
<pre><code>Memory usage to store result: &gt; 29.7 Mb 
Dividing work into 100 chunks...

Total: 84 s</code></pre>
<pre class="r"><code>fitmm8.full = dream(voomed8.full, formde, pseudo8.full$meta)</code></pre>
<pre><code>Dividing work into 100 chunks...

Total: 113 s</code></pre>
<pre class="r"><code>try(topTable(fitmm8.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>[1] 634</code></pre>
<pre class="r"><code>de.summary.full[9,1] &lt;- try(topTable(fitmm8.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()
topTable(fitmm8.full, coef=&#39;stimoxidation&#39;,  number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;Oxidation DE genes, cluster 8 of full dataset&quot;)</code></pre>
<table>
<caption>Oxidation DE genes, cluster 8 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>OXLD1</td>
<td align="right">0.6439721</td>
<td align="right">5.031777</td>
<td align="right">9.849353</td>
<td align="right">1.0e-07</td>
<td align="right">0.0008682</td>
<td align="right">5.324396</td>
</tr>
<tr class="even">
<td>LINC01937</td>
<td align="right">1.9758874</td>
<td align="right">3.579539</td>
<td align="right">9.354587</td>
<td align="right">2.0e-07</td>
<td align="right">0.0008682</td>
<td align="right">5.202965</td>
</tr>
<tr class="odd">
<td>BRWD1</td>
<td align="right">-0.2990852</td>
<td align="right">6.564589</td>
<td align="right">-8.100999</td>
<td align="right">2.0e-07</td>
<td align="right">0.0008682</td>
<td align="right">-5.194856</td>
</tr>
<tr class="even">
<td>XPO1</td>
<td align="right">-0.4355494</td>
<td align="right">6.713519</td>
<td align="right">-8.742466</td>
<td align="right">6.0e-07</td>
<td align="right">0.0017303</td>
<td align="right">-4.989280</td>
</tr>
<tr class="odd">
<td>RLF</td>
<td align="right">-0.7985341</td>
<td align="right">6.252294</td>
<td align="right">-7.438023</td>
<td align="right">7.0e-07</td>
<td align="right">0.0017303</td>
<td align="right">-4.966841</td>
</tr>
<tr class="even">
<td>PIAS2</td>
<td align="right">-0.5173223</td>
<td align="right">6.316515</td>
<td align="right">-8.255961</td>
<td align="right">9.0e-07</td>
<td align="right">0.0018213</td>
<td align="right">-4.919640</td>
</tr>
<tr class="odd">
<td>AC010210.1</td>
<td align="right">1.3746342</td>
<td align="right">3.307248</td>
<td align="right">8.024075</td>
<td align="right">1.3e-06</td>
<td align="right">0.0018213</td>
<td align="right">4.838157</td>
</tr>
<tr class="even">
<td>PPP1R10</td>
<td align="right">0.6175800</td>
<td align="right">5.691891</td>
<td align="right">7.434229</td>
<td align="right">1.4e-06</td>
<td align="right">0.0018213</td>
<td align="right">4.828662</td>
</tr>
<tr class="odd">
<td>GLG1</td>
<td align="right">-0.2741240</td>
<td align="right">7.014482</td>
<td align="right">-7.927377</td>
<td align="right">1.5e-06</td>
<td align="right">0.0018213</td>
<td align="right">-4.810784</td>
</tr>
<tr class="even">
<td>KAT6B</td>
<td align="right">-0.5419431</td>
<td align="right">6.845189</td>
<td align="right">-7.377024</td>
<td align="right">1.5e-06</td>
<td align="right">0.0018213</td>
<td align="right">-4.807146</td>
</tr>
</tbody>
</table>
<pre class="r"><code>try(topTable(fitmm8.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.full[9,2] &lt;- try(topTable(fitmm8.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.full[9, 2] &lt;- try(topTable(fitmm8.full, coef = &quot;stimpff&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm8.full, coef=&#39;stimpff&#39;, number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;PFF DE genes, cluster 8 of full dataset&quot;)</code></pre>
<table>
<caption>PFF DE genes, cluster 8 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>BMPR1A</td>
<td align="right">0.3850365</td>
<td align="right">6.501713</td>
<td align="right">5.885678</td>
<td align="right">0.0000350</td>
<td align="right">0.4449275</td>
<td align="right">4.138241</td>
</tr>
<tr class="even">
<td>BICRAL</td>
<td align="right">0.4473285</td>
<td align="right">5.240534</td>
<td align="right">4.710643</td>
<td align="right">0.0002379</td>
<td align="right">0.8207915</td>
<td align="right">3.674939</td>
</tr>
<tr class="odd">
<td>PRR5</td>
<td align="right">0.9980141</td>
<td align="right">4.209295</td>
<td align="right">4.582491</td>
<td align="right">0.0003336</td>
<td align="right">0.8207915</td>
<td align="right">3.587732</td>
</tr>
<tr class="even">
<td>HAUS2</td>
<td align="right">-0.3091288</td>
<td align="right">4.721788</td>
<td align="right">-4.531232</td>
<td align="right">0.0003391</td>
<td align="right">0.8207915</td>
<td align="right">-3.583419</td>
</tr>
<tr class="odd">
<td>CXorf40B</td>
<td align="right">-0.8463120</td>
<td align="right">2.636080</td>
<td align="right">-4.410177</td>
<td align="right">0.0004148</td>
<td align="right">0.8207915</td>
<td align="right">-3.530482</td>
</tr>
<tr class="even">
<td>SLC41A3</td>
<td align="right">0.3948520</td>
<td align="right">4.572912</td>
<td align="right">4.421568</td>
<td align="right">0.0004287</td>
<td align="right">0.8207915</td>
<td align="right">3.521781</td>
</tr>
<tr class="odd">
<td>AC009107.2</td>
<td align="right">-0.5017378</td>
<td align="right">4.784756</td>
<td align="right">-4.278706</td>
<td align="right">0.0004519</td>
<td align="right">0.8207915</td>
<td align="right">-3.507733</td>
</tr>
<tr class="even">
<td>CNNM2</td>
<td align="right">0.4911934</td>
<td align="right">4.882871</td>
<td align="right">4.249886</td>
<td align="right">0.0006204</td>
<td align="right">0.9193592</td>
<td align="right">3.422519</td>
</tr>
<tr class="odd">
<td>CP</td>
<td align="right">-1.5887899</td>
<td align="right">5.526436</td>
<td align="right">-4.239453</td>
<td align="right">0.0006508</td>
<td align="right">0.9193592</td>
<td align="right">-3.409489</td>
</tr>
<tr class="even">
<td>ZNF16</td>
<td align="right">0.5024517</td>
<td align="right">3.195021</td>
<td align="right">4.046279</td>
<td align="right">0.0007576</td>
<td align="right">0.9521909</td>
<td align="right">3.367841</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#now repeat with the neuron dataset
pseudo8.neuron &lt;- generate.pseudobulk(cluster8.neuron, labels = c(&quot;stim&quot;, &quot;replicate&quot;, &quot;cell.line&quot;))
pseudo8.neuron &lt;- filter.pseudobulk(pseudo8.neuron)
d8.neuron &lt;- DGEList(pseudo8.neuron$counts)
d8.neuron &lt;- calcNormFactors(d8.neuron)
keep8.neuron &lt;- filterByExpr(d8.neuron$counts)
d8.neuron &lt;- d8.neuron[keep8.neuron,]

voomed8.neuron &lt;- voomWithDreamWeights(d8.neuron, formde, pseudo8.neuron$meta, plot=FALSE)</code></pre>
<pre><code>Memory usage to store result: &gt; 24 Mb 
Dividing work into 100 chunks...

Total: 82 s</code></pre>
<pre class="r"><code>fitmm8.neuron = dream(voomed8.neuron, formde, pseudo8.neuron$meta)</code></pre>
<pre><code>Dividing work into 100 chunks...

Total: 106 s</code></pre>
<pre class="r"><code>try(topTable(fitmm8.neuron, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>[1] 2</code></pre>
<pre class="r"><code>de.summary.neuron[9,1] &lt;- try(topTable(fitmm8.neuron, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()
topTable(fitmm8.neuron, coef=&#39;stimoxidation&#39;,  number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;Oxidation DE genes, cluster 8 of full dataset&quot;)</code></pre>
<table>
<caption>Oxidation DE genes, cluster 8 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>NARF</td>
<td align="right">-0.4028042</td>
<td align="right">6.414485</td>
<td align="right">-7.661677</td>
<td align="right">0.0000022</td>
<td align="right">0.0228946</td>
<td align="right">-4.731978</td>
</tr>
<tr class="even">
<td>TET1</td>
<td align="right">-0.5648924</td>
<td align="right">7.660503</td>
<td align="right">-7.164071</td>
<td align="right">0.0000048</td>
<td align="right">0.0246268</td>
<td align="right">-4.574074</td>
</tr>
<tr class="odd">
<td>DDIT4</td>
<td align="right">-1.9342164</td>
<td align="right">5.500840</td>
<td align="right">-5.770934</td>
<td align="right">0.0000283</td>
<td align="right">0.0971284</td>
<td align="right">-4.186747</td>
</tr>
<tr class="even">
<td>MTFP1</td>
<td align="right">-1.4845529</td>
<td align="right">3.912765</td>
<td align="right">-5.448088</td>
<td align="right">0.0000517</td>
<td align="right">0.1146043</td>
<td align="right">-4.047986</td>
</tr>
<tr class="odd">
<td>HK2</td>
<td align="right">-1.4905555</td>
<td align="right">4.294197</td>
<td align="right">-5.636234</td>
<td align="right">0.0000618</td>
<td align="right">0.1146043</td>
<td align="right">-4.005710</td>
</tr>
<tr class="even">
<td>PDK3</td>
<td align="right">-0.8889310</td>
<td align="right">5.353684</td>
<td align="right">-5.152786</td>
<td align="right">0.0000668</td>
<td align="right">0.1146043</td>
<td align="right">-3.987479</td>
</tr>
<tr class="odd">
<td>CXCR4</td>
<td align="right">-1.1183089</td>
<td align="right">4.931837</td>
<td align="right">-4.936261</td>
<td align="right">0.0001442</td>
<td align="right">0.2120903</td>
<td align="right">-3.800884</td>
</tr>
<tr class="even">
<td>P4HA1</td>
<td align="right">-0.9900035</td>
<td align="right">7.137119</td>
<td align="right">-4.739535</td>
<td align="right">0.0002275</td>
<td align="right">0.2349946</td>
<td align="right">-3.686397</td>
</tr>
<tr class="odd">
<td>PDK1</td>
<td align="right">-1.5435413</td>
<td align="right">5.410659</td>
<td align="right">-4.734424</td>
<td align="right">0.0002317</td>
<td align="right">0.2349946</td>
<td align="right">-3.681630</td>
</tr>
<tr class="even">
<td>STAG2</td>
<td align="right">-0.1847200</td>
<td align="right">7.424414</td>
<td align="right">-4.723325</td>
<td align="right">0.0003155</td>
<td align="right">0.2349946</td>
<td align="right">-3.602250</td>
</tr>
</tbody>
</table>
<pre class="r"><code>try(topTable(fitmm8.neuron, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.neuron[9,2] &lt;- try(topTable(fitmm8.neuron, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.neuron[9, 2] &lt;- try(topTable(fitmm8.neuron, coef = &quot;stimpff&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm8.neuron, coef=&#39;stimpff&#39;, number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;PFF DE genes, cluster 8 of full dataset&quot;)</code></pre>
<table>
<caption>PFF DE genes, cluster 8 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>SPAG9</td>
<td align="right">-0.3226207</td>
<td align="right">8.085576</td>
<td align="right">-4.275762</td>
<td align="right">0.0005628</td>
<td align="right">0.9996081</td>
<td align="right">-3.448921</td>
</tr>
<tr class="even">
<td>ISCA2</td>
<td align="right">-0.3850110</td>
<td align="right">5.640302</td>
<td align="right">-4.378765</td>
<td align="right">0.0005726</td>
<td align="right">0.9996081</td>
<td align="right">-3.444270</td>
</tr>
<tr class="odd">
<td>ATP6V1H</td>
<td align="right">0.4453390</td>
<td align="right">5.474011</td>
<td align="right">4.180726</td>
<td align="right">0.0008299</td>
<td align="right">0.9996081</td>
<td align="right">3.342639</td>
</tr>
<tr class="even">
<td>FAXDC2</td>
<td align="right">0.9885577</td>
<td align="right">4.194718</td>
<td align="right">3.886058</td>
<td align="right">0.0011561</td>
<td align="right">0.9996081</td>
<td align="right">3.249496</td>
</tr>
<tr class="odd">
<td>RIN2</td>
<td align="right">-1.6385612</td>
<td align="right">5.038728</td>
<td align="right">-3.816086</td>
<td align="right">0.0015255</td>
<td align="right">0.9996081</td>
<td align="right">-3.169794</td>
</tr>
<tr class="even">
<td>ZNF625</td>
<td align="right">0.4482622</td>
<td align="right">5.349484</td>
<td align="right">3.792770</td>
<td align="right">0.0016516</td>
<td align="right">0.9996081</td>
<td align="right">3.146636</td>
</tr>
<tr class="odd">
<td>MFSD1</td>
<td align="right">-1.0187511</td>
<td align="right">3.470310</td>
<td align="right">-3.652369</td>
<td align="right">0.0021404</td>
<td align="right">0.9996081</td>
<td align="right">-3.070023</td>
</tr>
<tr class="even">
<td>SRSF5</td>
<td align="right">-0.2712445</td>
<td align="right">7.708079</td>
<td align="right">-3.588935</td>
<td align="right">0.0023609</td>
<td align="right">0.9996081</td>
<td align="right">-3.040627</td>
</tr>
<tr class="odd">
<td>GEN1</td>
<td align="right">-0.4066377</td>
<td align="right">6.735619</td>
<td align="right">-3.562299</td>
<td align="right">0.0028569</td>
<td align="right">0.9996081</td>
<td align="right">-2.982731</td>
</tr>
<tr class="even">
<td>RAB11A</td>
<td align="right">0.3068056</td>
<td align="right">6.600957</td>
<td align="right">3.461825</td>
<td align="right">0.0032476</td>
<td align="right">0.9996081</td>
<td align="right">2.943275</td>
</tr>
</tbody>
</table>
<p>For cluster 9:</p>
<pre class="r"><code>pseudo9.full &lt;- generate.pseudobulk(cluster9.full, labels = c(&quot;stim&quot;, &quot;replicate&quot;, &quot;cell.line&quot;))
pseudo9.full &lt;- filter.pseudobulk(pseudo9.full)
d9.full &lt;- DGEList(pseudo9.full$counts)
d9.full &lt;- calcNormFactors(d9.full)
keep9.full &lt;- filterByExpr(d9.full$counts)
d9.full &lt;- d9.full[keep9.full,]

voomed9.full &lt;- voomWithDreamWeights(d9.full, formde, pseudo9.full$meta, plot=FALSE)</code></pre>
<pre><code>Memory usage to store result: &gt; 25.9 Mb 
Dividing work into 100 chunks...

Total: 82 s</code></pre>
<pre class="r"><code>fitmm9.full = dream(voomed9.full, formde, pseudo9.full$meta)</code></pre>
<pre><code>Dividing work into 100 chunks...

Total: 111 s</code></pre>
<pre class="r"><code>try(topTable(fitmm9.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>[1] 92</code></pre>
<pre class="r"><code>de.summary.full[10,1] &lt;- try(topTable(fitmm9.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()
topTable(fitmm9.full, coef=&#39;stimoxidation&#39;,  number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;Oxidation DE genes, cluster 9 of full dataset&quot;)</code></pre>
<table>
<caption>Oxidation DE genes, cluster 9 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>OGG1</td>
<td align="right">0.3944510</td>
<td align="right">5.582161</td>
<td align="right">8.691846</td>
<td align="right">6.00e-07</td>
<td align="right">0.0064052</td>
<td align="right">4.995868</td>
</tr>
<tr class="even">
<td>SLC2A1</td>
<td align="right">-1.1811466</td>
<td align="right">3.249990</td>
<td align="right">-8.238553</td>
<td align="right">1.20e-06</td>
<td align="right">0.0064052</td>
<td align="right">-4.852242</td>
</tr>
<tr class="odd">
<td>AP3B2</td>
<td align="right">0.3515158</td>
<td align="right">6.009994</td>
<td align="right">7.418748</td>
<td align="right">1.70e-06</td>
<td align="right">0.0064052</td>
<td align="right">4.789418</td>
</tr>
<tr class="even">
<td>RLF</td>
<td align="right">-0.8476715</td>
<td align="right">6.913787</td>
<td align="right">-6.401983</td>
<td align="right">6.60e-06</td>
<td align="right">0.0141712</td>
<td align="right">-4.507329</td>
</tr>
<tr class="odd">
<td>STIM1</td>
<td align="right">0.7944202</td>
<td align="right">4.423345</td>
<td align="right">7.074414</td>
<td align="right">7.40e-06</td>
<td align="right">0.0141712</td>
<td align="right">4.480432</td>
</tr>
<tr class="even">
<td>SLU7</td>
<td align="right">-0.5196390</td>
<td align="right">5.426772</td>
<td align="right">-6.924047</td>
<td align="right">9.20e-06</td>
<td align="right">0.0141712</td>
<td align="right">-4.434328</td>
</tr>
<tr class="odd">
<td>KDM3A</td>
<td align="right">-1.6152357</td>
<td align="right">5.551316</td>
<td align="right">-6.608019</td>
<td align="right">1.17e-05</td>
<td align="right">0.0141712</td>
<td align="right">-4.382486</td>
</tr>
<tr class="even">
<td>DDIT4</td>
<td align="right">-2.6554540</td>
<td align="right">4.758925</td>
<td align="right">-6.280554</td>
<td align="right">1.30e-05</td>
<td align="right">0.0141712</td>
<td align="right">-4.360287</td>
</tr>
<tr class="odd">
<td>MIR210HG</td>
<td align="right">-1.6679630</td>
<td align="right">3.475411</td>
<td align="right">-6.251492</td>
<td align="right">1.59e-05</td>
<td align="right">0.0141712</td>
<td align="right">-4.316367</td>
</tr>
<tr class="even">
<td>WNK3</td>
<td align="right">-0.3316223</td>
<td align="right">6.623639</td>
<td align="right">-6.215498</td>
<td align="right">1.63e-05</td>
<td align="right">0.0141712</td>
<td align="right">-4.310234</td>
</tr>
</tbody>
</table>
<pre class="r"><code>try(topTable(fitmm9.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.full[10,2] &lt;- try(topTable(fitmm9.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.full[10, 2] &lt;- try(topTable(fitmm9.full, coef = &quot;stimpff&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm9.full, coef=&#39;stimpff&#39;, number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;PFF DE genes, cluster 9 of full dataset&quot;)</code></pre>
<table>
<caption>PFF DE genes, cluster 9 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>AP3B2</td>
<td align="right">0.3381361</td>
<td align="right">6.009994</td>
<td align="right">4.933228</td>
<td align="right">0.0001679</td>
<td align="right">0.9987553</td>
<td align="right">3.763031</td>
</tr>
<tr class="even">
<td>STIM1</td>
<td align="right">0.6568728</td>
<td align="right">4.423345</td>
<td align="right">4.220234</td>
<td align="right">0.0009717</td>
<td align="right">0.9987553</td>
<td align="right">3.298585</td>
</tr>
<tr class="odd">
<td>SPSB3</td>
<td align="right">-0.4717105</td>
<td align="right">5.066873</td>
<td align="right">-4.021591</td>
<td align="right">0.0009731</td>
<td align="right">0.9987553</td>
<td align="right">-3.298178</td>
</tr>
<tr class="even">
<td>AL049840.2</td>
<td align="right">0.7536917</td>
<td align="right">2.919947</td>
<td align="right">3.902535</td>
<td align="right">0.0011454</td>
<td align="right">0.9987553</td>
<td align="right">3.252144</td>
</tr>
<tr class="odd">
<td>LDOC1</td>
<td align="right">-0.4757087</td>
<td align="right">5.040858</td>
<td align="right">-3.810440</td>
<td align="right">0.0020356</td>
<td align="right">0.9987553</td>
<td align="right">-3.084982</td>
</tr>
<tr class="even">
<td>DRG1</td>
<td align="right">-0.2476880</td>
<td align="right">5.529154</td>
<td align="right">-3.627163</td>
<td align="right">0.0023576</td>
<td align="right">0.9987553</td>
<td align="right">-3.041038</td>
</tr>
<tr class="odd">
<td>LINC02789</td>
<td align="right">0.9999326</td>
<td align="right">3.437371</td>
<td align="right">3.560923</td>
<td align="right">0.0024155</td>
<td align="right">0.9987553</td>
<td align="right">3.033733</td>
</tr>
<tr class="even">
<td>CHKB-DT</td>
<td align="right">0.6198912</td>
<td align="right">4.234429</td>
<td align="right">3.651176</td>
<td align="right">0.0025798</td>
<td align="right">0.9987553</td>
<td align="right">3.013818</td>
</tr>
<tr class="odd">
<td>SENP5</td>
<td align="right">-0.2499175</td>
<td align="right">6.942381</td>
<td align="right">-3.603056</td>
<td align="right">0.0030376</td>
<td align="right">0.9987553</td>
<td align="right">-2.963903</td>
</tr>
<tr class="even">
<td>UBA2</td>
<td align="right">-0.2745685</td>
<td align="right">6.798440</td>
<td align="right">-3.411688</td>
<td align="right">0.0033226</td>
<td align="right">0.9987553</td>
<td align="right">-2.936196</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#now repeat with the neuron dataset
pseudo9.neuron &lt;- generate.pseudobulk(cluster9.neuron, labels = c(&quot;stim&quot;, &quot;replicate&quot;, &quot;cell.line&quot;))
pseudo9.neuron &lt;- filter.pseudobulk(pseudo9.neuron)
d9.neuron &lt;- DGEList(pseudo9.neuron$counts)
d9.neuron &lt;- calcNormFactors(d9.neuron)
keep9.neuron &lt;- filterByExpr(d9.neuron$counts)
d9.neuron &lt;- d9.neuron[keep9.neuron,]

voomed9.neuron &lt;- voomWithDreamWeights(d9.neuron, formde, pseudo9.neuron$meta, plot=FALSE)</code></pre>
<pre><code>Memory usage to store result: &gt; 1.9 Mb 
Dividing work into 100 chunks...

Total: 39 s</code></pre>
<pre class="r"><code>fitmm9.neuron = dream(voomed9.neuron, formde, pseudo9.neuron$meta)</code></pre>
<pre><code>Dividing work into 100 chunks...

Total: 41 s</code></pre>
<pre class="r"><code>try(topTable(fitmm9.neuron, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.neuron[10,1] &lt;- try(topTable(fitmm9.neuron, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.neuron[10, 1] &lt;- try(topTable(fitmm9.neuron, coef = &quot;stimoxidation&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm9.neuron, coef=&#39;stimoxidation&#39;,  number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;Oxidation DE genes, cluster 9 of full dataset&quot;)</code></pre>
<table>
<caption>Oxidation DE genes, cluster 9 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>NDUFA5</td>
<td align="right">-0.2624297</td>
<td align="right">7.702222</td>
<td align="right">-4.125184</td>
<td align="right">0.0024281</td>
<td align="right">0.5583572</td>
<td align="right">-3.032161</td>
</tr>
<tr class="even">
<td>SSH2</td>
<td align="right">0.8491772</td>
<td align="right">8.935699</td>
<td align="right">3.447705</td>
<td align="right">0.0043271</td>
<td align="right">0.5583572</td>
<td align="right">2.853273</td>
</tr>
<tr class="odd">
<td>EIF4A2</td>
<td align="right">-0.5985248</td>
<td align="right">8.345432</td>
<td align="right">-3.200702</td>
<td align="right">0.0077919</td>
<td align="right">0.5583572</td>
<td align="right">-2.660957</td>
</tr>
<tr class="even">
<td>HECW1</td>
<td align="right">0.8102224</td>
<td align="right">7.920828</td>
<td align="right">3.195681</td>
<td align="right">0.0084432</td>
<td align="right">0.5583572</td>
<td align="right">2.633814</td>
</tr>
<tr class="odd">
<td>ANK3</td>
<td align="right">-0.8927624</td>
<td align="right">8.074280</td>
<td align="right">-3.279693</td>
<td align="right">0.0091609</td>
<td align="right">0.5583572</td>
<td align="right">-2.605992</td>
</tr>
<tr class="even">
<td>PRMT1</td>
<td align="right">0.5417838</td>
<td align="right">7.086969</td>
<td align="right">3.102500</td>
<td align="right">0.0099073</td>
<td align="right">0.5583572</td>
<td align="right">2.579047</td>
</tr>
<tr class="odd">
<td>PHACTR4</td>
<td align="right">0.8152513</td>
<td align="right">8.109921</td>
<td align="right">2.924232</td>
<td align="right">0.0118440</td>
<td align="right">0.5583572</td>
<td align="right">2.516759</td>
</tr>
<tr class="even">
<td>ZSWIM6</td>
<td align="right">0.7887880</td>
<td align="right">8.544273</td>
<td align="right">3.030665</td>
<td align="right">0.0123245</td>
<td align="right">0.5583572</td>
<td align="right">2.502715</td>
</tr>
<tr class="odd">
<td>SOD1</td>
<td align="right">-0.4935112</td>
<td align="right">7.408366</td>
<td align="right">-2.961232</td>
<td align="right">0.0123322</td>
<td align="right">0.5583572</td>
<td align="right">-2.502494</td>
</tr>
<tr class="even">
<td>TIAM1</td>
<td align="right">0.6223398</td>
<td align="right">7.449668</td>
<td align="right">2.964357</td>
<td align="right">0.0127642</td>
<td align="right">0.5583572</td>
<td align="right">2.490282</td>
</tr>
</tbody>
</table>
<pre class="r"><code>try(topTable(fitmm9.neuron, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code>de.summary.neuron[10,2] &lt;- try(topTable(fitmm9.neuron, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()
topTable(fitmm9.neuron, coef=&#39;stimpff&#39;, number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;PFF DE genes, cluster 9 of full dataset&quot;)</code></pre>
<table>
<caption>PFF DE genes, cluster 9 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PPIG</td>
<td align="right">0.7261857</td>
<td align="right">7.472972</td>
<td align="right">6.326409</td>
<td align="right">0.0000473</td>
<td align="right">0.0461188</td>
<td align="right">4.068337</td>
</tr>
<tr class="even">
<td>UBXN4</td>
<td align="right">0.9065993</td>
<td align="right">7.300632</td>
<td align="right">5.372696</td>
<td align="right">0.0002456</td>
<td align="right">0.0852378</td>
<td align="right">3.666762</td>
</tr>
<tr class="odd">
<td>SPECC1L</td>
<td align="right">-0.6577117</td>
<td align="right">7.490254</td>
<td align="right">-4.987812</td>
<td align="right">0.0002625</td>
<td align="right">0.0852378</td>
<td align="right">-3.649707</td>
</tr>
<tr class="even">
<td>APC</td>
<td align="right">1.1534285</td>
<td align="right">8.578407</td>
<td align="right">4.097781</td>
<td align="right">0.0023945</td>
<td align="right">0.5830613</td>
<td align="right">3.036364</td>
</tr>
<tr class="odd">
<td>NDUFA5</td>
<td align="right">0.2950840</td>
<td align="right">7.702222</td>
<td align="right">3.441897</td>
<td align="right">0.0068093</td>
<td align="right">0.6568372</td>
<td align="right">2.706028</td>
</tr>
<tr class="even">
<td>NCOA1</td>
<td align="right">1.0979404</td>
<td align="right">8.481534</td>
<td align="right">3.297487</td>
<td align="right">0.0078145</td>
<td align="right">0.6568372</td>
<td align="right">2.659981</td>
</tr>
<tr class="odd">
<td>CLTC</td>
<td align="right">0.5264435</td>
<td align="right">7.094060</td>
<td align="right">3.247366</td>
<td align="right">0.0080893</td>
<td align="right">0.6568372</td>
<td align="right">2.648319</td>
</tr>
<tr class="even">
<td>PRMT1</td>
<td align="right">0.6602890</td>
<td align="right">7.086969</td>
<td align="right">3.149154</td>
<td align="right">0.0092015</td>
<td align="right">0.6568372</td>
<td align="right">2.604477</td>
</tr>
<tr class="odd">
<td>VCAN</td>
<td align="right">-1.2269902</td>
<td align="right">7.941789</td>
<td align="right">-3.056293</td>
<td align="right">0.0110097</td>
<td align="right">0.6568372</td>
<td align="right">-2.542391</td>
</tr>
<tr class="even">
<td>CAPRIN1</td>
<td align="right">1.0474862</td>
<td align="right">7.566935</td>
<td align="right">2.960722</td>
<td align="right">0.0114944</td>
<td align="right">0.6568372</td>
<td align="right">2.527296</td>
</tr>
</tbody>
</table>
<p>For cluster 10:</p>
<pre class="r"><code>pseudo10.full &lt;- generate.pseudobulk(cluster10.full, labels = c(&quot;stim&quot;, &quot;replicate&quot;, &quot;cell.line&quot;))
pseudo10.full &lt;- filter.pseudobulk(pseudo10.full)
d10.full &lt;- DGEList(pseudo10.full$counts)
d10.full &lt;- calcNormFactors(d10.full)
keep10.full &lt;- filterByExpr(d10.full$counts)
d10.full &lt;- d10.full[keep10.full,]

voomed10.full &lt;- voomWithDreamWeights(d10.full, formde, pseudo10.full$meta, plot=FALSE)</code></pre>
<pre><code>Memory usage to store result: &gt; 19.8 Mb 
Dividing work into 100 chunks...

Total: 64 s</code></pre>
<pre class="r"><code>fitmm10.full = dream(voomed10.full, formde, pseudo10.full$meta)</code></pre>
<pre><code>Dividing work into 100 chunks...

Total: 100 s</code></pre>
<pre class="r"><code>try(topTable(fitmm10.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>[1] 2</code></pre>
<pre class="r"><code>de.summary.full[11,1] &lt;- try(topTable(fitmm10.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()
topTable(fitmm10.full, coef=&#39;stimoxidation&#39;,  number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;Oxidation DE genes, cluster 10 of full dataset&quot;)</code></pre>
<table>
<caption>Oxidation DE genes, cluster 10 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>MT-ND3</td>
<td align="right">0.6262128</td>
<td align="right">10.953900</td>
<td align="right">6.992900</td>
<td align="right">0.0000062</td>
<td align="right">0.0463826</td>
<td align="right">4.519137</td>
</tr>
<tr class="even">
<td>MRPL17</td>
<td align="right">-0.7716178</td>
<td align="right">4.814286</td>
<td align="right">-6.265423</td>
<td align="right">0.0000110</td>
<td align="right">0.0463826</td>
<td align="right">-4.397440</td>
</tr>
<tr class="odd">
<td>STAU2</td>
<td align="right">0.3696664</td>
<td align="right">7.631731</td>
<td align="right">6.072519</td>
<td align="right">0.0000276</td>
<td align="right">0.0778367</td>
<td align="right">4.192643</td>
</tr>
<tr class="even">
<td>CASTOR3</td>
<td align="right">0.4087530</td>
<td align="right">6.934596</td>
<td align="right">5.231275</td>
<td align="right">0.0000564</td>
<td align="right">0.0917692</td>
<td align="right">4.027248</td>
</tr>
<tr class="odd">
<td>BNIP3</td>
<td align="right">-1.7795777</td>
<td align="right">8.420559</td>
<td align="right">-5.417273</td>
<td align="right">0.0000624</td>
<td align="right">0.0917692</td>
<td align="right">-4.003708</td>
</tr>
<tr class="even">
<td>IDI1</td>
<td align="right">0.7205479</td>
<td align="right">6.261624</td>
<td align="right">5.531014</td>
<td align="right">0.0000660</td>
<td align="right">0.0917692</td>
<td align="right">3.990212</td>
</tr>
<tr class="odd">
<td>TNIP1</td>
<td align="right">-1.1186340</td>
<td align="right">4.996028</td>
<td align="right">-5.232976</td>
<td align="right">0.0000816</td>
<td align="right">0.0917692</td>
<td align="right">-3.939559</td>
</tr>
<tr class="even">
<td>PDK1</td>
<td align="right">-1.9225913</td>
<td align="right">5.442998</td>
<td align="right">-5.212536</td>
<td align="right">0.0000867</td>
<td align="right">0.0917692</td>
<td align="right">-3.925118</td>
</tr>
<tr class="odd">
<td>DDIT4</td>
<td align="right">-2.3745030</td>
<td align="right">5.407379</td>
<td align="right">-5.006397</td>
<td align="right">0.0001294</td>
<td align="right">0.1073228</td>
<td align="right">-3.827658</td>
</tr>
<tr class="even">
<td>MT-ND1</td>
<td align="right">0.6087295</td>
<td align="right">10.180528</td>
<td align="right">5.201969</td>
<td align="right">0.0001347</td>
<td align="right">0.1073228</td>
<td align="right">3.817646</td>
</tr>
</tbody>
</table>
<pre class="r"><code>try(topTable(fitmm10.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.full[11,2] &lt;- try(topTable(fitmm10.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.full[11, 2] &lt;- try(topTable(fitmm10.full, coef = &quot;stimpff&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm10.full, coef=&#39;stimpff&#39;, number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;PFF DE genes, cluster 10 of full dataset&quot;)</code></pre>
<table>
<caption>PFF DE genes, cluster 10 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>CLTA</td>
<td align="right">-0.2831678</td>
<td align="right">7.923346</td>
<td align="right">-4.176746</td>
<td align="right">0.0008537</td>
<td align="right">0.9215077</td>
<td align="right">-3.334768</td>
</tr>
<tr class="even">
<td>NAA30</td>
<td align="right">-0.6726816</td>
<td align="right">4.259897</td>
<td align="right">-4.074448</td>
<td align="right">0.0008862</td>
<td align="right">0.9215077</td>
<td align="right">-3.324379</td>
</tr>
<tr class="odd">
<td>SF3B4</td>
<td align="right">-0.9036566</td>
<td align="right">4.930739</td>
<td align="right">-3.935821</td>
<td align="right">0.0012055</td>
<td align="right">0.9215077</td>
<td align="right">-3.237572</td>
</tr>
<tr class="even">
<td>POLI</td>
<td align="right">-0.9576693</td>
<td align="right">4.675478</td>
<td align="right">-3.799001</td>
<td align="right">0.0013143</td>
<td align="right">0.9215077</td>
<td align="right">-3.212836</td>
</tr>
<tr class="odd">
<td>TBC1D10B</td>
<td align="right">-0.7409350</td>
<td align="right">4.113693</td>
<td align="right">-3.880578</td>
<td align="right">0.0013256</td>
<td align="right">0.9215077</td>
<td align="right">-3.210383</td>
</tr>
<tr class="even">
<td>TNFAIP1</td>
<td align="right">-0.9284795</td>
<td align="right">4.214943</td>
<td align="right">-3.782707</td>
<td align="right">0.0015219</td>
<td align="right">0.9215077</td>
<td align="right">-3.170469</td>
</tr>
<tr class="odd">
<td>ARF6</td>
<td align="right">-0.8997721</td>
<td align="right">5.084519</td>
<td align="right">-3.702449</td>
<td align="right">0.0016482</td>
<td align="right">0.9215077</td>
<td align="right">-3.147244</td>
</tr>
<tr class="even">
<td>WBP1L</td>
<td align="right">-0.7951241</td>
<td align="right">4.698240</td>
<td align="right">-3.691902</td>
<td align="right">0.0016771</td>
<td align="right">0.9215077</td>
<td align="right">-3.142145</td>
</tr>
<tr class="odd">
<td>PRPF19</td>
<td align="right">-0.7217409</td>
<td align="right">4.480509</td>
<td align="right">-3.576891</td>
<td align="right">0.0021552</td>
<td align="right">0.9215077</td>
<td align="right">-3.067972</td>
</tr>
<tr class="even">
<td>IPO5</td>
<td align="right">-1.1385801</td>
<td align="right">5.886346</td>
<td align="right">-3.540342</td>
<td align="right">0.0023376</td>
<td align="right">0.9215077</td>
<td align="right">-3.043610</td>
</tr>
</tbody>
</table>
<p>For cluster 11:</p>
<pre class="r"><code>pseudo11.full &lt;- generate.pseudobulk(cluster11.full, labels = c(&quot;stim&quot;, &quot;replicate&quot;, &quot;cell.line&quot;))
pseudo11.full &lt;- filter.pseudobulk(pseudo11.full)
d11.full &lt;- DGEList(pseudo11.full$counts)
d11.full &lt;- calcNormFactors(d11.full)
keep11.full &lt;- filterByExpr(d11.full$counts)
d11.full &lt;- d11.full[keep11.full,]

voomed11.full &lt;- voomWithDreamWeights(d11.full, formde, pseudo11.full$meta, plot=FALSE)</code></pre>
<pre><code>Memory usage to store result: &gt; 9.3 Mb 
Dividing work into 100 chunks...

Total: 44 s</code></pre>
<pre class="r"><code>fitmm11.full = dream(voomed11.full, formde, pseudo11.full$meta)</code></pre>
<pre><code>Dividing work into 100 chunks...

Total: 73 s</code></pre>
<pre class="r"><code>try(topTable(fitmm11.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.full[12,1] &lt;- try(topTable(fitmm11.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.full[12, 1] &lt;- try(topTable(fitmm11.full, coef = &quot;stimoxidation&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm11.full, coef=&#39;stimoxidation&#39;,  number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;Oxidation DE genes, cluster 11 of full dataset&quot;)</code></pre>
<table>
<caption>Oxidation DE genes, cluster 11 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>UFD1</td>
<td align="right">0.3337069</td>
<td align="right">5.781937</td>
<td align="right">205.689082</td>
<td align="right">0.0000237</td>
<td align="right">0.1508227</td>
<td align="right">4.227302</td>
</tr>
<tr class="even">
<td>CASP3</td>
<td align="right">0.1468733</td>
<td align="right">4.615093</td>
<td align="right">119.494534</td>
<td align="right">0.0000700</td>
<td align="right">0.1518321</td>
<td align="right">3.976253</td>
</tr>
<tr class="odd">
<td>RAB5IF</td>
<td align="right">0.3853477</td>
<td align="right">6.098160</td>
<td align="right">16.935988</td>
<td align="right">0.0000714</td>
<td align="right">0.1518321</td>
<td align="right">3.971478</td>
</tr>
<tr class="even">
<td>HOOK2</td>
<td align="right">-0.3167323</td>
<td align="right">4.903409</td>
<td align="right">-10.960099</td>
<td align="right">0.0003681</td>
<td align="right">0.3643722</td>
<td align="right">-3.561956</td>
</tr>
<tr class="odd">
<td>NBR1</td>
<td align="right">-0.3818205</td>
<td align="right">4.987230</td>
<td align="right">-9.905540</td>
<td align="right">0.0005025</td>
<td align="right">0.3643722</td>
<td align="right">-3.479433</td>
</tr>
<tr class="even">
<td>PIAS2</td>
<td align="right">-0.8611081</td>
<td align="right">6.214527</td>
<td align="right">-10.457732</td>
<td align="right">0.0005120</td>
<td align="right">0.3643722</td>
<td align="right">-3.474402</td>
</tr>
<tr class="odd">
<td>C16orf91</td>
<td align="right">0.5100347</td>
<td align="right">4.476341</td>
<td align="right">6.411009</td>
<td align="right">0.0006794</td>
<td align="right">0.3643722</td>
<td align="right">3.397753</td>
</tr>
<tr class="even">
<td>ZNF711</td>
<td align="right">-0.6341825</td>
<td align="right">4.858036</td>
<td align="right">-9.243156</td>
<td align="right">0.0006915</td>
<td align="right">0.3643722</td>
<td align="right">-3.392916</td>
</tr>
<tr class="odd">
<td>INTS13</td>
<td align="right">-0.6639122</td>
<td align="right">4.257998</td>
<td align="right">-6.360292</td>
<td align="right">0.0007088</td>
<td align="right">0.3643722</td>
<td align="right">-3.386167</td>
</tr>
<tr class="even">
<td>THUMPD3-AS1</td>
<td align="right">0.4535166</td>
<td align="right">6.501627</td>
<td align="right">37.310383</td>
<td align="right">0.0007228</td>
<td align="right">0.3643722</td>
<td align="right">3.380780</td>
</tr>
</tbody>
</table>
<pre class="r"><code>try(topTable(fitmm11.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.full[12,2] &lt;- try(topTable(fitmm11.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.full[12, 2] &lt;- try(topTable(fitmm11.full, coef = &quot;stimpff&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm11.full, coef=&#39;stimpff&#39;, number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;PFF DE genes, cluster 11 of full dataset&quot;)</code></pre>
<table>
<caption>PFF DE genes, cluster 11 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>CASP3</td>
<td align="right">-0.1350093</td>
<td align="right">4.615093</td>
<td align="right">-62.325797</td>
<td align="right">0.0002573</td>
<td align="right">0.7050192</td>
<td align="right">-3.654899</td>
</tr>
<tr class="even">
<td>HOOK2</td>
<td align="right">-0.4805890</td>
<td align="right">4.903409</td>
<td align="right">-11.998546</td>
<td align="right">0.0002626</td>
<td align="right">0.7050192</td>
<td align="right">-3.649650</td>
</tr>
<tr class="odd">
<td>HNRNPUL1</td>
<td align="right">-0.1913315</td>
<td align="right">6.183334</td>
<td align="right">-49.151248</td>
<td align="right">0.0004126</td>
<td align="right">0.7050192</td>
<td align="right">-3.531911</td>
</tr>
<tr class="even">
<td>RRP12</td>
<td align="right">0.8265785</td>
<td align="right">4.193446</td>
<td align="right">43.304187</td>
<td align="right">0.0005342</td>
<td align="right">0.7050192</td>
<td align="right">3.463009</td>
</tr>
<tr class="odd">
<td>GLO1</td>
<td align="right">-0.2191587</td>
<td align="right">6.721759</td>
<td align="right">-37.210032</td>
<td align="right">0.0007261</td>
<td align="right">0.7050192</td>
<td align="right">-3.379530</td>
</tr>
<tr class="even">
<td>SMIM8</td>
<td align="right">0.4099462</td>
<td align="right">5.492173</td>
<td align="right">9.296700</td>
<td align="right">0.0007351</td>
<td align="right">0.7050192</td>
<td align="right">3.376145</td>
</tr>
<tr class="odd">
<td>FAM107B</td>
<td align="right">-0.7524282</td>
<td align="right">6.381878</td>
<td align="right">-8.769472</td>
<td align="right">0.0009430</td>
<td align="right">0.7050192</td>
<td align="right">-3.307009</td>
</tr>
<tr class="even">
<td>TAB2</td>
<td align="right">-0.2779572</td>
<td align="right">5.687088</td>
<td align="right">-8.599719</td>
<td align="right">0.0010087</td>
<td align="right">0.7050192</td>
<td align="right">-3.288094</td>
</tr>
<tr class="odd">
<td>COPS6</td>
<td align="right">-0.1710529</td>
<td align="right">6.393741</td>
<td align="right">-31.257820</td>
<td align="right">0.0010381</td>
<td align="right">0.7050192</td>
<td align="right">-3.280000</td>
</tr>
<tr class="even">
<td>KIAA1191</td>
<td align="right">-0.9060563</td>
<td align="right">4.595903</td>
<td align="right">-12.835114</td>
<td align="right">0.0011616</td>
<td align="right">0.7050192</td>
<td align="right">-3.248139</td>
</tr>
</tbody>
</table>
<p>For cluster 12:</p>
<pre class="r"><code>pseudo12.full &lt;- generate.pseudobulk(cluster12.full, labels = c(&quot;stim&quot;, &quot;replicate&quot;, &quot;cell.line&quot;))
pseudo12.full &lt;- filter.pseudobulk(pseudo12.full)
d12.full &lt;- DGEList(pseudo12.full$counts)
d12.full &lt;- calcNormFactors(d12.full)
keep12.full &lt;- filterByExpr(d12.full$counts)
d12.full &lt;- d12.full[keep12.full,]

voomed12.full &lt;- voomWithDreamWeights(d12.full, formde, pseudo12.full$meta, plot=FALSE)</code></pre>
<pre><code>Memory usage to store result: &gt; 19.1 Mb 
Dividing work into 100 chunks...

Total: 64 s</code></pre>
<pre class="r"><code>fitmm12.full = dream(voomed12.full, formde, pseudo12.full$meta)</code></pre>
<pre><code>Dividing work into 100 chunks...

Total: 99 s</code></pre>
<pre class="r"><code>try(topTable(fitmm12.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.full[13,1] &lt;- try(topTable(fitmm12.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.full[13, 1] &lt;- try(topTable(fitmm12.full, coef = &quot;stimoxidation&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm12.full, coef=&#39;stimoxidation&#39;,  number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;Oxidation DE genes, cluster 12 of full dataset&quot;)</code></pre>
<table>
<caption>Oxidation DE genes, cluster 12 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>STAG2</td>
<td align="right">-0.2795359</td>
<td align="right">7.049219</td>
<td align="right">-4.881581</td>
<td align="right">0.0001200</td>
<td align="right">0.3682276</td>
<td align="right">-3.846038</td>
</tr>
<tr class="even">
<td>HSPH1</td>
<td align="right">0.9722630</td>
<td align="right">6.382503</td>
<td align="right">5.067813</td>
<td align="right">0.0001682</td>
<td align="right">0.3682276</td>
<td align="right">3.762476</td>
</tr>
<tr class="odd">
<td>ZNF292</td>
<td align="right">-0.6390110</td>
<td align="right">7.735089</td>
<td align="right">-4.639751</td>
<td align="right">0.0002649</td>
<td align="right">0.3682276</td>
<td align="right">-3.647415</td>
</tr>
<tr class="even">
<td>TUBA1B</td>
<td align="right">-0.1791506</td>
<td align="right">10.745009</td>
<td align="right">-4.792449</td>
<td align="right">0.0002882</td>
<td align="right">0.3682276</td>
<td align="right">-3.625714</td>
</tr>
<tr class="odd">
<td>TOPBP1</td>
<td align="right">-0.3677642</td>
<td align="right">6.286182</td>
<td align="right">-4.667494</td>
<td align="right">0.0003319</td>
<td align="right">0.3682276</td>
<td align="right">-3.589051</td>
</tr>
<tr class="even">
<td>CENPH</td>
<td align="right">-0.4744264</td>
<td align="right">6.365170</td>
<td align="right">-4.502492</td>
<td align="right">0.0003573</td>
<td align="right">0.3682276</td>
<td align="right">-3.569785</td>
</tr>
<tr class="odd">
<td>PCMTD1</td>
<td align="right">-0.2743295</td>
<td align="right">6.464623</td>
<td align="right">-4.507899</td>
<td align="right">0.0004359</td>
<td align="right">0.3682276</td>
<td align="right">-3.517332</td>
</tr>
<tr class="even">
<td>PFDN2</td>
<td align="right">0.2797929</td>
<td align="right">6.759779</td>
<td align="right">4.396224</td>
<td align="right">0.0004507</td>
<td align="right">0.3682276</td>
<td align="right">3.508469</td>
</tr>
<tr class="odd">
<td>DDIT4</td>
<td align="right">-1.2298513</td>
<td align="right">5.696838</td>
<td align="right">-4.260260</td>
<td align="right">0.0004708</td>
<td align="right">0.3682276</td>
<td align="right">-3.496831</td>
</tr>
<tr class="even">
<td>NEMP1</td>
<td align="right">-0.5299124</td>
<td align="right">5.538425</td>
<td align="right">-4.222753</td>
<td align="right">0.0005117</td>
<td align="right">0.3682276</td>
<td align="right">-3.474558</td>
</tr>
</tbody>
</table>
<pre class="r"><code>try(topTable(fitmm12.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.full[13,2] &lt;- try(topTable(fitmm12.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.full[13, 2] &lt;- try(topTable(fitmm12.full, coef = &quot;stimpff&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm12.full, coef=&#39;stimpff&#39;, number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;PFF DE genes, cluster 12 of full dataset&quot;)</code></pre>
<table>
<caption>PFF DE genes, cluster 12 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>INPP1</td>
<td align="right">-0.7008030</td>
<td align="right">4.487767</td>
<td align="right">-4.832796</td>
<td align="right">0.0002017</td>
<td align="right">0.9995362</td>
<td align="right">-3.716833</td>
</tr>
<tr class="even">
<td>CTNND1</td>
<td align="right">0.3024524</td>
<td align="right">6.857478</td>
<td align="right">3.631563</td>
<td align="right">0.0019084</td>
<td align="right">0.9995362</td>
<td align="right">3.104135</td>
</tr>
<tr class="odd">
<td>IFITM2</td>
<td align="right">-0.9479504</td>
<td align="right">6.359395</td>
<td align="right">-3.727766</td>
<td align="right">0.0021404</td>
<td align="right">0.9995362</td>
<td align="right">-3.070022</td>
</tr>
<tr class="even">
<td>PML</td>
<td align="right">-1.4520108</td>
<td align="right">3.913760</td>
<td align="right">-3.546126</td>
<td align="right">0.0023382</td>
<td align="right">0.9995362</td>
<td align="right">-3.043525</td>
</tr>
<tr class="odd">
<td>PTP4A2</td>
<td align="right">-0.3874001</td>
<td align="right">7.505268</td>
<td align="right">-3.578691</td>
<td align="right">0.0024838</td>
<td align="right">0.9995362</td>
<td align="right">-3.025313</td>
</tr>
<tr class="even">
<td>TUBA1B</td>
<td align="right">-0.2064426</td>
<td align="right">10.745009</td>
<td align="right">-3.536465</td>
<td align="right">0.0030298</td>
<td align="right">0.9995362</td>
<td align="right">-2.964698</td>
</tr>
<tr class="odd">
<td>UTP14A</td>
<td align="right">-1.1613630</td>
<td align="right">4.342799</td>
<td align="right">-3.434613</td>
<td align="right">0.0033976</td>
<td align="right">0.9995362</td>
<td align="right">-2.929270</td>
</tr>
<tr class="even">
<td>GGA3</td>
<td align="right">-1.0139933</td>
<td align="right">3.727073</td>
<td align="right">-3.423978</td>
<td align="right">0.0035269</td>
<td align="right">0.9995362</td>
<td align="right">-2.917637</td>
</tr>
<tr class="odd">
<td>GYPC</td>
<td align="right">0.5532099</td>
<td align="right">6.445678</td>
<td align="right">3.400544</td>
<td align="right">0.0035400</td>
<td align="right">0.9995362</td>
<td align="right">2.916488</td>
</tr>
<tr class="even">
<td>MYO9B</td>
<td align="right">0.6573729</td>
<td align="right">5.779973</td>
<td align="right">3.431084</td>
<td align="right">0.0039455</td>
<td align="right">0.9995362</td>
<td align="right">2.882489</td>
</tr>
</tbody>
</table>
<p>For cluster 13:</p>
<pre class="r"><code>pseudo13.full &lt;- generate.pseudobulk(cluster13.full, labels = c(&quot;stim&quot;, &quot;replicate&quot;, &quot;cell.line&quot;))
pseudo13.full &lt;- filter.pseudobulk(pseudo13.full)
d13.full &lt;- DGEList(pseudo13.full$counts)
d13.full &lt;- calcNormFactors(d13.full)
keep13.full &lt;- filterByExpr(d13.full$counts)
d13.full &lt;- d13.full[keep13.full,]

voomed13.full &lt;- voomWithDreamWeights(d13.full, formde, pseudo13.full$meta, plot=FALSE)</code></pre>
<pre><code>Memory usage to store result: &gt; 2 Mb 
Dividing work into 100 chunks...

Total: 39 s</code></pre>
<pre class="r"><code>fitmm13.full = dream(voomed13.full, formde, pseudo13.full$meta)</code></pre>
<pre><code>Dividing work into 100 chunks...

Total: 42 s</code></pre>
<pre class="r"><code>try(topTable(fitmm13.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.full[14,1] &lt;- try(topTable(fitmm13.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.full[14, 1] &lt;- try(topTable(fitmm13.full, coef = &quot;stimoxidation&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm13.full, coef=&#39;stimoxidation&#39;,  number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;Oxidation DE genes, cluster 13 of full dataset&quot;)</code></pre>
<table>
<caption>Oxidation DE genes, cluster 13 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>SLC3A2</td>
<td align="right">-1.1320152</td>
<td align="right">7.394891</td>
<td align="right">-3.915997</td>
<td align="right">0.0022590</td>
<td align="right">0.9987138</td>
<td align="right">-3.053883</td>
</tr>
<tr class="even">
<td>RTN3</td>
<td align="right">-1.0731347</td>
<td align="right">7.135955</td>
<td align="right">-3.096018</td>
<td align="right">0.0089467</td>
<td align="right">0.9987138</td>
<td align="right">-2.614084</td>
</tr>
<tr class="odd">
<td>OCIAD1</td>
<td align="right">-0.7038799</td>
<td align="right">7.485484</td>
<td align="right">-3.040669</td>
<td align="right">0.0094685</td>
<td align="right">0.9987138</td>
<td align="right">-2.594660</td>
</tr>
<tr class="even">
<td>DDX24</td>
<td align="right">0.6094921</td>
<td align="right">7.637112</td>
<td align="right">2.982782</td>
<td align="right">0.0107632</td>
<td align="right">0.9987138</td>
<td align="right">2.550293</td>
</tr>
<tr class="odd">
<td>ZEB1</td>
<td align="right">1.1211542</td>
<td align="right">7.635778</td>
<td align="right">2.948504</td>
<td align="right">0.0121365</td>
<td align="right">0.9987138</td>
<td align="right">2.508151</td>
</tr>
<tr class="even">
<td>EIF4G3</td>
<td align="right">0.9371707</td>
<td align="right">7.419199</td>
<td align="right">2.769583</td>
<td align="right">0.0160426</td>
<td align="right">0.9987138</td>
<td align="right">2.407945</td>
</tr>
<tr class="odd">
<td>MTPN</td>
<td align="right">-0.8288746</td>
<td align="right">7.213717</td>
<td align="right">-2.735103</td>
<td align="right">0.0170152</td>
<td align="right">0.9987138</td>
<td align="right">-2.386379</td>
</tr>
<tr class="even">
<td>EIF4EBP1</td>
<td align="right">-1.2259637</td>
<td align="right">7.707041</td>
<td align="right">-2.678921</td>
<td align="right">0.0189398</td>
<td align="right">0.9987138</td>
<td align="right">-2.346714</td>
</tr>
<tr class="odd">
<td>KMT2E</td>
<td align="right">-0.6181670</td>
<td align="right">6.754347</td>
<td align="right">-2.694991</td>
<td align="right">0.0203287</td>
<td align="right">0.9987138</td>
<td align="right">-2.320225</td>
</tr>
<tr class="even">
<td>TMEM14C</td>
<td align="right">-0.7881404</td>
<td align="right">7.676246</td>
<td align="right">-2.644494</td>
<td align="right">0.0204691</td>
<td align="right">0.9987138</td>
<td align="right">-2.317637</td>
</tr>
</tbody>
</table>
<pre class="r"><code>try(topTable(fitmm13.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.full[14,2] &lt;- try(topTable(fitmm13.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.full[14, 2] &lt;- try(topTable(fitmm13.full, coef = &quot;stimpff&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm13.full, coef=&#39;stimpff&#39;, number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;PFF DE genes, cluster 13 of full dataset&quot;)</code></pre>
<table>
<caption>PFF DE genes, cluster 13 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>TMED4</td>
<td align="right">-1.8434579</td>
<td align="right">6.455944</td>
<td align="right">-3.867491</td>
<td align="right">0.0025769</td>
<td align="right">0.8777568</td>
<td align="right">-3.014164</td>
</tr>
<tr class="even">
<td>NARS</td>
<td align="right">-0.9494682</td>
<td align="right">6.878313</td>
<td align="right">-3.198280</td>
<td align="right">0.0070066</td>
<td align="right">0.8777568</td>
<td align="right">-2.696530</td>
</tr>
<tr class="odd">
<td>BASP1</td>
<td align="right">-1.3850795</td>
<td align="right">7.949752</td>
<td align="right">-3.227206</td>
<td align="right">0.0076988</td>
<td align="right">0.8777568</td>
<td align="right">-2.665000</td>
</tr>
<tr class="even">
<td>BRD4</td>
<td align="right">-1.1463911</td>
<td align="right">6.983364</td>
<td align="right">-3.476846</td>
<td align="right">0.0077063</td>
<td align="right">0.8777568</td>
<td align="right">-2.664672</td>
</tr>
<tr class="odd">
<td>CLIC1</td>
<td align="right">-1.2189258</td>
<td align="right">7.576930</td>
<td align="right">-3.190252</td>
<td align="right">0.0079482</td>
<td align="right">0.8777568</td>
<td align="right">-2.654262</td>
</tr>
<tr class="even">
<td>ITM2B</td>
<td align="right">-0.9607348</td>
<td align="right">8.717527</td>
<td align="right">-3.140328</td>
<td align="right">0.0087803</td>
<td align="right">0.8777568</td>
<td align="right">-2.620491</td>
</tr>
<tr class="odd">
<td>ENSA</td>
<td align="right">0.6691687</td>
<td align="right">6.713721</td>
<td align="right">3.169453</td>
<td align="right">0.0088312</td>
<td align="right">0.8777568</td>
<td align="right">2.618521</td>
</tr>
<tr class="even">
<td>MACF1</td>
<td align="right">-1.1527632</td>
<td align="right">7.186827</td>
<td align="right">-3.079351</td>
<td align="right">0.0104180</td>
<td align="right">0.8777568</td>
<td align="right">-2.561636</td>
</tr>
<tr class="odd">
<td>EIF4A2</td>
<td align="right">-0.8480838</td>
<td align="right">7.603126</td>
<td align="right">-2.959081</td>
<td align="right">0.0110769</td>
<td align="right">0.8777568</td>
<td align="right">-2.540265</td>
</tr>
<tr class="even">
<td>EMC7</td>
<td align="right">-0.8323500</td>
<td align="right">6.656162</td>
<td align="right">-2.948624</td>
<td align="right">0.0113066</td>
<td align="right">0.8777568</td>
<td align="right">-2.533076</td>
</tr>
</tbody>
</table>
<p>For cluster 14:</p>
<pre class="r"><code>pseudo14.full &lt;- generate.pseudobulk(cluster14.full, labels = c(&quot;stim&quot;, &quot;replicate&quot;, &quot;cell.line&quot;))
pseudo14.full &lt;- filter.pseudobulk(pseudo14.full)
d14.full &lt;- DGEList(pseudo14.full$counts)
d14.full &lt;- calcNormFactors(d14.full)
keep14.full &lt;- filterByExpr(d14.full$counts)
d14.full &lt;- d14.full[keep14.full,]

voomed14.full &lt;- voomWithDreamWeights(d14.full, formde, pseudo14.full$meta, plot=FALSE)</code></pre>
<pre><code>Memory usage to store result: &gt; 10.7 Mb 
Dividing work into 100 chunks...

Total: 57 s</code></pre>
<pre class="r"><code>fitmm14.full = dream(voomed14.full, formde, pseudo14.full$meta)</code></pre>
<pre><code>Dividing work into 100 chunks...

Total: 72 s</code></pre>
<pre class="r"><code>try(topTable(fitmm14.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.full[15,1] &lt;- try(topTable(fitmm14.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.full[15, 1] &lt;- try(topTable(fitmm14.full, coef = &quot;stimoxidation&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm14.full, coef=&#39;stimoxidation&#39;,  number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;Oxidation DE genes, cluster 14 of full dataset&quot;)</code></pre>
<table>
<caption>Oxidation DE genes, cluster 14 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>DNAJC2</td>
<td align="right">0.5740027</td>
<td align="right">5.862365</td>
<td align="right">5.379651</td>
<td align="right">0.0000614</td>
<td align="right">0.1647116</td>
<td align="right">4.007434</td>
</tr>
<tr class="even">
<td>ZNF292</td>
<td align="right">-0.4914593</td>
<td align="right">7.500371</td>
<td align="right">-5.588733</td>
<td align="right">0.0000667</td>
<td align="right">0.1647116</td>
<td align="right">-3.987670</td>
</tr>
<tr class="odd">
<td>HSPA8</td>
<td align="right">0.4683029</td>
<td align="right">9.008456</td>
<td align="right">5.652030</td>
<td align="right">0.0001136</td>
<td align="right">0.1868733</td>
<td align="right">3.859639</td>
</tr>
<tr class="even">
<td>SRRT</td>
<td align="right">0.6759025</td>
<td align="right">5.111548</td>
<td align="right">4.811318</td>
<td align="right">0.0002759</td>
<td align="right">0.1943205</td>
<td align="right">3.636913</td>
</tr>
<tr class="odd">
<td>RFC1</td>
<td align="right">-0.4241919</td>
<td align="right">7.006057</td>
<td align="right">-4.574730</td>
<td align="right">0.0003116</td>
<td align="right">0.1943205</td>
<td align="right">-3.605488</td>
</tr>
<tr class="even">
<td>PNN</td>
<td align="right">0.3408272</td>
<td align="right">7.385116</td>
<td align="right">4.742263</td>
<td align="right">0.0003132</td>
<td align="right">0.1943205</td>
<td align="right">3.604099</td>
</tr>
<tr class="odd">
<td>DARS</td>
<td align="right">-0.7065557</td>
<td align="right">6.634966</td>
<td align="right">-4.724262</td>
<td align="right">0.0003282</td>
<td align="right">0.1943205</td>
<td align="right">-3.591998</td>
</tr>
<tr class="even">
<td>NRF1</td>
<td align="right">0.4538313</td>
<td align="right">5.228099</td>
<td align="right">4.658297</td>
<td align="right">0.0003674</td>
<td align="right">0.1943205</td>
<td align="right">3.562487</td>
</tr>
<tr class="odd">
<td>KDM4B</td>
<td align="right">-0.6777921</td>
<td align="right">5.709377</td>
<td align="right">-4.628753</td>
<td align="right">0.0003762</td>
<td align="right">0.1943205</td>
<td align="right">-3.556213</td>
</tr>
<tr class="even">
<td>SRSF3</td>
<td align="right">0.6147437</td>
<td align="right">7.394637</td>
<td align="right">4.461610</td>
<td align="right">0.0003936</td>
<td align="right">0.1943205</td>
<td align="right">3.544338</td>
</tr>
</tbody>
</table>
<pre class="r"><code>try(topTable(fitmm14.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.full[15,2] &lt;- try(topTable(fitmm14.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.full[15, 2] &lt;- try(topTable(fitmm14.full, coef = &quot;stimpff&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm14.full, coef=&#39;stimpff&#39;, number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;PFF DE genes, cluster 14 of full dataset&quot;)</code></pre>
<table>
<caption>PFF DE genes, cluster 14 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>NRF1</td>
<td align="right">0.5714660</td>
<td align="right">5.228099</td>
<td align="right">4.561519</td>
<td align="right">0.0004379</td>
<td align="right">0.9998006</td>
<td align="right">3.516111</td>
</tr>
<tr class="even">
<td>ACVR1</td>
<td align="right">-0.8726210</td>
<td align="right">5.744578</td>
<td align="right">-4.476672</td>
<td align="right">0.0006656</td>
<td align="right">0.9998006</td>
<td align="right">-3.403351</td>
</tr>
<tr class="odd">
<td>PSMC1</td>
<td align="right">0.5374138</td>
<td align="right">6.907646</td>
<td align="right">3.876406</td>
<td align="right">0.0016883</td>
<td align="right">0.9998006</td>
<td align="right">3.140211</td>
</tr>
<tr class="even">
<td>WDPCP</td>
<td align="right">0.8764340</td>
<td align="right">6.424013</td>
<td align="right">3.624533</td>
<td align="right">0.0022780</td>
<td align="right">0.9998006</td>
<td align="right">3.051365</td>
</tr>
<tr class="odd">
<td>SLMAP</td>
<td align="right">0.7943469</td>
<td align="right">5.785561</td>
<td align="right">3.599962</td>
<td align="right">0.0027937</td>
<td align="right">0.9998006</td>
<td align="right">2.989568</td>
</tr>
<tr class="even">
<td>PDE10A</td>
<td align="right">1.2632021</td>
<td align="right">6.703481</td>
<td align="right">3.648538</td>
<td align="right">0.0030005</td>
<td align="right">0.9998006</td>
<td align="right">2.967687</td>
</tr>
<tr class="odd">
<td>ARFGEF2</td>
<td align="right">0.6416379</td>
<td align="right">5.503697</td>
<td align="right">3.538134</td>
<td align="right">0.0032584</td>
<td align="right">0.9998006</td>
<td align="right">2.942245</td>
</tr>
<tr class="even">
<td>FABP5</td>
<td align="right">-1.2502629</td>
<td align="right">8.375723</td>
<td align="right">-3.589111</td>
<td align="right">0.0034102</td>
<td align="right">0.9998006</td>
<td align="right">-2.928114</td>
</tr>
<tr class="odd">
<td>SNHG19</td>
<td align="right">-0.7129359</td>
<td align="right">5.391383</td>
<td align="right">-3.567179</td>
<td align="right">0.0038721</td>
<td align="right">0.9998006</td>
<td align="right">-2.888396</td>
</tr>
<tr class="even">
<td>SCAF8</td>
<td align="right">0.7482173</td>
<td align="right">5.315758</td>
<td align="right">3.332448</td>
<td align="right">0.0042206</td>
<td align="right">0.9998006</td>
<td align="right">2.861185</td>
</tr>
</tbody>
</table>
<p>For cluster 15:</p>
<pre class="r"><code>pseudo15.full &lt;- generate.pseudobulk(cluster15.full, labels = c(&quot;stim&quot;, &quot;replicate&quot;, &quot;cell.line&quot;))
pseudo15.full &lt;- filter.pseudobulk(pseudo15.full)
d15.full &lt;- DGEList(pseudo15.full$counts)
d15.full &lt;- calcNormFactors(d15.full)
keep15.full &lt;- filterByExpr(d15.full$counts)
d15.full &lt;- d15.full[keep15.full,]

voomed15.full &lt;- voomWithDreamWeights(d15.full, formde, pseudo15.full$meta, plot=FALSE)</code></pre>
<pre><code>Memory usage to store result: &gt; 5.7 Mb 
Dividing work into 100 chunks...

Total: 40 s</code></pre>
<pre class="r"><code>fitmm15.full = dream(voomed15.full, formde, pseudo15.full$meta)</code></pre>
<pre><code>Dividing work into 100 chunks...

Total: 60 s</code></pre>
<pre class="r"><code>try(topTable(fitmm15.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>[1] 2</code></pre>
<pre class="r"><code>de.summary.full[16,1] &lt;- try(topTable(fitmm15.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()
topTable(fitmm15.full, coef=&#39;stimoxidation&#39;,  number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;Oxidation DE genes, cluster 15 of full dataset&quot;)</code></pre>
<table>
<caption>Oxidation DE genes, cluster 15 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>BNIP3</td>
<td align="right">-2.3788632</td>
<td align="right">7.255471</td>
<td align="right">-7.000481</td>
<td align="right">0.0000030</td>
<td align="right">0.0073533</td>
<td align="right">-4.669213</td>
</tr>
<tr class="even">
<td>LRRFIP1</td>
<td align="right">-0.8298665</td>
<td align="right">7.016044</td>
<td align="right">-5.607233</td>
<td align="right">0.0000391</td>
<td align="right">0.0475339</td>
<td align="right">-4.112791</td>
</tr>
<tr class="odd">
<td>RAB32</td>
<td align="right">-1.3017998</td>
<td align="right">7.505210</td>
<td align="right">-4.770844</td>
<td align="right">0.0001528</td>
<td align="right">0.1238906</td>
<td align="right">-3.786433</td>
</tr>
<tr class="even">
<td>C1QBP</td>
<td align="right">0.4654185</td>
<td align="right">6.706781</td>
<td align="right">4.942556</td>
<td align="right">0.0002174</td>
<td align="right">0.1321639</td>
<td align="right">3.697918</td>
</tr>
<tr class="odd">
<td>BNIP3L</td>
<td align="right">-0.9258576</td>
<td align="right">7.792949</td>
<td align="right">-4.709839</td>
<td align="right">0.0003332</td>
<td align="right">0.1620677</td>
<td align="right">-3.588020</td>
</tr>
<tr class="even">
<td>TCF12</td>
<td align="right">0.4501994</td>
<td align="right">8.167328</td>
<td align="right">4.176448</td>
<td align="right">0.0007125</td>
<td align="right">0.2429316</td>
<td align="right">3.384718</td>
</tr>
<tr class="odd">
<td>API5</td>
<td align="right">-0.7550817</td>
<td align="right">6.314472</td>
<td align="right">-3.972095</td>
<td align="right">0.0008937</td>
<td align="right">0.2429316</td>
<td align="right">-3.322019</td>
</tr>
<tr class="even">
<td>ERBB3</td>
<td align="right">0.8502367</td>
<td align="right">8.106173</td>
<td align="right">4.047147</td>
<td align="right">0.0009265</td>
<td align="right">0.2429316</td>
<td align="right">3.311954</td>
</tr>
<tr class="odd">
<td>YWHAH</td>
<td align="right">-1.3293554</td>
<td align="right">7.720302</td>
<td align="right">-3.962038</td>
<td align="right">0.0011178</td>
<td align="right">0.2429316</td>
<td align="right">-3.259077</td>
</tr>
<tr class="even">
<td>NDUFB3</td>
<td align="right">1.0673068</td>
<td align="right">6.539035</td>
<td align="right">4.074979</td>
<td align="right">0.0011567</td>
<td align="right">0.2429316</td>
<td align="right">3.249339</td>
</tr>
</tbody>
</table>
<pre class="r"><code>try(topTable(fitmm15.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.full[16,2] &lt;- try(topTable(fitmm15.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in sign(tab$t) : non-numeric argument to mathematical function</code></pre>
<pre><code>Error in de.summary.full[16, 2] &lt;- try(topTable(fitmm15.full, coef = &quot;stimpff&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm15.full, coef=&#39;stimpff&#39;, number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;PFF DE genes, cluster 15 of full dataset&quot;)</code></pre>
<table>
<caption>PFF DE genes, cluster 15 of full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">z.std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ATP5MF</td>
<td align="right">1.0031209</td>
<td align="right">7.725800</td>
<td align="right">4.090115</td>
<td align="right">0.0008326</td>
<td align="right">0.993082</td>
<td align="right">3.341708</td>
</tr>
<tr class="even">
<td>PPIL4</td>
<td align="right">-0.7158346</td>
<td align="right">6.602625</td>
<td align="right">-3.595591</td>
<td align="right">0.0024296</td>
<td align="right">0.993082</td>
<td align="right">-3.031973</td>
</tr>
<tr class="odd">
<td>UFC1</td>
<td align="right">0.6147131</td>
<td align="right">6.829605</td>
<td align="right">3.503578</td>
<td align="right">0.0026095</td>
<td align="right">0.993082</td>
<td align="right">3.010351</td>
</tr>
<tr class="even">
<td>RYBP</td>
<td align="right">-0.7668373</td>
<td align="right">6.761500</td>
<td align="right">-3.509350</td>
<td align="right">0.0028980</td>
<td align="right">0.993082</td>
<td align="right">-2.978354</td>
</tr>
<tr class="odd">
<td>NFIB</td>
<td align="right">1.2943289</td>
<td align="right">7.008338</td>
<td align="right">3.494298</td>
<td align="right">0.0030932</td>
<td align="right">0.993082</td>
<td align="right">2.958319</td>
</tr>
<tr class="even">
<td>WTAP</td>
<td align="right">-0.6469612</td>
<td align="right">6.617095</td>
<td align="right">-3.430963</td>
<td align="right">0.0034244</td>
<td align="right">0.993082</td>
<td align="right">-2.926829</td>
</tr>
<tr class="odd">
<td>PLAC9</td>
<td align="right">0.9389682</td>
<td align="right">7.667369</td>
<td align="right">3.279131</td>
<td align="right">0.0046718</td>
<td align="right">0.993082</td>
<td align="right">2.828836</td>
</tr>
<tr class="even">
<td>CHD1</td>
<td align="right">-0.6901266</td>
<td align="right">6.568151</td>
<td align="right">-3.195336</td>
<td align="right">0.0050995</td>
<td align="right">0.993082</td>
<td align="right">-2.800677</td>
</tr>
<tr class="odd">
<td>UBE2K</td>
<td align="right">0.5539354</td>
<td align="right">7.077142</td>
<td align="right">3.144072</td>
<td align="right">0.0056108</td>
<td align="right">0.993082</td>
<td align="right">2.769700</td>
</tr>
<tr class="even">
<td>SERF2</td>
<td align="right">0.6754677</td>
<td align="right">10.240344</td>
<td align="right">3.174493</td>
<td align="right">0.0058148</td>
<td align="right">0.993082</td>
<td align="right">2.758048</td>
</tr>
</tbody>
</table>
<p>For cluster 16:</p>
<pre class="r"><code>pseudo16.full &lt;- generate.pseudobulk(cluster16.full, labels = c(&quot;stim&quot;, &quot;replicate&quot;, &quot;cell.line&quot;))
pseudo16.full &lt;- filter.pseudobulk(pseudo16.full)
d16.full &lt;- DGEList(pseudo16.full$counts)
d16.full &lt;- calcNormFactors(d16.full)
keep16.full &lt;- filterByExpr(d16.full$counts)
d16.full &lt;- d16.full[keep16.full,]

voomed16.full &lt;- voomWithDreamWeights(d16.full, formde, pseudo16.full$meta, plot=FALSE)</code></pre>
<pre><code>Error in if (inherits(possibleError, &quot;error&quot;) &amp;&amp; grep(&quot;the fixed-effects model matrix is column rank deficient&quot;, : missing value where TRUE/FALSE needed</code></pre>
<pre class="r"><code>fitmm16.full = dream(voomed16.full, formde, pseudo16.full$meta)</code></pre>
<pre><code>Error in dream(voomed16.full, formde, pseudo16.full$meta): object &#39;voomed16.full&#39; not found</code></pre>
<pre class="r"><code>try(topTable(fitmm16.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in topTable(fitmm16.full, coef = &quot;stimoxidation&quot;, p.value = 0.05,  : 
  object &#39;fitmm16.full&#39; not found</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.full[17,1] &lt;- try(topTable(fitmm16.full, coef=&#39;stimoxidation&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in topTable(fitmm16.full, coef = &quot;stimoxidation&quot;, p.value = 0.05,  : 
  object &#39;fitmm16.full&#39; not found</code></pre>
<pre><code>Error in de.summary.full[17, 1] &lt;- try(topTable(fitmm16.full, coef = &quot;stimoxidation&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm16.full, coef=&#39;stimoxidation&#39;,  number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;Oxidation DE genes, cluster 16 of full dataset&quot;)</code></pre>
<pre><code>Error in topTable(fitmm16.full, coef = &quot;stimoxidation&quot;, number = 10000): object &#39;fitmm16.full&#39; not found</code></pre>
<pre class="r"><code>try(topTable(fitmm16.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in topTable(fitmm16.full, coef = &quot;stimpff&quot;, p.value = 0.05, number = 10000) : 
  object &#39;fitmm16.full&#39; not found</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>de.summary.full[17,2] &lt;- try(topTable(fitmm16.full, coef=&#39;stimpff&#39;, p.value = 0.05, number = 10000)) %&gt;% nrow()</code></pre>
<pre><code>Error in topTable(fitmm16.full, coef = &quot;stimpff&quot;, p.value = 0.05, number = 10000) : 
  object &#39;fitmm16.full&#39; not found</code></pre>
<pre><code>Error in de.summary.full[17, 2] &lt;- try(topTable(fitmm16.full, coef = &quot;stimpff&quot;, : number of items to replace is not a multiple of replacement length</code></pre>
<pre class="r"><code>topTable(fitmm16.full, coef=&#39;stimpff&#39;, number = 10000) %&gt;% head(10) %&gt;% kable(caption = &quot;PFF DE genes, cluster 16 of full dataset&quot;)</code></pre>
<pre><code>Error in topTable(fitmm16.full, coef = &quot;stimpff&quot;, number = 10000): object &#39;fitmm16.full&#39; not found</code></pre>
<p>To summarize, at 5% FDR threshold, I detect the following number of DE genes in each cluster:</p>
<pre class="r"><code>de.summary.full %&gt;% kable(caption = &quot;DE genes, full dataset&quot;)</code></pre>
<table>
<caption>DE genes, full dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">oxidation</th>
<th align="right">pff</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>cluster0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td>cluster1</td>
<td align="right">1360</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td>cluster2</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td>cluster3</td>
<td align="right">30</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td>cluster4</td>
<td align="right">93</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td>cluster5</td>
<td align="right">63</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td>cluster6</td>
<td align="right">108</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td>cluster7</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td>cluster8</td>
<td align="right">634</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td>cluster9</td>
<td align="right">92</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td>cluster10</td>
<td align="right">2</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td>cluster11</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td>cluster12</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td>cluster13</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td>cluster14</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td>cluster15</td>
<td align="right">2</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td>cluster16</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<pre class="r"><code>de.summary.neuron %&gt;% kable(caption = &quot;DE genes, neuronal dataset&quot;)</code></pre>
<table>
<caption>DE genes, neuronal dataset</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">oxidation</th>
<th align="right">pff</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>cluster0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td>cluster1</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td>cluster2</td>
<td align="right">92</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td>cluster3</td>
<td align="right">152</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td>cluster4</td>
<td align="right">9</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td>cluster5</td>
<td align="right">628</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td>cluster6</td>
<td align="right">91</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td>cluster7</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td>cluster8</td>
<td align="right">2</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td>cluster9</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td>cluster10</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td>cluster11</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] knitr_1.23               cowplot_1.0.0           
 [3] RColorBrewer_1.1-2       forcats_0.5.1           
 [5] stringr_1.4.0            dplyr_1.0.7             
 [7] purrr_0.3.4              readr_1.4.0             
 [9] tidyr_1.1.3              tibble_3.1.2            
[11] tidyverse_1.3.1          Seurat_3.1.3            
[13] BiocParallel_1.18.0      variancePartition_1.14.1
[15] Biobase_2.44.0           BiocGenerics_0.30.0     
[17] scales_1.1.1             foreach_1.5.1           
[19] ggplot2_3.3.5            edgeR_3.26.5            
[21] limma_3.40.6             workflowr_1.6.2         

loaded via a namespace (and not attached):
  [1] snow_0.4-3          readxl_1.3.1        backports_1.2.1    
  [4] plyr_1.8.6          igraph_1.2.4.1      lazyeval_0.2.2     
  [7] splines_3.6.1       listenv_0.7.0       digest_0.6.27      
 [10] htmltools_0.5.1.1   lmerTest_3.1-0      gdata_2.18.0       
 [13] fansi_0.5.0         magrittr_2.0.1      cluster_2.1.0      
 [16] doParallel_1.0.14   ROCR_1.0-7          globals_0.12.4     
 [19] modelr_0.1.8        RcppParallel_5.1.4  prettyunits_1.1.1  
 [22] colorspace_2.0-2    rvest_1.0.0         rappdirs_0.3.1     
 [25] ggrepel_0.8.1       haven_2.3.1         xfun_0.24          
 [28] crayon_1.4.1        jsonlite_1.7.2      lme4_1.1-21        
 [31] survival_3.2-11     zoo_1.8-6           iterators_1.0.13   
 [34] ape_5.4-1           glue_1.4.2          gtable_0.3.0       
 [37] leiden_0.3.1        future.apply_1.3.0  DBI_1.1.0          
 [40] bibtex_0.4.2        Rcpp_1.0.6          metap_1.1          
 [43] viridisLite_0.3.0   progress_1.2.2      reticulate_1.16    
 [46] rsvd_1.0.1          tsne_0.1-3          htmlwidgets_1.5.3  
 [49] httr_1.4.2          gplots_3.0.1.1      ellipsis_0.3.2     
 [52] ica_1.0-2           pkgconfig_2.0.3     uwot_0.1.5         
 [55] dbplyr_2.1.1        locfit_1.5-9.1      utf8_1.1.4         
 [58] tidyselect_1.1.0    rlang_0.4.11        reshape2_1.4.4     
 [61] later_1.2.0         cellranger_1.1.0    munsell_0.5.0      
 [64] tools_3.6.1         cli_2.5.0           generics_0.1.0     
 [67] pacman_0.5.1        broom_0.7.8         ggridges_0.5.1     
 [70] evaluate_0.14       yaml_2.2.1          npsurv_0.4-0       
 [73] fs_1.3.1            fitdistrplus_1.0-14 caTools_1.17.1.2   
 [76] RANN_2.6.1          pbapply_1.4-0       future_1.14.0      
 [79] nlme_3.1-140        whisker_0.3-2       xml2_1.3.2         
 [82] rstudioapi_0.13     compiler_3.6.1      pbkrtest_0.4-7     
 [85] plotly_4.9.4.1      png_0.1-7           lsei_1.2-0         
 [88] reprex_2.0.0        stringi_1.6.2       highr_0.8          
 [91] lattice_0.20-38     Matrix_1.2-18       nloptr_1.2.2.1     
 [94] vctrs_0.3.8         pillar_1.6.1        lifecycle_1.0.0    
 [97] Rdpack_0.11-0       lmtest_0.9-37       RcppAnnoy_0.0.12   
[100] data.table_1.14.0   bitops_1.0-6        irlba_2.3.3        
[103] gbRd_0.4-11         httpuv_1.6.1        colorRamps_2.3     
[106] R6_2.5.0            promises_1.2.0.1    KernSmooth_2.23-15 
[109] gridExtra_2.3       codetools_0.2-16    boot_1.3-23        
[112] MASS_7.3-51.4       gtools_3.8.1        assertthat_0.2.1   
[115] rprojroot_2.0.2     withr_2.4.2         sctransform_0.2.0  
[118] hms_1.1.0           grid_3.6.1          minqa_1.2.4        
[121] rmarkdown_1.13      Rtsne_0.15          git2r_0.26.1       
[124] numDeriv_2016.8-1.1 lubridate_1.7.10   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
