<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Ben Umans" />

<meta name="date" content="2022-02-03" />

<title>hippo</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Organoid_Stress_Pilot</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/bumans/Organoid_Stress_Pilot">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">hippo</h1>
<h4 class="author">Ben Umans</h4>
<h4 class="date">2022-02-03</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-08
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Organoid_Stress_Pilot/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210705code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210705)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210705code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210705)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombumansOrganoidStressPilottree33d0b074aa4e6e095804dbcf979e0d479b04ed51targetblank33d0b07a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bumans/Organoid_Stress_Pilot/tree/33d0b074aa4e6e095804dbcf979e0d479b04ed51" target="_blank">33d0b07</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombumansOrganoidStressPilottree33d0b074aa4e6e095804dbcf979e0d479b04ed51targetblank33d0b07a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/bumans/Organoid_Stress_Pilot/tree/33d0b074aa4e6e095804dbcf979e0d479b04ed51" target="_blank">33d0b07</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store
    Ignored:    output/4vs_avg-expression.png
    Ignored:    output/4vs_logfc.png
    Ignored:    output/4vs_pvals.png
    Ignored:    output/de_clusters_interaction.Rmd
    Ignored:    output/de_clusters_interaction2.Rmd
    Ignored:    output/fitmm0.neuron.RDS
    Ignored:    output/fitmm1.neuron.RDS
    Ignored:    output/fitmm2.neuron.RDS
    Ignored:    output/fitmm3.neuron.RDS
    Ignored:    output/fitmm4.neuron.RDS
    Ignored:    output/fitmm5.neuron.RDS
    Ignored:    output/fitmm6.neuron.RDS
    Ignored:    output/fitmm7.neuron.RDS
    Ignored:    output/fitmm8.neuron.RDS
    Ignored:    output/fitmmall.neuron.RDS
    Ignored:    output/genes_MT.png
    Ignored:    output/genes_cells.png
    Ignored:    output/genes_umi-cluster.png
    Ignored:    output/genes_umi.png
    Ignored:    output/interaction_varpart.csv
    Ignored:    output/merged_full_dataset.RDS
    Ignored:    output/neuron0_oxidation_table.csv
    Ignored:    output/neuron1_oxidation_table.csv
    Ignored:    output/neuron2_oxidation_table.csv
    Ignored:    output/neuron3_oxidation_table.csv
    Ignored:    output/neuron4_oxidation_table.csv
    Ignored:    output/neuron5_oxidation_table.csv
    Ignored:    output/neuron6_oxidation_table.csv
    Ignored:    output/neuron7_oxidation_table.csv
    Ignored:    output/neuron8_oxidation_table.csv
    Ignored:    output/neuronall_oxidation_table.csv
    Ignored:    output/organoid.combined.full.sct.neuron.RDS
    Ignored:    output/organoid.combined.sct.neuron.markers.RDS
    Ignored:    output/organoid.combined.sct.neuron.unstim.markers.RDS
    Ignored:    output/organoid.combined.sct03.RDS
    Ignored:    output/organoid.combined.sct05.RDS
    Ignored:    output/organoid.combined.sct1.RDS
    Ignored:    output/organoid_combined_sct_neuron_resolution_0.02.png
    Ignored:    output/organoid_combined_sct_neuron_resolution_0.05.png
    Ignored:    output/organoid_combined_sct_neuron_resolution_0.1.png
    Ignored:    output/organoid_combined_sct_neuron_resolution_0.3.png
    Ignored:    output/organoid_combined_sct_neuron_resolution_0.5.png
    Ignored:    output/resolution_test.csv
    Ignored:    output/resolution_test_pff.csv
    Ignored:    output/sce_hippo_test.RDS

Untracked files:
    Untracked:  analysis/cell_cycle_full.Rmd
    Untracked:  analysis/topics.Rmd
    Untracked:  data/Snakefile_cellranger
    Untracked:  data/all.vcf.bed
    Untracked:  data/australia.rds
    Untracked:  data/cellranger.38.dict
    Untracked:  data/cluster.json
    Untracked:  data/config.yaml
    Untracked:  data/environment.yaml
    Untracked:  data/exclude_fulldata.rds
    Untracked:  data/human.unsorted.vcf
    Untracked:  data/human.vcf
    Untracked:  data/human.vcf.bed
    Untracked:  data/library9.rds
    Untracked:  data/log/
    Untracked:  data/mysterycell.rds
    Untracked:  data/neuron.control.sce.RDS
    Untracked:  data/organoid.combined.sct03.neuron.RDS
    Untracked:  data/organoid_combined_sct03.RDS
    Untracked:  data/reheaded.vcf
    Untracked:  data/snakelog.out
    Untracked:  data/snakemake_cellranger.batch
    Untracked:  data/sorted.vcf
    Untracked:  data/submit.sh
    Untracked:  data/unsorted.vcf
    Untracked:  data/unsorted_header
    Untracked:  data_S2/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/hippo.Rmd</code>) and HTML (<code>docs/hippo.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bumans/Organoid_Stress_Pilot/blob/33d0b074aa4e6e095804dbcf979e0d479b04ed51/analysis/hippo.Rmd" target="_blank">33d0b07</a>
</td>
<td>
Ben Umans
</td>
<td>
2022-02-08
</td>
<td>
wflow_publish(files = c(“analysis/hippo.Rmd”, “analysis/index.Rmd”))
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Mengjie Chen gave a really interesting talk this week about her scRNAseq pre-processing pipeline, HIPPO. The routine is very different from the Seurat approach, iteratively splitting clusters in a hierarchical manner, rather than identifying highly variable genes across the whole map. One can imagine helpful aspects of this–particularly for a developmental tree, where you want to be able to compare at similar levels of resolution (ie, not comparing retinal pigmented epithelium to mesoderm). I’d like to try it out with my organoid data just to see what would happen.</p>
<pre class="r"><code>Sys.setenv(TZ=&#39;America/Chicago&#39;)
library(HIPPO)
library(Seurat)</code></pre>
<pre><code>Attaching SeuratObject</code></pre>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>Registered S3 method overwritten by &#39;cli&#39;:
  method     from         
  print.boxx spatstat.geom</code></pre>
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──</code></pre>
<pre><code>✓ ggplot2 3.3.5     ✓ purrr   0.3.4
✓ tibble  3.1.2     ✓ dplyr   1.0.7
✓ tidyr   1.1.3     ✓ stringr 1.4.0
✓ readr   1.4.0     ✓ forcats 0.5.1</code></pre>
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(ggplot2)
library(RColorBrewer)
library(knitr)
library(viridis)</code></pre>
<pre><code>Loading required package: viridisLite</code></pre>
<pre class="r"><code>library(SingleCellExperiment)</code></pre>
<pre><code>Loading required package: SummarizedExperiment</code></pre>
<pre><code>Loading required package: MatrixGenerics</code></pre>
<pre><code>Loading required package: matrixStats</code></pre>
<pre><code>
Attaching package: &#39;matrixStats&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:dplyr&#39;:

    count</code></pre>
<pre><code>
Attaching package: &#39;MatrixGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:matrixStats&#39;:

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
    colWeightedMeans, colWeightedMedians, colWeightedSds,
    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
    rowWeightedSds, rowWeightedVars</code></pre>
<pre><code>Loading required package: GenomicRanges</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    combine, intersect, setdiff, union</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, append, as.data.frame, basename, cbind, colnames,
    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
    union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    first, rename</code></pre>
<pre><code>The following object is masked from &#39;package:tidyr&#39;:

    expand</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    expand.grid, I, unname</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>
Attaching package: &#39;IRanges&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    collapse, desc, slice</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    reduce</code></pre>
<pre><code>Loading required package: GenomeInfoDb</code></pre>
<pre><code>Loading required package: Biobase</code></pre>
<pre><code>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    &#39;browseVignettes()&#39;. To cite Bioconductor, see
    &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>
Attaching package: &#39;Biobase&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:MatrixGenerics&#39;:

    rowMedians</code></pre>
<pre><code>The following objects are masked from &#39;package:matrixStats&#39;:

    anyMissing, rowMedians</code></pre>
<pre><code>
Attaching package: &#39;SummarizedExperiment&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:SeuratObject&#39;:

    Assays</code></pre>
<pre><code>The following object is masked from &#39;package:Seurat&#39;:

    Assays</code></pre>
<pre class="r"><code>organoid.combined.sct.neuron &lt;- readRDS(file=&quot;output/organoid.combined.full.sct.neuron.RDS&quot;) 
organoid.combined.control &lt;- subset(organoid.combined.sct.neuron, subset = stim==&quot;control&quot;)

sce &lt;- SingleCellExperiment(assays=list(counts=organoid.combined.control@assays$RNA@counts))
rm(organoid.combined.sct.neuron)
rm(organoid.combined.control)

#to use only genes that are expressed above a certain count value
# keep_feature &lt;- rowSums(counts(sce)&gt;10) &gt;0
# sum(keep_feature)
# sce &lt;- sce[keep_feature,]

#to randomly subsample genes
keep_feature &lt;- sample(rownames(sce), size=8000)
sce &lt;- sce[keep_feature,]</code></pre>
<p>##Initial diagnostics I’mm using a random subset of 8000 genes to speed computation. Let’s see what zero-inflation looks like here in the whole dataset.</p>
<pre class="r"><code>hippo_diagnostic_plot(sce,
                      show_outliers = TRUE,
                      zvalue_thresh = 2)</code></pre>
<p><img src="figure/hippo.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>##HIPPO clustering</p>
<pre class="r"><code>set.seed(20200321)
sce = hippo(sce, K = 25, z_threshold = 2, outlier_proportion = 0.01, verbose=TRUE)</code></pre>
<pre><code>K = 2..</code></pre>
<pre><code>K = 3..</code></pre>
<pre><code>Warning: Quick-TRANSfer stage steps exceeded maximum (= 523600)</code></pre>
<pre><code>K = 4..</code></pre>
<pre><code>K = 5..</code></pre>
<pre><code>K = 6..</code></pre>
<pre><code>K = 7..</code></pre>
<pre><code>K = 8..</code></pre>
<pre><code>K = 9..</code></pre>
<pre><code>K = 10..</code></pre>
<pre><code>K = 11..</code></pre>
<pre><code>K = 12..</code></pre>
<pre><code>K = 13..</code></pre>
<pre><code>K = 14..</code></pre>
<pre><code>K = 15..</code></pre>
<pre><code>K = 16..</code></pre>
<pre><code>K = 17..</code></pre>
<pre><code>K = 18..</code></pre>
<pre><code>K = 19..</code></pre>
<pre><code>K = 20..</code></pre>
<pre><code>K = 21..</code></pre>
<pre><code>K = 22..</code></pre>
<pre><code>K = 23..</code></pre>
<pre><code>K = 24..</code></pre>
<pre><code>K = 25..</code></pre>
<pre class="r"><code>saveRDS(sce, &quot;output/sce_hippo_test.RDS&quot;)</code></pre>
<p>HIPPO lets us visualize dimensionality reduction for each iteration:</p>
<pre class="r"><code># sce = hippo_dimension_reduction(sce, method=&quot;umap&quot;)
# hippo_umap_plot(sce)

zero_proportion_plot(sce)</code></pre>
<pre><code>Warning: Removed 833 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 43 row(s) containing missing values (geom_path).</code></pre>
<pre><code>Warning: Removed 93 rows containing missing values (geom_label_repel).</code></pre>
<p><img src="figure/hippo.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>NB: at the moment the function to look at iterative splitting does not work. Will be working on this with Karl.</p>
<p>And the resulting heatmap of clustering genes:</p>
<pre class="r"><code>hippo_feature_heatmap(sce, kk = 25,
                      top.n = 10)</code></pre>
<p><img src="figure/hippo.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>We can finally test for DE between clusters:</p>
<pre class="r"><code>sce = hippo_diffexp(sce, 
                  top.n = 5)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] SingleCellExperiment_1.16.0 SummarizedExperiment_1.24.0
 [3] Biobase_2.54.0              GenomicRanges_1.46.0       
 [5] GenomeInfoDb_1.30.0         IRanges_2.28.0             
 [7] S4Vectors_0.32.0            BiocGenerics_0.40.0        
 [9] MatrixGenerics_1.6.0        matrixStats_0.60.1         
[11] viridis_0.6.1               viridisLite_0.4.0          
[13] knitr_1.33                  RColorBrewer_1.1-2         
[15] forcats_0.5.1               stringr_1.4.0              
[17] dplyr_1.0.7                 purrr_0.3.4                
[19] readr_1.4.0                 tidyr_1.1.3                
[21] tibble_3.1.2                ggplot2_3.3.5              
[23] tidyverse_1.3.1             SeuratObject_4.0.2         
[25] Seurat_4.0.3                HIPPO_1.6.0                
[27] workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] readxl_1.3.1           backports_1.2.1        plyr_1.8.6            
  [4] igraph_1.2.6           lazyeval_0.2.2         splines_4.1.0         
  [7] listenv_0.8.0          scattermore_0.7        digest_0.6.27         
 [10] htmltools_0.5.1.1      fansi_0.5.0            magrittr_2.0.1        
 [13] tensor_1.5             cluster_2.1.2          ROCR_1.0-11           
 [16] globals_0.14.0         modelr_0.1.8           spatstat.sparse_2.0-0 
 [19] colorspace_2.0-2       rvest_1.0.0            ggrepel_0.9.1         
 [22] haven_2.4.1            xfun_0.24              crayon_1.4.1          
 [25] RCurl_1.98-1.5         jsonlite_1.7.2         spatstat.data_2.1-0   
 [28] survival_3.2-11        zoo_1.8-9              glue_1.4.2            
 [31] polyclip_1.10-0        gtable_0.3.0           zlibbioc_1.40.0       
 [34] XVector_0.34.0         leiden_0.3.8           DelayedArray_0.20.0   
 [37] future.apply_1.7.0     abind_1.4-5            scales_1.1.1          
 [40] DBI_1.1.1              miniUI_0.1.1.1         Rcpp_1.0.7            
 [43] xtable_1.8-4           reticulate_1.20        spatstat.core_2.2-0   
 [46] htmlwidgets_1.5.3      httr_1.4.2             ellipsis_0.3.2        
 [49] ica_1.0-2              farver_2.1.0           pkgconfig_2.0.3       
 [52] sass_0.4.0             uwot_0.1.10            dbplyr_2.1.1          
 [55] deldir_0.2-10          utf8_1.2.1             labeling_0.4.2        
 [58] tidyselect_1.1.1       rlang_0.4.11           reshape2_1.4.4        
 [61] later_1.2.0            cellranger_1.1.0       munsell_0.5.0         
 [64] tools_4.1.0            cli_3.0.0              generics_0.1.0        
 [67] broom_0.7.8            ggridges_0.5.3         evaluate_0.14         
 [70] fastmap_1.1.0          yaml_2.2.1             goftest_1.2-2         
 [73] fs_1.5.0               fitdistrplus_1.1-5     RANN_2.6.1            
 [76] pbapply_1.4-3          future_1.21.0          nlme_3.1-152          
 [79] whisker_0.4            mime_0.11              xml2_1.3.2            
 [82] rstudioapi_0.13        compiler_4.1.0         plotly_4.9.4.1        
 [85] png_0.1-7              spatstat.utils_2.2-0   reprex_2.0.0          
 [88] bslib_0.2.5.1          stringi_1.6.2          highr_0.9             
 [91] lattice_0.20-44        Matrix_1.3-3           vctrs_0.3.8           
 [94] pillar_1.6.1           lifecycle_1.0.0        spatstat.geom_2.2-0   
 [97] lmtest_0.9-38          jquerylib_0.1.4        RcppAnnoy_0.0.18      
[100] data.table_1.14.0      cowplot_1.1.1          bitops_1.0-7          
[103] irlba_2.3.3            httpuv_1.6.1           patchwork_1.1.1       
[106] R6_2.5.0               promises_1.2.0.1       KernSmooth_2.23-20    
[109] gridExtra_2.3          parallelly_1.26.1      codetools_0.2-18      
[112] MASS_7.3-54            assertthat_0.2.1       rprojroot_2.0.2       
[115] withr_2.4.2            sctransform_0.3.2      GenomeInfoDbData_1.2.7
[118] mgcv_1.8-35            parallel_4.1.0         hms_1.1.0             
[121] grid_4.1.0             rpart_4.1-15           rmarkdown_2.9         
[124] Rtsne_0.15             git2r_0.28.0           shiny_1.6.0           
[127] lubridate_1.7.10      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
