<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Home</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Organoid_Stress_Pilot</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/bumans/Organoid_Stress_Pilot">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Home</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-08
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Organoid_Stress_Pilot/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombumansOrganoidStressPilottree33d0b074aa4e6e095804dbcf979e0d479b04ed51targetblank33d0b07a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bumans/Organoid_Stress_Pilot/tree/33d0b074aa4e6e095804dbcf979e0d479b04ed51" target="_blank">33d0b07</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombumansOrganoidStressPilottree33d0b074aa4e6e095804dbcf979e0d479b04ed51targetblank33d0b07a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/bumans/Organoid_Stress_Pilot/tree/33d0b074aa4e6e095804dbcf979e0d479b04ed51" target="_blank">33d0b07</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store
    Ignored:    output/4vs_avg-expression.png
    Ignored:    output/4vs_logfc.png
    Ignored:    output/4vs_pvals.png
    Ignored:    output/de_clusters_interaction.Rmd
    Ignored:    output/de_clusters_interaction2.Rmd
    Ignored:    output/fitmm0.neuron.RDS
    Ignored:    output/fitmm1.neuron.RDS
    Ignored:    output/fitmm2.neuron.RDS
    Ignored:    output/fitmm3.neuron.RDS
    Ignored:    output/fitmm4.neuron.RDS
    Ignored:    output/fitmm5.neuron.RDS
    Ignored:    output/fitmm6.neuron.RDS
    Ignored:    output/fitmm7.neuron.RDS
    Ignored:    output/fitmm8.neuron.RDS
    Ignored:    output/fitmmall.neuron.RDS
    Ignored:    output/genes_MT.png
    Ignored:    output/genes_cells.png
    Ignored:    output/genes_umi-cluster.png
    Ignored:    output/genes_umi.png
    Ignored:    output/interaction_varpart.csv
    Ignored:    output/merged_full_dataset.RDS
    Ignored:    output/neuron0_oxidation_table.csv
    Ignored:    output/neuron1_oxidation_table.csv
    Ignored:    output/neuron2_oxidation_table.csv
    Ignored:    output/neuron3_oxidation_table.csv
    Ignored:    output/neuron4_oxidation_table.csv
    Ignored:    output/neuron5_oxidation_table.csv
    Ignored:    output/neuron6_oxidation_table.csv
    Ignored:    output/neuron7_oxidation_table.csv
    Ignored:    output/neuron8_oxidation_table.csv
    Ignored:    output/neuronall_oxidation_table.csv
    Ignored:    output/organoid.combined.full.sct.neuron.RDS
    Ignored:    output/organoid.combined.sct.neuron.markers.RDS
    Ignored:    output/organoid.combined.sct.neuron.unstim.markers.RDS
    Ignored:    output/organoid.combined.sct03.RDS
    Ignored:    output/organoid.combined.sct05.RDS
    Ignored:    output/organoid.combined.sct1.RDS
    Ignored:    output/organoid_combined_sct_neuron_resolution_0.02.png
    Ignored:    output/organoid_combined_sct_neuron_resolution_0.05.png
    Ignored:    output/organoid_combined_sct_neuron_resolution_0.1.png
    Ignored:    output/organoid_combined_sct_neuron_resolution_0.3.png
    Ignored:    output/organoid_combined_sct_neuron_resolution_0.5.png
    Ignored:    output/resolution_test.csv
    Ignored:    output/resolution_test_pff.csv
    Ignored:    output/sce_hippo_test.RDS

Untracked files:
    Untracked:  analysis/cell_cycle_full.Rmd
    Untracked:  analysis/topics.Rmd
    Untracked:  data/Snakefile_cellranger
    Untracked:  data/all.vcf.bed
    Untracked:  data/australia.rds
    Untracked:  data/cellranger.38.dict
    Untracked:  data/cluster.json
    Untracked:  data/config.yaml
    Untracked:  data/environment.yaml
    Untracked:  data/exclude_fulldata.rds
    Untracked:  data/human.unsorted.vcf
    Untracked:  data/human.vcf
    Untracked:  data/human.vcf.bed
    Untracked:  data/library9.rds
    Untracked:  data/log/
    Untracked:  data/mysterycell.rds
    Untracked:  data/neuron.control.sce.RDS
    Untracked:  data/organoid.combined.sct03.neuron.RDS
    Untracked:  data/organoid_combined_sct03.RDS
    Untracked:  data/reheaded.vcf
    Untracked:  data/snakelog.out
    Untracked:  data/snakemake_cellranger.batch
    Untracked:  data/sorted.vcf
    Untracked:  data/submit.sh
    Untracked:  data/unsorted.vcf
    Untracked:  data/unsorted_header
    Untracked:  data_S2/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML (<code>docs/index.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bumans/Organoid_Stress_Pilot/blob/33d0b074aa4e6e095804dbcf979e0d479b04ed51/analysis/index.Rmd" target="_blank">33d0b07</a>
</td>
<td>
Ben Umans
</td>
<td>
2022-02-08
</td>
<td>
wflow_publish(files = c(“analysis/hippo.Rmd”, “analysis/index.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bumans/Organoid_Stress_Pilot/cd56fef278b7da46fe594c4a8a9b46dd8f12a389/docs/index.html" target="_blank">cd56fef</a>
</td>
<td>
Ben Umans
</td>
<td>
2021-08-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bumans/Organoid_Stress_Pilot/blob/2b88108674514c625c892dd3196dcc740e2f93a8/analysis/index.Rmd" target="_blank">2b88108</a>
</td>
<td>
Ben Umans
</td>
<td>
2021-08-02
</td>
<td>
wflow_publish(files = c(“analysis/index.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bumans/Organoid_Stress_Pilot/65cfdc535cf9d00fefc85dc5fa8e8fda3ec828f6/docs/index.html" target="_blank">65cfdc5</a>
</td>
<td>
Ben Umans
</td>
<td>
2021-07-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bumans/Organoid_Stress_Pilot/blob/1722f300fa1f927b33833b267b03b1995f89673a/analysis/index.Rmd" target="_blank">1722f30</a>
</td>
<td>
Ben Umans
</td>
<td>
2021-07-29
</td>
<td>
Initial publish to build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bumans/Organoid_Stress_Pilot/e24a33c3872be3b5a04e8a83c2ac9456d48612c4/docs/index.html" target="_blank">e24a33c</a>
</td>
<td>
Ben Umans
</td>
<td>
2021-07-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bumans/Organoid_Stress_Pilot/blob/5690f09288c5494b8d6646c2d3e2e9dbf72ca83d/analysis/index.Rmd" target="_blank">5690f09</a>
</td>
<td>
Ben Umans
</td>
<td>
2021-07-27
</td>
<td>
wflow_publish(files = “analysis/index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bumans/Organoid_Stress_Pilot/a0305dd7983946aa7f6401de18caca09d3fa04ba/docs/index.html" target="_blank">a0305dd</a>
</td>
<td>
Ben Umans
</td>
<td>
2021-07-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bumans/Organoid_Stress_Pilot/blob/957c1dc1005c2e26120f9d9ea339eeea0b5945cb/analysis/index.Rmd" target="_blank">957c1dc</a>
</td>
<td>
Ben Umans
</td>
<td>
2021-07-16
</td>
<td>
wflow_publish(files = c(“analysis/remove-mesenchyme.Rmd”, “analysis/index.Rmd”))
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bumans/Organoid_Stress_Pilot/blob/d97e0a55536a926b6cadf2c343ee2cd67b68e736/analysis/index.Rmd" target="_blank">d97e0a5</a>
</td>
<td>
Ben Umans
</td>
<td>
2021-07-14
</td>
<td>
wflow_publish(files = c(“analysis/index.Rmd”, “analysis/combine_experiments.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bumans/Organoid_Stress_Pilot/fb024f59f543969f9166c1527da7991dd7053895/docs/index.html" target="_blank">fb024f5</a>
</td>
<td>
Ben Umans
</td>
<td>
2021-07-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bumans/Organoid_Stress_Pilot/blob/b70cd9ef06095499611a9a1e48e2f85cfcdb1b6f/analysis/index.Rmd" target="_blank">b70cd9e</a>
</td>
<td>
Ben Umans
</td>
<td>
2021-07-08
</td>
<td>
wflow_publish(“analysis/index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bumans/Organoid_Stress_Pilot/40ae463bc2afcc36af5546ad205c26c4811405df/docs/index.html" target="_blank">40ae463</a>
</td>
<td>
Ben Umans
</td>
<td>
2021-07-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bumans/Organoid_Stress_Pilot/blob/4cb6d4f2fa063d5149d09eaced92c56194aad2ef/analysis/index.Rmd" target="_blank">4cb6d4f</a>
</td>
<td>
Ben Umans
</td>
<td>
2021-07-08
</td>
<td>
wflow_publish(c(“analysis/index.Rmd”, “analysis/precheck.Rmd”))
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bumans/Organoid_Stress_Pilot/blob/a57ca2293b0882c5737370ae79a9eec422e22a9b/analysis/index.Rmd" target="_blank">a57ca22</a>
</td>
<td>
Ben Umans
</td>
<td>
2021-07-05
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>This is the <code>workflowr</code> project for analysis of single-cell RNA sequencing from my organoid pilot project. Three individuals were used to differentiate cerebral organoids in triplicate, which were then left as controls or stimulated before single-cell (10x) data were collected. A total of 11 libraries were constructed, of which 9 are immediately relevant to this project.</p>
<p>###Shallow sequencing data analysis Initially, we sequenced these libraries with an aim to collect ~50,000 reads/cell in order to assess cell type composition. To <a href="precheck.html">start</a>, I used one of the control libraries to check the quality of the data and its correspondence to previous datasets.</p>
<p><a href="combine_experiments.html">Next</a>, I used SCTransform-ed versions of all datasets to create an integrated Seurat object for clustering and expression analysis. In this document I also look for expression of a few stress-related genes and begin to use the <code>variancePartition</code> package to assess reproducibility and the factors contributing to gene expression variation in my dataset. I also include suggestions from KB and YG after an initial discussion of these findings.</p>
<p><a href="variance-estimation.html">To explore their suggestions further</a>, I looked at the effect or running the analysis on separated clusters and downsampling cell numbers.</p>
<p>Not all of the cells collected during this experiment appear to fall along the neural development trajectory. In order to remove overtly different cells, I subsetted out the “mesenchyme” and reclustered using the remaining cells <a href="remove-mesenchyme.html">here</a>.</p>
<p><a href="de.html">Next</a>, I used this filtered dataset to test for differential expression across stimulus conditions.</p>
<p>###Full dataset After completing these analyses, we sent these sequencing libraries back for further sequencing using a NovaSeq S2 chip, with the library composition rebalanced to yield equal numbers of reads across all samples (which will yield slightly different numbers of reads/cell owing to differing capture rates of cells between replicates). Upon receipt of these new data, the combined fastq files from the first and second sequencing batches were processed using <code>cellranger</code> and <code>demuxlet</code> and then filtered and <a href="merge_full_data.html">merged</a> to form integrated datasets.</p>
<p>From these datasets, I attempted two forms of analysis. First, <a href="de_full.html">differential expression</a>, going cluster by cluster. Second, <a href="variance_estimation_full.html">variance decomposition</a>, using what is essentially an ANOVA framework implemented in the <code>variancePartition</code> package. I also tested the effects of different clustering parameters on the differential expression <a href="de_parameter_effects.html">results</a>.</p>
<p>After Mengjie Chen presented some information about her pre-processing pipeline HIPPO, I attempted to <a href="hippo.html">implement</a> this using my data in order to get an indication of how it compares with Seurat’s results. This is an ongoing work in progress.</p>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
