<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Ben Umans" />

<meta name="date" content="2021-07-28" />

<title>merge_full_data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Organoid_Stress_Pilot</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/bumans/Organoid_Stress_Pilot">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">merge_full_data</h1>
<h4 class="author">Ben Umans</h4>
<h4 class="date">2021-07-28</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-07-31
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Organoid_Stress_Pilot/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210705code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210705)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210705code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210705)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombumansOrganoidStressPilottree5d8afa5b0840cd65c8111901f4455b45b85abe82targetblank5d8afa5a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bumans/Organoid_Stress_Pilot/tree/5d8afa5b0840cd65c8111901f4455b45b85abe82" target="_blank">5d8afa5</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombumansOrganoidStressPilottree5d8afa5b0840cd65c8111901f4455b45b85abe82targetblank5d8afa5a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/bumans/Organoid_Stress_Pilot/tree/5d8afa5b0840cd65c8111901f4455b45b85abe82" target="_blank">5d8afa5</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    data/.DS_Store
    Ignored:    data/fastq/
    Ignored:    output/.DS_Store
    Ignored:    output/YG-BU-01_human/
    Ignored:    output/YG-BU-02_human/
    Ignored:    output/YG-BU-03_human/
    Ignored:    output/YG-BU-04_human/
    Ignored:    output/YG-BU-05_human/
    Ignored:    output/YG-BU-06_human/
    Ignored:    output/YG-BU-07_human/
    Ignored:    output/YG-BU-08_human/
    Ignored:    output/YG-BU-09_human/
    Ignored:    output/YG-BU-10_human/
    Ignored:    output/YG-BU-11_human/
    Ignored:    output/merged_full_dataset.RDS
    Ignored:    output/organoid.combined.full.sct.neuron.RDS
    Ignored:    output/organoid.combined.sct.neuron.markers.RDS
    Ignored:    output/organoid.combined.sct03.RDS
    Ignored:    output/organoid.combined.sct05.RDS
    Ignored:    output/organoid.combined.sct1.RDS

Untracked files:
    Untracked:  data/Snakefile_cellranger
    Untracked:  data/all.vcf.bed
    Untracked:  data/australia.rds
    Untracked:  data/cellranger.38.dict
    Untracked:  data/cluster.json
    Untracked:  data/config.yaml
    Untracked:  data/environment.yaml
    Untracked:  data/exclude_fulldata.rds
    Untracked:  data/human.unsorted.vcf
    Untracked:  data/human.vcf
    Untracked:  data/human.vcf.bed
    Untracked:  data/library9.rds
    Untracked:  data/log/
    Untracked:  data/mysterycell.rds
    Untracked:  data/organoid.combined.sct03.neuron.RDS
    Untracked:  data/organoid_combined_sct03.RDS
    Untracked:  data/reheaded.vcf
    Untracked:  data/snakelog.out
    Untracked:  data/snakemake_cellranger.batch
    Untracked:  data/sorted.vcf
    Untracked:  data/submit.sh
    Untracked:  data/unsorted.vcf
    Untracked:  data/unsorted_header
    Untracked:  data_S2/

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/index.Rmd
    Modified:   analysis/precheck.Rmd
    Modified:   analysis/remove-mesenchyme.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/merge_full_data.Rmd</code>) and HTML (<code>docs/merge_full_data.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bumans/Organoid_Stress_Pilot/blob/5d8afa5b0840cd65c8111901f4455b45b85abe82/analysis/merge_full_data.Rmd" target="_blank">5d8afa5</a>
</td>
<td>
Ben Umans
</td>
<td>
2021-07-30
</td>
<td>
Building all files for initial upload
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>I have now received more extensive sequencing data from these libraries and run the combined data (from the S1 and S2 chips) through <code>cellranger</code> together to generate combined data, and through <code>demuxlet</code> to resolve each library into the 3 constituent individuals. Here, I will combine the (now much larger) Seurat objects like before, integrate them, and create both a combined total object and a combined neural-specific object.</p>
</div>
<div id="combined-seurat-objects" class="section level2">
<h2>Combined Seurat objects</h2>
<p>Helpfully, the <code>demuxlet</code> pipeline we have preprocesses each library with SCTransform and saves a Seurat object, so combining them in much easier.</p>
<pre class="r"><code>pacman::p_load(edgeR, variancePartition, BiocParallel, limma)
library(Seurat)
library(tidyverse)
library(ggplot2)
library(RColorBrewer)
library(cowplot)
library(knitr)</code></pre>
<pre class="r"><code>control1 &lt;- readRDS(file = &quot;data_S2/YG-BU-01_human/obj.rds&quot;)
#replace the names of all individuals with the actual line names
oldnames1 &lt;- control1$individual
newnames1 &lt;- str_replace(oldnames1, &quot;SCM-10&quot;, &quot;28126&quot;) %&gt;% str_replace(&quot;SCM-12&quot;, &quot;20157&quot;) %&gt;% str_replace(&quot;SCM-13&quot;, &quot;23555&quot;)
control1 &lt;- AddMetaData(control1, newnames1, col.name = &quot;cell.line&quot;)
control1$replicate &lt;- &quot;2&quot;
control1$stim &lt;- &quot;control&quot;


oxid1 &lt;- readRDS(file = &quot;data_S2/YG-BU-02_human/obj.rds&quot;)
#replace the names of all individuals with the actual line names
oldnames2 &lt;- oxid1$individual
newnames2 &lt;- str_replace(oldnames2, &quot;SCM-10&quot;, &quot;28126&quot;) %&gt;% str_replace(&quot;SCM-12&quot;, &quot;20157&quot;) %&gt;% str_replace(&quot;SCM-13&quot;, &quot;23555&quot;)
oxid1 &lt;- AddMetaData(oxid1, newnames2, col.name = &quot;cell.line&quot;)
oxid1$replicate &lt;- &quot;2&quot;
oxid1$stim &lt;- &quot;oxidation&quot;

pff1 &lt;- readRDS(file = &quot;data_S2/YG-BU-03_human/obj.rds&quot;)
#replace the names of all individuals with the actual line names
oldnames3 &lt;- pff1$individual
newnames3 &lt;- str_replace(oldnames3, &quot;SCM-10&quot;, &quot;28126&quot;) %&gt;% str_replace(&quot;SCM-12&quot;, &quot;20157&quot;) %&gt;% str_replace(&quot;SCM-13&quot;, &quot;23555&quot;)
pff1 &lt;- AddMetaData(pff1, newnames3, col.name = &quot;cell.line&quot;)
pff1$replicate &lt;- &quot;2&quot;
pff1$stim &lt;- &quot;pff&quot;

control2 &lt;- readRDS(file = &quot;data_S2/YG-BU-06_human/obj.rds&quot;)
#replace the names of all individuals with the actual line names
oldnames6 &lt;- control2$individual
newnames6 &lt;- str_replace(oldnames6, &quot;SCM-10&quot;, &quot;28126&quot;) %&gt;% str_replace(&quot;SCM-12&quot;, &quot;20157&quot;) %&gt;% str_replace(&quot;SCM-13&quot;, &quot;23555&quot;)
control2 &lt;- AddMetaData(control2, newnames6, col.name = &quot;cell.line&quot;)
control2$replicate &lt;- &quot;3&quot;
control2$stim &lt;- &quot;control&quot;


oxid2 &lt;- readRDS(file = &quot;data_S2/YG-BU-07_human/obj.rds&quot;)
#replace the names of all individuals with the actual line names
oldnames7 &lt;- oxid2$individual
newnames7 &lt;- str_replace(oldnames7, &quot;SCM-10&quot;, &quot;28126&quot;) %&gt;% str_replace(&quot;SCM-12&quot;, &quot;20157&quot;) %&gt;% str_replace(&quot;SCM-13&quot;, &quot;23555&quot;)
oxid2 &lt;- AddMetaData(oxid2, newnames7, col.name = &quot;cell.line&quot;)
oxid2$replicate &lt;- &quot;3&quot;
oxid2$stim &lt;- &quot;oxidation&quot;


control3 &lt;- readRDS(file = &quot;data_S2/YG-BU-09_human/obj.rds&quot;)
#replace the names of all individuals with the actual line names
oldnames9 &lt;- control3$individual
newnames9 &lt;- str_replace(oldnames9, &quot;SCM-10&quot;, &quot;28126&quot;) %&gt;% str_replace(&quot;SCM-12&quot;, &quot;20157&quot;) %&gt;% str_replace(&quot;SCM-13&quot;, &quot;23555&quot;)
control3 &lt;- AddMetaData(control3, newnames9, col.name = &quot;cell.line&quot;)
control3$replicate &lt;- &quot;4&quot;
control3$stim &lt;- &quot;control&quot;

oxid3 &lt;- readRDS(file = &quot;data_S2/YG-BU-10_human/obj.rds&quot;)
#replace the names of all individuals with the actual line names
oldnames10 &lt;- oxid3$individual
newnames10 &lt;- str_replace(oldnames10, &quot;SCM-10&quot;, &quot;28126&quot;) %&gt;% str_replace(&quot;SCM-12&quot;, &quot;20157&quot;) %&gt;% str_replace(&quot;SCM-13&quot;, &quot;23555&quot;)
oxid3 &lt;- AddMetaData(oxid3, newnames10, col.name = &quot;cell.line&quot;)
oxid3$replicate &lt;- &quot;4&quot;
oxid3$stim &lt;- &quot;oxidation&quot;

pff3 &lt;- readRDS(file = &quot;data_S2/YG-BU-11_human/obj.rds&quot;)
#replace the names of all individuals with the actual line names
oldnames11 &lt;- pff3$individual
newnames11 &lt;- str_replace(oldnames11, &quot;SCM-10&quot;, &quot;28126&quot;) %&gt;% str_replace(&quot;SCM-12&quot;, &quot;20157&quot;) %&gt;% str_replace(&quot;SCM-13&quot;, &quot;23555&quot;)
pff3 &lt;- AddMetaData(pff3, newnames11, col.name = &quot;cell.line&quot;)
pff3$replicate &lt;- &quot;4&quot;
pff3$stim &lt;- &quot;oxidation&quot;


combined.data &lt;- list(control1, control2, control3, oxid1, oxid2, oxid3, pff1, pff3)

rm(control1, control2, control3, oxid1, oxid2, oxid3, pff1, pff3)</code></pre>
<p>Because each of these datasets was prepared using <code>SCTransform</code> it should be straightforward to now integrate them. Following the <a href="https://satijalab.org/seurat/articles/integration_introduction.html#performing-integration-on-datasets-normalized-with-sctransform-1">guide</a> from the Satija lab for integrating these datasets, I run the following steps:</p>
<pre class="r"><code>features &lt;- SelectIntegrationFeatures(object.list = combined.data, nfeatures = 5000, verbose = F)
combined.data &lt;- PrepSCTIntegration(object.list = combined.data, anchor.features = features)

organoid.anchors &lt;- FindIntegrationAnchors(object.list = combined.data, normalization.method = &quot;SCT&quot;,
    anchor.features = features)</code></pre>
<pre><code>Warning in CheckDuplicateCellNames(object.list = object.list): Some cell
names are duplicated across objects provided. Renaming to enforce unique
cell names.</code></pre>
<pre><code>Finding all pairwise anchors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 11409 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 10721 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 13717 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 11711 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 13454 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 10663 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 13267 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 12681 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 11149 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 9970 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 16234 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 14731 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 11270 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 10846 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 11293 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 10894 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 13226 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 12172 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 11707 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 11115 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 13569 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 11729 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 13473 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 11179 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 23613 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 21580 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 16034 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 14021 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 13611 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 11640 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 13080 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 12460 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 11084 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 9993 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 16385 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 15016 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 14667 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 13917 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 11324 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 10227 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 15696 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 14487 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 12588 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 11458 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 12440 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 11021 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 18379 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 17075 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 13519 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 12186 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 11959 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 10630 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 18284 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 17215 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 13408 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 12039 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre class="r"><code>organoid.combined.sct &lt;- IntegrateData(anchorset = organoid.anchors, normalization.method = &quot;SCT&quot;)</code></pre>
<pre><code>Merging dataset 5 into 3</code></pre>
<pre><code>Extracting anchors for merged samples</code></pre>
<pre><code>Finding integration vectors</code></pre>
<pre><code>Finding integration vector weights</code></pre>
<pre><code>Integrating data</code></pre>
<pre><code>Merging dataset 8 into 6</code></pre>
<pre><code>Extracting anchors for merged samples</code></pre>
<pre><code>Finding integration vectors</code></pre>
<pre><code>Finding integration vector weights</code></pre>
<pre><code>Integrating data</code></pre>
<pre><code>Merging dataset 4 into 3 5</code></pre>
<pre><code>Extracting anchors for merged samples</code></pre>
<pre><code>Finding integration vectors</code></pre>
<pre><code>Finding integration vector weights</code></pre>
<pre><code>Integrating data</code></pre>
<pre><code>Merging dataset 1 into 7</code></pre>
<pre><code>Extracting anchors for merged samples</code></pre>
<pre><code>Finding integration vectors</code></pre>
<pre><code>Finding integration vector weights</code></pre>
<pre><code>Integrating data</code></pre>
<pre><code>Merging dataset 7 1 into 3 5 4</code></pre>
<pre><code>Extracting anchors for merged samples</code></pre>
<pre><code>Finding integration vectors</code></pre>
<pre><code>Finding integration vector weights</code></pre>
<pre><code>Integrating data</code></pre>
<pre><code>Merging dataset 6 8 into 3 5 4 7 1</code></pre>
<pre><code>Extracting anchors for merged samples</code></pre>
<pre><code>Finding integration vectors</code></pre>
<pre><code>Finding integration vector weights</code></pre>
<pre><code>Integrating data</code></pre>
<pre><code>Merging dataset 2 into 3 5 4 7 1 6 8</code></pre>
<pre><code>Extracting anchors for merged samples</code></pre>
<pre><code>Finding integration vectors</code></pre>
<pre><code>Finding integration vector weights</code></pre>
<pre><code>Integrating data</code></pre>
<pre><code>Warning: Adding a command log without an assay associated with it</code></pre>
<pre class="r"><code>DefaultAssay(organoid.combined.sct) &lt;- &quot;integrated&quot;
rm(combined.data)</code></pre>
<div id="evaluate-qc-for-further-filtering" class="section level3">
<h3>Evaluate QC for further filtering</h3>
<p>The pre-processing script already subsetted to less than 15% mitochondrial reads, but this is still pretty liberal.</p>
<pre class="r"><code>FeatureScatter(organoid.combined.sct, feature1 = &quot;nCount_RNA&quot;, feature2 = &quot;percent.mt&quot;) + theme(legend.position = &quot;none&quot;) + ggtitle(&quot;&quot;)</code></pre>
<p><img src="figure/merge_full_data.Rmd/pre-qc-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeatureScatter(organoid.combined.sct, feature1 = &quot;nCount_RNA&quot;, feature2 = &quot;nFeature_RNA&quot;) + theme(legend.position = &quot;none&quot;) + ggtitle(&quot;&quot;)</code></pre>
<p><img src="figure/merge_full_data.Rmd/pre-qc-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeatureScatter(organoid.combined.sct, feature1 = &quot;nFeature_RNA&quot;, feature2 = &quot;percent.mt&quot;) + theme(legend.position = &quot;none&quot;) + ggtitle(&quot;&quot;)</code></pre>
<p><img src="figure/merge_full_data.Rmd/pre-qc-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>organoid.combined.sct@meta.data %&gt;% 
  ggplot(aes(x=percent.mt)) +
  geom_freqpoly() +
  ggtitle(&quot;Mitochondrial read percent per cell&quot;)</code></pre>
<p><img src="figure/merge_full_data.Rmd/pre-qc-4.png" width="672" style="display: block; margin: auto;" /> 15% is not a terrible cutoff, but we could easily do 8% and not lose anything worth missing. Let’s also check the number of genes detected, understanding that with low-pass sequencing there’s a lot less data here.</p>
<pre class="r"><code>organoid.combined.sct@meta.data %&gt;% 
  ggplot(aes(x=nFeature_RNA)) +
  geom_freqpoly() +
  ggtitle(&quot;Number of genes per cell&quot;)</code></pre>
<pre><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="figure/merge_full_data.Rmd/genes-cell-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>We could set a minimum threshold of 2500 genes/cell to be safe. This would certainly hit some clusters, which might actually be synthetic owing to their low gene counts</p>
<pre class="r"><code>organoid.combined.sct &lt;- subset(organoid.combined.sct, subset = percent.mt &lt; 8 &amp; nFeature_RNA &gt; 2500 )</code></pre>
</div>
<div id="dimensionality-reduction-and-clustering" class="section level3">
<h3>Dimensionality reduction and clustering</h3>
<p>Now, I go ahead and perform dimensionality reduction and clustering using the integrated dataset.</p>
<pre class="r"><code>organoid.combined.sct &lt;- RunPCA(organoid.combined.sct, npcs = 100)</code></pre>
<pre><code>PC_ 1 
Positive:  COL1A1, MGP, COL3A1, LGALS1, VIM, NEAT1, COL1A2, SPARC, TIMP1, PTN 
       COL11A1, SFRP2, S100A10, TPM2, TPT1, IFITM3, FN1, S100A11, LDHA, CLU 
       OGN, FOXP2, RPL41, BNC2, PRKG1, IGFBP5, NDUFA4L2, ANXA2, TAGLN, RPLP1 
Negative:  STMN2, CADM2, LRRC7, RBFOX1, NRXN1, MYT1L, SYT1, SLC44A5, CCSER1, DLGAP1 
       CTNNA2, PTPRD, ANKS1B, DSCAM, LRRTM4, DAB1, NKAIN2, ARPP21, TUBB3, GPM6A 
       SOX11, OPCML, NTM, CELF2, DCX, TUBA1A, DLG2, BHLHE22, TMEM108, RTN1 
PC_ 2 
Positive:  COL1A1, MGP, COL3A1, LGALS1, RBFOX1, COL1A2, TIMP1, STMN2, SFRP2, LRRC7 
       SPARC, TPM2, OGN, FN1, FOXP2, KCNQ1OT1, ARPP21, S100A11, BNC2, POSTN 
       NDUFA4L2, DAB1, NELL2, NKAIN2, TAGLN, TPT1, KCNQ5, FBLN1, LDHA, S100A10 
Negative:  AL589740.1, LINC01965, CKB, C1orf61, NKAIN3, KCNH7, PTPRZ1, CENPF, ADGRV1, SYNE2 
       SHROOM3, SOX2, HMGB2, TOP2A, KCNIP4, NPAS3, NUSAP1, DACH1, GFAP, MKI67 
       NFIA, TUBA1B, DIAPH3, GPM6B, ZBTB20, SMC4, MPPED2, DBI, QKI, DLEU2 
PC_ 3 
Positive:  KCNQ3, SOX11, EPHA3, AL589740.1, SEMA3C, UNC5D, DOK6, CASC15, ZNF704, MLLT3 
       PLXNA2, MEIS2, CADM2, SOX4, PANTR1, AUTS2, CUX2, ZNF804A, PLCB1, GRAMD1B 
       DLEU2, SORBS2, SEZ6L, NRXN1, FRMD4A, HES6, PALMD, IGFBPL1, CDH4, TTC28 
Negative:  NRG3, FABP7, PCDH9, CLU, GFAP, PTPRZ1, DLG2, CSMD1, NPAS3, MEF2C 
       GPM6B, GRIN2B, RGS6, CKB, BCAN, METRN, C1orf61, IQCJ-SCHIP1, KAZN, DBI 
       MGAT4C, CSMD3, SLC35F1, GRM5, MDGA2, TTYH1, RYR2, BAALC, KHDRBS2, DAB1 
PC_ 4 
Positive:  C1orf61, GFAP, CLU, GPM6B, BCAN, KCNQ3, NRG1, AL589740.1, SEMA3C, KCNN3 
       BAALC, EPHA3, METRN, TTYH1, ATP1B2, SOX11, CNN3, CKB, AQP4, PTPRZ1 
       NPAS3, SORBS2, EZR, DBI, AP1S2, GPR158, MEIS2, UNC5D, ID4, CRYAB 
Negative:  TOP2A, HMGB2, MKI67, NUSAP1, CENPF, TPX2, DIAPH3, SMC4, ASPM, KNL1 
       DLG2, CDK1, PRC1, DLGAP5, UBE2C, CENPE, APOLD1, GTSE1, KIF11, TUBA1B 
       COL1A1, KIF18B, NDC80, PTTG1, SPC25, PBK, BIRC5, KIF15, HIST1H4C, RRM2 
PC_ 5 
Positive:  ROBO2, CNTNAP5, SOX4, DCC, NRXN3, BCL11B, DPP10, RALYL, GRIK2, GRM5 
       NLGN1, MARCH1, PCDH9, CNTNAP2, DOK6, TRPM3, CSMD3, PEX5L, MDGA2, DSCAML1 
       KCNIP4, SORCS1, KAZN, NFIA, ANK3, KCNH7, NEGR1, GRIA2, SLC24A2, SOX5 
Negative:  SATB2, SLC44A5, FABP7, CUX2, NKAIN2, HS6ST3, TAFA1, DLGAP2, RBFOX1, BHLHE22 
       PLXNA4, DPYD, FGF12, MCTP1, CCSER1, MGAT4C, SYT4, DAB1, SATB2-AS1, TENM4 
       ARPP21, KIF26B, ZBTB18, CDH4, DOK5, LRRC7, MIR137HG, PTPRZ1, SNTG1, CLU </code></pre>
<pre class="r"><code>organoid.combined.sct &lt;- RunUMAP(organoid.combined.sct, reduction = &quot;pca&quot;, dims = 1:100)</code></pre>
<pre><code>Warning: The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric
To use Python UMAP via reticulate, set umap.method to &#39;umap-learn&#39; and metric to &#39;correlation&#39;
This message will be shown once per session</code></pre>
<pre><code>04:02:17 UMAP embedding parameters a = 0.9922 b = 1.112</code></pre>
<pre><code>04:02:17 Read 50978 rows and found 100 numeric columns</code></pre>
<pre><code>04:02:17 Using Annoy for neighbor search, n_neighbors = 30</code></pre>
<pre><code>04:02:17 Building Annoy index with metric = cosine, n_trees = 50</code></pre>
<pre><code>0%   10   20   30   40   50   60   70   80   90   100%</code></pre>
<pre><code>[----|----|----|----|----|----|----|----|----|----|</code></pre>
<pre><code>**************************************************|
04:02:31 Writing NN index file to temp file /tmp/RtmpbzcQQZ/file12eb6238795ff
04:02:31 Searching Annoy index using 1 thread, search_k = 3000
04:02:50 Annoy recall = 100%
04:02:54 Commencing smooth kNN distance calibration using 1 thread
04:02:56 Initializing from normalized Laplacian + noise
04:03:03 Commencing optimization for 200 epochs, with 2357470 positive edges
04:04:02 Optimization finished</code></pre>
<pre class="r"><code>FeaturePlot(organoid.combined.sct, features = c(&quot;rna_PAX6&quot;, &quot;rna_MAP2&quot;), max.cutoff = 50)</code></pre>
<p><img src="figure/merge_full_data.Rmd/integrated-umap-plots-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(organoid.combined.sct, reduction = &quot;umap&quot;, group.by = &quot;stim&quot;)</code></pre>
<p><img src="figure/merge_full_data.Rmd/integrated-umap-plots-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(organoid.combined.sct, reduction = &quot;umap&quot;, group.by = &quot;replicate&quot;)</code></pre>
<p><img src="figure/merge_full_data.Rmd/integrated-umap-plots-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>And now we can look for clusters:</p>
<pre class="r"><code>organoid.combined.sct &lt;- FindNeighbors(organoid.combined.sct, dims=1:100)
organoid.combined.sct &lt;- FindClusters(organoid.combined.sct, resolution=0.3)</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 50978
Number of edges: 2509805

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9473
Number of communities: 17
Elapsed time: 15 seconds</code></pre>
<pre class="r"><code>DimPlot(organoid.combined.sct, reduction = &quot;umap&quot;) + ggtitle(&quot;resolution=0.3&quot;)</code></pre>
<p><img src="figure/merge_full_data.Rmd/integrated-clusters-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>This is a good time to save this output.</p>
<pre class="r"><code>saveRDS(organoid.combined.sct, file = &quot;output/merged_full_dataset.RDS&quot;)</code></pre>
</div>
</div>
<div id="filter-to-just-neurons" class="section level2">
<h2>Filter to just neurons</h2>
<p>Once again, I want to check that that large cluster isn’t something I want to hold onto. Markers that Genevieve uses for chondrogenic differentiation include:</p>
<pre class="r"><code>FeaturePlot(organoid.combined.sct, features = c(&quot;COL2A1&quot;, &quot;SOX5&quot;, &quot;SOX6&quot;, &quot;ACAN&quot;))</code></pre>
<p><img src="figure/merge_full_data.Rmd/chondro-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Yikes!! One pole of this cell cluster still looks pretty mesenchymal, with particularly high expression of collagen 2 and aggregan, which could mean any number of things.</p>
<p>At the other end of this cluster I definitely saw expression of some glial genes, for instance:</p>
<pre class="r"><code>FeaturePlot(organoid.combined.sct, features = c(&quot;APOE&quot;))</code></pre>
<p><img src="figure/merge_full_data.Rmd/glia-cluster-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>What could be going on here? Neural crest that started to differentiate? Consider looking at the CNCC marker trajectory that Sara Prescott worked out for her human-chimp paper, which was mostly designed for FACS but found some good markers (CD99, CD266(TNFRSF12A), CD10(MME), CD105(ENG), p75(NGFR)).</p>
<pre class="r"><code>FeaturePlot(organoid.combined.sct, features = c(&quot;sct_CD99&quot;, &quot;sct_TNFRSF12A&quot;, &quot;sct_MME&quot;, &quot;sct_ENG&quot;, &quot;sct_NGFR&quot;))</code></pre>
<p><img src="figure/merge_full_data.Rmd/CNCC-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(organoid.combined.sct, features = c(&quot;sct_SOX10&quot;, &quot;sct_TUBB3&quot;))</code></pre>
<p><img src="figure/merge_full_data.Rmd/CNCC-2.png" width="672" style="display: block; margin: auto;" /> There’s pretty good expression of all of these in the non-chondrogenic lobe of the cluster. BUT, SOX10 is strongly expressed in that smaller separate cluster, which would tend to indicate those are bona fide NCC. Hmm!</p>
<p>Any chance that these cells are microglia?</p>
<pre class="r"><code>FeaturePlot(organoid.combined.sct, features = c(&quot;sct_MMP9&quot;, &quot;sct_SALL1&quot;, &quot;sct_P2RY12&quot;, &quot;sct_TMEM119&quot;, &quot;sct_CX3CR1&quot;))</code></pre>
<p><img src="figure/merge_full_data.Rmd/microglia-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(organoid.combined.sct, features = c(&quot;sct_ITGAM&quot;, &quot;sct_AIF1&quot;), max.cutoff = 0.5)</code></pre>
<p><img src="figure/merge_full_data.Rmd/microglia-2.png" width="672" style="display: block; margin: auto;" /> There’s some expression of MMP9, TMEM119, and a tiny bit of IBA1 throughout the cluster, but not in the APOE-positive area and not in the COL2A1/ACAN area.</p>
<p>This cluster is probably not homogenously any one cell type, but it’s not neuronal.</p>
<pre class="r"><code># p1 &lt;- DimPlot(organoid.combined.sct)
# select.cells &lt;- CellSelector(plot = p1)
# saveRDS(select.cells, &quot;data/exclude_fulldata.rds&quot;)
select.cells &lt;- readRDS(file = &quot;data/exclude_fulldata.rds&quot;)
organoid.combined.sct.neuron &lt;- subset(organoid.combined.sct, cells = select.cells, invert=TRUE)
DimPlot(organoid.combined.sct, cells.highlight = select.cells)</code></pre>
<p><img src="figure/merge_full_data.Rmd/subset-data-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>saveRDS(organoid.combined.sct.neuron, file = &quot;output/organoid.combined.full.sct.neuron.RDS&quot;)</code></pre>
<p>Now recluster with this reduced dataset</p>
<pre class="r"><code>organoid.combined.sct.neuron &lt;- RunPCA(organoid.combined.sct.neuron, npcs = 100)
organoid.combined.sct.neuron &lt;- RunUMAP(organoid.combined.sct.neuron, reduction = &quot;pca&quot;, dims = 1:100)

DimPlot(organoid.combined.sct.neuron, reduction = &quot;umap&quot;, group.by = &quot;stim&quot;)
DimPlot(organoid.combined.sct.neuron, reduction = &quot;umap&quot;, group.by = &quot;replicate&quot;)

organoid.combined.sct.neuron &lt;- FindNeighbors(organoid.combined.sct.neuron, dims=1:100)
organoid.combined.sct.neuron &lt;- FindClusters(organoid.combined.sct.neuron, resolution=0.3)

DimPlot(organoid.combined.sct.neuron, reduction = &quot;umap&quot;) + ggtitle(&quot;resolution=0.3&quot;)

organoid.combined.sct.neuron.markers &lt;- FindAllMarkers(organoid.combined.sct.neuron, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
saveRDS(organoid.combined.sct.neuron.markers, file=&quot;output/organoid.combined.sct.neuron.markers.RDS&quot;)
organoid.combined.sct.neuron.markers %&gt;% group_by(cluster) %&gt;% top_n(n = 5, wt = avg_logFC) %&gt;% kable()

#saveRDS(organoid.combined.sct.neuron, file = &quot;data/organoid.combined.full.sct.neuron.RDS&quot;)
# organoid.combined.sct.neuron &lt;- readRDS(file = &quot;data/organoid.combined.full.sct.neuron.RDS&quot;)</code></pre>
<p>The relevant output of the above markers and the cluster-labeled UMAP representation:</p>
<pre class="r"><code>organoid.combined.sct.neuron.markers &lt;- readRDS(file=&quot;output/organoid.combined.sct.neuron.markers.RDS&quot;)
DimPlot(organoid.combined.sct.neuron, reduction = &quot;umap&quot;) + ggtitle(&quot;resolution=0.3&quot;)</code></pre>
<p><img src="figure/merge_full_data.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>organoid.combined.sct.neuron.markers %&gt;% group_by(cluster) %&gt;% top_n(n = 5, wt = avg_logFC) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="right">p_val</th>
<th align="right">avg_logFC</th>
<th align="right">pct.1</th>
<th align="right">pct.2</th>
<th align="right">p_val_adj</th>
<th align="left">cluster</th>
<th align="left">gene</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">0.0000000</td>
<td align="right">6.892666</td>
<td align="right">0.893</td>
<td align="right">0.260</td>
<td align="right">0.0000000</td>
<td align="left">0</td>
<td align="left">DPY19L1</td>
</tr>
<tr class="even">
<td align="right">0.0000000</td>
<td align="right">6.468132</td>
<td align="right">0.819</td>
<td align="right">0.298</td>
<td align="right">0.0000000</td>
<td align="left">0</td>
<td align="left">ADRA2A</td>
</tr>
<tr class="odd">
<td align="right">0.0000000</td>
<td align="right">6.284193</td>
<td align="right">0.873</td>
<td align="right">0.377</td>
<td align="right">0.0000000</td>
<td align="left">0</td>
<td align="left">PRKX</td>
</tr>
<tr class="even">
<td align="right">0.0000000</td>
<td align="right">5.669388</td>
<td align="right">0.950</td>
<td align="right">0.476</td>
<td align="right">0.0000000</td>
<td align="left">0</td>
<td align="left">PTPN4</td>
</tr>
<tr class="odd">
<td align="right">0.0000000</td>
<td align="right">6.027549</td>
<td align="right">0.538</td>
<td align="right">0.336</td>
<td align="right">0.0000000</td>
<td align="left">0</td>
<td align="left">AC105916.1</td>
</tr>
<tr class="even">
<td align="right">0.0000000</td>
<td align="right">10.800391</td>
<td align="right">0.821</td>
<td align="right">0.181</td>
<td align="right">0.0000000</td>
<td align="left">1</td>
<td align="left">SYBU</td>
</tr>
<tr class="odd">
<td align="right">0.0000000</td>
<td align="right">8.366472</td>
<td align="right">0.857</td>
<td align="right">0.282</td>
<td align="right">0.0000000</td>
<td align="left">1</td>
<td align="left">GPR85</td>
</tr>
<tr class="even">
<td align="right">0.0000000</td>
<td align="right">7.388974</td>
<td align="right">0.587</td>
<td align="right">0.074</td>
<td align="right">0.0000000</td>
<td align="left">1</td>
<td align="left">CELF2-DT</td>
</tr>
<tr class="odd">
<td align="right">0.0000000</td>
<td align="right">7.292979</td>
<td align="right">0.972</td>
<td align="right">0.436</td>
<td align="right">0.0000000</td>
<td align="left">1</td>
<td align="left">R3HDM1</td>
</tr>
<tr class="even">
<td align="right">0.0000000</td>
<td align="right">7.249213</td>
<td align="right">0.984</td>
<td align="right">0.605</td>
<td align="right">0.0000000</td>
<td align="left">1</td>
<td align="left">ADD2</td>
</tr>
<tr class="odd">
<td align="right">0.0000000</td>
<td align="right">9.614571</td>
<td align="right">0.678</td>
<td align="right">0.466</td>
<td align="right">0.0000000</td>
<td align="left">2</td>
<td align="left">AC053513.1</td>
</tr>
<tr class="even">
<td align="right">0.0000000</td>
<td align="right">8.793939</td>
<td align="right">0.740</td>
<td align="right">0.126</td>
<td align="right">0.0000000</td>
<td align="left">2</td>
<td align="left">CITED1</td>
</tr>
<tr class="odd">
<td align="right">0.0000000</td>
<td align="right">8.688715</td>
<td align="right">0.433</td>
<td align="right">0.099</td>
<td align="right">0.0000000</td>
<td align="left">2</td>
<td align="left">SFRP4</td>
</tr>
<tr class="even">
<td align="right">0.0000000</td>
<td align="right">8.224377</td>
<td align="right">0.487</td>
<td align="right">0.113</td>
<td align="right">0.0000000</td>
<td align="left">2</td>
<td align="left">GAPDH</td>
</tr>
<tr class="odd">
<td align="right">0.0000000</td>
<td align="right">9.378618</td>
<td align="right">0.287</td>
<td align="right">0.176</td>
<td align="right">0.0000000</td>
<td align="left">2</td>
<td align="left">ERO1A</td>
</tr>
<tr class="even">
<td align="right">0.0000000</td>
<td align="right">7.749870</td>
<td align="right">0.847</td>
<td align="right">0.541</td>
<td align="right">0.0000000</td>
<td align="left">3</td>
<td align="left">RAPGEF2</td>
</tr>
<tr class="odd">
<td align="right">0.0000000</td>
<td align="right">7.440431</td>
<td align="right">0.774</td>
<td align="right">0.541</td>
<td align="right">0.0000000</td>
<td align="left">3</td>
<td align="left">DDX3Y</td>
</tr>
<tr class="even">
<td align="right">0.0000000</td>
<td align="right">7.162033</td>
<td align="right">0.604</td>
<td align="right">0.393</td>
<td align="right">0.0000000</td>
<td align="left">3</td>
<td align="left">ERO1B</td>
</tr>
<tr class="odd">
<td align="right">0.0000000</td>
<td align="right">9.436481</td>
<td align="right">0.298</td>
<td align="right">0.141</td>
<td align="right">0.0000000</td>
<td align="left">3</td>
<td align="left">TSPYL2</td>
</tr>
<tr class="even">
<td align="right">0.0000000</td>
<td align="right">7.467664</td>
<td align="right">0.352</td>
<td align="right">0.180</td>
<td align="right">0.0000000</td>
<td align="left">3</td>
<td align="left">IL12RB2</td>
</tr>
<tr class="odd">
<td align="right">0.0000000</td>
<td align="right">10.142848</td>
<td align="right">0.990</td>
<td align="right">0.206</td>
<td align="right">0.0000000</td>
<td align="left">4</td>
<td align="left">FANCI</td>
</tr>
<tr class="even">
<td align="right">0.0000000</td>
<td align="right">9.293580</td>
<td align="right">0.957</td>
<td align="right">0.120</td>
<td align="right">0.0000000</td>
<td align="left">4</td>
<td align="left">BRCA2</td>
</tr>
<tr class="odd">
<td align="right">0.0000000</td>
<td align="right">8.976039</td>
<td align="right">0.940</td>
<td align="right">0.134</td>
<td align="right">0.0000000</td>
<td align="left">4</td>
<td align="left">BLM</td>
</tr>
<tr class="even">
<td align="right">0.0000000</td>
<td align="right">8.781641</td>
<td align="right">0.948</td>
<td align="right">0.081</td>
<td align="right">0.0000000</td>
<td align="left">4</td>
<td align="left">MYBL2</td>
</tr>
<tr class="odd">
<td align="right">0.0000000</td>
<td align="right">8.632552</td>
<td align="right">0.982</td>
<td align="right">0.111</td>
<td align="right">0.0000000</td>
<td align="left">4</td>
<td align="left">RAD51AP1</td>
</tr>
<tr class="even">
<td align="right">0.0000000</td>
<td align="right">12.531469</td>
<td align="right">0.711</td>
<td align="right">0.013</td>
<td align="right">0.0000000</td>
<td align="left">5</td>
<td align="left">SERPINA3</td>
</tr>
<tr class="odd">
<td align="right">0.0000000</td>
<td align="right">12.460638</td>
<td align="right">0.715</td>
<td align="right">0.024</td>
<td align="right">0.0000000</td>
<td align="left">5</td>
<td align="left">WDR38</td>
</tr>
<tr class="even">
<td align="right">0.0000000</td>
<td align="right">12.151465</td>
<td align="right">0.791</td>
<td align="right">0.021</td>
<td align="right">0.0000000</td>
<td align="left">5</td>
<td align="left">C7orf57</td>
</tr>
<tr class="odd">
<td align="right">0.0000000</td>
<td align="right">11.898922</td>
<td align="right">0.693</td>
<td align="right">0.046</td>
<td align="right">0.0000000</td>
<td align="left">5</td>
<td align="left">C11orf88</td>
</tr>
<tr class="even">
<td align="right">0.0000000</td>
<td align="right">12.056751</td>
<td align="right">0.530</td>
<td align="right">0.015</td>
<td align="right">0.0000000</td>
<td align="left">5</td>
<td align="left">MS4A8</td>
</tr>
<tr class="odd">
<td align="right">0.0000000</td>
<td align="right">10.278164</td>
<td align="right">0.997</td>
<td align="right">0.600</td>
<td align="right">0.0000000</td>
<td align="left">6</td>
<td align="left">WASF1</td>
</tr>
<tr class="even">
<td align="right">0.0000000</td>
<td align="right">10.104331</td>
<td align="right">0.859</td>
<td align="right">0.293</td>
<td align="right">0.0000000</td>
<td align="left">6</td>
<td align="left">SATB1</td>
</tr>
<tr class="odd">
<td align="right">0.0000000</td>
<td align="right">8.843072</td>
<td align="right">0.907</td>
<td align="right">0.261</td>
<td align="right">0.0000000</td>
<td align="left">6</td>
<td align="left">MAL2</td>
</tr>
<tr class="even">
<td align="right">0.0000000</td>
<td align="right">8.516076</td>
<td align="right">0.842</td>
<td align="right">0.389</td>
<td align="right">0.0000000</td>
<td align="left">6</td>
<td align="left">HECW2</td>
</tr>
<tr class="odd">
<td align="right">0.0000000</td>
<td align="right">9.887219</td>
<td align="right">0.270</td>
<td align="right">0.055</td>
<td align="right">0.0000000</td>
<td align="left">6</td>
<td align="left">VRK2</td>
</tr>
<tr class="even">
<td align="right">0.0000000</td>
<td align="right">7.257082</td>
<td align="right">0.972</td>
<td align="right">0.404</td>
<td align="right">0.0000000</td>
<td align="left">7</td>
<td align="left">RBPJ</td>
</tr>
<tr class="odd">
<td align="right">0.0000000</td>
<td align="right">6.882316</td>
<td align="right">0.550</td>
<td align="right">0.144</td>
<td align="right">0.0000000</td>
<td align="left">7</td>
<td align="left">GAS2</td>
</tr>
<tr class="even">
<td align="right">0.0000000</td>
<td align="right">6.253624</td>
<td align="right">0.383</td>
<td align="right">0.151</td>
<td align="right">0.0000000</td>
<td align="left">7</td>
<td align="left">AC078881.1</td>
</tr>
<tr class="odd">
<td align="right">0.0000000</td>
<td align="right">6.065486</td>
<td align="right">0.815</td>
<td align="right">0.685</td>
<td align="right">0.0000000</td>
<td align="left">7</td>
<td align="left">PPP1R9A</td>
</tr>
<tr class="even">
<td align="right">0.0000539</td>
<td align="right">7.637421</td>
<td align="right">0.529</td>
<td align="right">0.562</td>
<td align="right">0.2692528</td>
<td align="left">7</td>
<td align="left">GALNT8</td>
</tr>
<tr class="odd">
<td align="right">0.0000000</td>
<td align="right">9.920117</td>
<td align="right">0.955</td>
<td align="right">0.531</td>
<td align="right">0.0000000</td>
<td align="left">8</td>
<td align="left">SCLT1</td>
</tr>
<tr class="even">
<td align="right">0.0000000</td>
<td align="right">9.713181</td>
<td align="right">0.931</td>
<td align="right">0.165</td>
<td align="right">0.0000000</td>
<td align="left">8</td>
<td align="left">PARPBP</td>
</tr>
<tr class="odd">
<td align="right">0.0000000</td>
<td align="right">9.432273</td>
<td align="right">0.977</td>
<td align="right">0.097</td>
<td align="right">0.0000000</td>
<td align="left">8</td>
<td align="left">TROAP</td>
</tr>
<tr class="even">
<td align="right">0.0000000</td>
<td align="right">8.916473</td>
<td align="right">0.940</td>
<td align="right">0.092</td>
<td align="right">0.0000000</td>
<td align="left">8</td>
<td align="left">TTK</td>
</tr>
<tr class="odd">
<td align="right">0.0000000</td>
<td align="right">8.695318</td>
<td align="right">0.894</td>
<td align="right">0.127</td>
<td align="right">0.0000000</td>
<td align="left">8</td>
<td align="left">FAM83D</td>
</tr>
<tr class="even">
<td align="right">0.0000000</td>
<td align="right">6.672513</td>
<td align="right">0.959</td>
<td align="right">0.750</td>
<td align="right">0.0000000</td>
<td align="left">9</td>
<td align="left">SLAIN1</td>
</tr>
<tr class="odd">
<td align="right">0.0000000</td>
<td align="right">6.592629</td>
<td align="right">0.329</td>
<td align="right">0.107</td>
<td align="right">0.0000000</td>
<td align="left">9</td>
<td align="left">TNN</td>
</tr>
<tr class="even">
<td align="right">0.0000000</td>
<td align="right">7.664381</td>
<td align="right">0.718</td>
<td align="right">0.339</td>
<td align="right">0.0000000</td>
<td align="left">9</td>
<td align="left">MTUS1</td>
</tr>
<tr class="odd">
<td align="right">0.0000000</td>
<td align="right">7.321903</td>
<td align="right">0.566</td>
<td align="right">0.286</td>
<td align="right">0.0000000</td>
<td align="left">9</td>
<td align="left">AC090241.2</td>
</tr>
<tr class="even">
<td align="right">0.0000000</td>
<td align="right">7.791721</td>
<td align="right">0.393</td>
<td align="right">0.224</td>
<td align="right">0.0000000</td>
<td align="left">9</td>
<td align="left">RAP2C-AS1</td>
</tr>
<tr class="odd">
<td align="right">0.0000000</td>
<td align="right">7.995611</td>
<td align="right">0.840</td>
<td align="right">0.064</td>
<td align="right">0.0000000</td>
<td align="left">10</td>
<td align="left">GATA3-AS1</td>
</tr>
<tr class="even">
<td align="right">0.0000000</td>
<td align="right">8.192247</td>
<td align="right">0.631</td>
<td align="right">0.041</td>
<td align="right">0.0000000</td>
<td align="left">10</td>
<td align="left">GATA3</td>
</tr>
<tr class="odd">
<td align="right">0.0000000</td>
<td align="right">9.055618</td>
<td align="right">0.652</td>
<td align="right">0.144</td>
<td align="right">0.0000000</td>
<td align="left">10</td>
<td align="left">NR2F2</td>
</tr>
<tr class="even">
<td align="right">0.0000000</td>
<td align="right">12.871942</td>
<td align="right">0.464</td>
<td align="right">0.040</td>
<td align="right">0.0000000</td>
<td align="left">10</td>
<td align="left">SMIM32</td>
</tr>
<tr class="odd">
<td align="right">0.0000000</td>
<td align="right">7.304059</td>
<td align="right">0.764</td>
<td align="right">0.553</td>
<td align="right">0.0000000</td>
<td align="left">10</td>
<td align="left">CDKN2D</td>
</tr>
<tr class="even">
<td align="right">0.0000000</td>
<td align="right">10.992154</td>
<td align="right">0.885</td>
<td align="right">0.108</td>
<td align="right">0.0000000</td>
<td align="left">11</td>
<td align="left">LGR4</td>
</tr>
<tr class="odd">
<td align="right">0.0000000</td>
<td align="right">10.700430</td>
<td align="right">0.344</td>
<td align="right">0.080</td>
<td align="right">0.0001744</td>
<td align="left">11</td>
<td align="left">SEMA5A</td>
</tr>
<tr class="even">
<td align="right">0.0000007</td>
<td align="right">11.301041</td>
<td align="right">0.393</td>
<td align="right">0.145</td>
<td align="right">0.0035772</td>
<td align="left">11</td>
<td align="left">LGI4</td>
</tr>
<tr class="odd">
<td align="right">0.0000141</td>
<td align="right">11.596306</td>
<td align="right">0.262</td>
<td align="right">0.050</td>
<td align="right">0.0704529</td>
<td align="left">11</td>
<td align="left">EGFLAM</td>
</tr>
<tr class="even">
<td align="right">0.0012444</td>
<td align="right">11.650612</td>
<td align="right">0.295</td>
<td align="right">0.117</td>
<td align="right">1.0000000</td>
<td align="left">11</td>
<td align="left">ANO2</td>
</tr>
</tbody>
</table>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] knitr_1.23               cowplot_1.0.0           
 [3] RColorBrewer_1.1-2       forcats_0.5.1           
 [5] stringr_1.4.0            dplyr_1.0.7             
 [7] purrr_0.3.4              readr_1.4.0             
 [9] tidyr_1.1.3              tibble_3.1.2            
[11] tidyverse_1.3.1          Seurat_3.1.3            
[13] BiocParallel_1.18.0      variancePartition_1.14.1
[15] Biobase_2.44.0           BiocGenerics_0.30.0     
[17] scales_1.1.1             foreach_1.5.1           
[19] ggplot2_3.3.5            edgeR_3.26.5            
[21] limma_3.40.6             workflowr_1.6.2         

loaded via a namespace (and not attached):
  [1] readxl_1.3.1        backports_1.2.1     plyr_1.8.6         
  [4] igraph_1.2.4.1      lazyeval_0.2.2      splines_3.6.1      
  [7] listenv_0.7.0       digest_0.6.27       htmltools_0.5.1.1  
 [10] gdata_2.18.0        fansi_0.5.0         magrittr_2.0.1     
 [13] cluster_2.1.0       doParallel_1.0.14   ROCR_1.0-7         
 [16] globals_0.12.4      modelr_0.1.8        RcppParallel_5.1.4 
 [19] prettyunits_1.1.1   colorspace_2.0-2    rvest_1.0.0        
 [22] rappdirs_0.3.1      ggrepel_0.8.1       haven_2.3.1        
 [25] xfun_0.24           crayon_1.4.1        jsonlite_1.7.2     
 [28] lme4_1.1-21         survival_3.2-11     zoo_1.8-6          
 [31] iterators_1.0.13    ape_5.4-1           glue_1.4.2         
 [34] gtable_0.3.0        leiden_0.3.1        future.apply_1.3.0 
 [37] DBI_1.1.0           bibtex_0.4.2        Rcpp_1.0.6         
 [40] metap_1.1           viridisLite_0.3.0   progress_1.2.2     
 [43] reticulate_1.16     rsvd_1.0.1          tsne_0.1-3         
 [46] htmlwidgets_1.5.3   httr_1.4.2          gplots_3.0.1.1     
 [49] ellipsis_0.3.2      ica_1.0-2           farver_2.1.0       
 [52] pkgconfig_2.0.3     uwot_0.1.5          dbplyr_2.1.1       
 [55] locfit_1.5-9.1      utf8_1.1.4          labeling_0.4.2     
 [58] tidyselect_1.1.0    rlang_0.4.11        reshape2_1.4.4     
 [61] later_1.2.0         cellranger_1.1.0    munsell_0.5.0      
 [64] tools_3.6.1         cli_2.5.0           generics_0.1.0     
 [67] pacman_0.5.1        broom_0.7.8         ggridges_0.5.1     
 [70] evaluate_0.14       yaml_2.2.1          npsurv_0.4-0       
 [73] fs_1.3.1            fitdistrplus_1.0-14 caTools_1.17.1.2   
 [76] RANN_2.6.1          pbapply_1.4-0       future_1.14.0      
 [79] nlme_3.1-140        whisker_0.3-2       xml2_1.3.2         
 [82] rstudioapi_0.13     compiler_3.6.1      pbkrtest_0.4-7     
 [85] plotly_4.9.4.1      png_0.1-7           lsei_1.2-0         
 [88] reprex_2.0.0        stringi_1.6.2       highr_0.8          
 [91] RSpectra_0.15-0     lattice_0.20-38     Matrix_1.2-18      
 [94] nloptr_1.2.2.1      vctrs_0.3.8         pillar_1.6.1       
 [97] lifecycle_1.0.0     Rdpack_0.11-0       lmtest_0.9-37      
[100] RcppAnnoy_0.0.12    data.table_1.14.0   bitops_1.0-6       
[103] irlba_2.3.3         gbRd_0.4-11         httpuv_1.6.1       
[106] colorRamps_2.3      R6_2.5.0            promises_1.2.0.1   
[109] KernSmooth_2.23-15  gridExtra_2.3       codetools_0.2-16   
[112] boot_1.3-23         MASS_7.3-51.4       gtools_3.8.1       
[115] assertthat_0.2.1    rprojroot_2.0.2     withr_2.4.2        
[118] sctransform_0.2.0   hms_1.1.0           grid_3.6.1         
[121] minqa_1.2.4         rmarkdown_1.13      Rtsne_0.15         
[124] git2r_0.26.1        lubridate_1.7.10   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
